\chapter{Effects of SVs on gene expression and chromatin
organization in \texorpdfstring{\textit{D. melanogaster}}{D. melanogaster}}
\label{sec:balancer}



The research projects described in previous chapters focused on the technical
aspects of identifying and characterizing \acp{sv}. For the project described in
this chapter I teamed up with \yad, \alek and \eileen to study the functional
impact of \acp{sv}, specifically of large chromosomal rearrangements, in the model
organism \textit{Drosophila melanogaster}. Yet also in this chapter recent
sequencing technologies, notably \hic, were instrumental in the characterization
of \acp{sv}, which was an essential step of the project. All wet lab experiments
described below were performed by \yad with the help of \rebecca.
\alek implemented all \hic- and Capture \hic-related analyses, which are hence
only depicted briefly. All other computational analyses including all figures
are my own work if not explicitly stated differently. Supplementary information
can be found in \cref{sec:suppl_balancer}. At the time of writing the study was
not yet complete - thus the following should be considered a status report of
ongoing work.



\section{Background}
\label{sec:balancer_background}


\subsection{Impact of \texorpdfstring{\acp{sv}}{SVs} in health and disease}

Population-scale studies revealed that \acp{sv} are a major contributor to
genetic variation in the human population \citep{Conrad2010} â€“ in fact they
contribute more base pair differences than \acp{snv} \citep{Sudmant2015}.
Besides their abundance in healthy individuals, \acp{sv} had earlier already
been recognized for their role in a number of diseases, as
\citet{Zhang2009,Weischenfeldt2013,Carvalho2016} review in more detail.

The consequences certain \acp{sv} imply can range from purely molecular
phenotypes, such as altered gene expression, to little severe (e.g. the efficiency of
starch digestion depending on the copy number of \textit{AMY1} \citep{Perry2007})
or more severe phenotypes (e.g. red-green blindness, caused by \acp{cnv} on
chromosome X \citep{Nathans1986}), to disease or to an increased susceptibility
towards a disease (e.g. to complex diseases such as autoimmunity
\citep{Fanciulli2007} or autism \citep{Sebat2007}). Especially \acp{cnv}, the
most-frequently studied type of \sv, have been linked to Mendelian diseases
\citep[see table 2]{Zhang2009} and are frequently involved in cancer
\citep{Beroukhim2010}. Yet the types of \acp{sv} linked to disease are by far
not restricted to localized deletions or duplications: Trisomy 21, as an example
for aneuploidy in general, causes Down syndrome and was among the earliest
identified genetic diseases \citep{Lejeune1959}. Also inversions were identified
as the cause of several Mendelian diseases \citep{Feuk2010}. Recurrent
translocations, often creating gene fusions, are known to be driving many
different cancer types \citep{Mertens2015} and also transposable elements have
been noted be to play a role in cancer \citep{Burns2017}.

The mechanisms by which \acp{sv} can induce these phenotypic consequences are
manifold, too: Breakpoints of \acp{sv} that fall into a gene may cause its loss
of function. Aforementioned gene fusions can create novel chimeras that
potentially exert fundamentally different functions from the original genes
\citep{Mertens2015}. Copy number gains or losses lead to dosage imbalance, with
critical consequences for many cancer-associated genes \citep{Fehrmann2015}, and
a heterozygous deletion or a \loh event can bring recessive mutations into play
that had been masked beforehand.

Furthermore \acp{sv} may impact on the molecular level, e.g. on gene expression,
which can partly be studied even if not associated with a real phenotype. In
the context of \explain{expression quantitiative trail loci}{are genomic loci
with different alleles that influence the expression of (typically a single) genes
not neccessarily in close proximity.} analyses
more and more such effects have lately been found \citep{Sudmant2015,Chiang2017}.
Causative \acp{sv} need not affect the actual gene itself, but could target
regulatory sequences in proximity or elsewhere in the genome instead. This is
typically the case for aforementioned expression quantitative trait loci, yet
the (then very surprising) discovery was already made in 1979 in a row of
heritable blood diseases named thalassaemia \citep{Fritsch1979}.

At last, another mechanism has recently gained attention in which \acp{sv}
neither affect genes nor regulatory sequences, but re-model the
three-dimensional chromatin organization instead. As this is a central aspect of
our study, it is further elaborated in \cref{sec:disrupting_tads}.




\subsection{Three-dimensional chromatin conformation}
\label{sec:chromatin_conformation}


\figuremargin{TAD_schematic_zaugg.jpg}{TAD_schematic}{Schematic of
\texorpdfstring{\aclp{tad}}{topologically associating domains}}{Characteristic
triangles in a mammalian \hic map (top) belong to spatial domains of DNA
inside the nucleus (bottom). Figure taken from \citep{Ruiz-Velasco2017} licensed
under \acl{ccby4}.}





\subsection{Consequences of disrupting chromatin conformation}
\label{sec:disrupting_tads}






\section{Design of the study}
\label{sec:balancer_study_design}

In this study we set out to understand how genomic aberrations such as the
disruption of \acp{tad} caused by large chromosomal rearrangements and other
\acp{sv} can affect the regulation of surrounding genes. In contrast to previous
studies (as outlined in \cref{sec:disrupting_tads}) we explicitly
focused on a phenotypically healthy system where no dominating effects, but
rather modest changes in molecular phenotypes were to be expected. We think this
work will complement previous studies, which had investigated rather
pathological situations, and open a new perspective on the functional impact of
\acp{tad}. Moreover, instead of selecting a single locus we aimed at testing
multiple rearrangements in a genome-wide fashion to gain a broader understanding
on the generality of such effects.




\subsection{Balancer chromosomes carry large genomic rearrangements}
\label{sec:balancer_balancers}

We found a suitable model system for this task in so-called balancer chromosomes
of \textit{Drosophila melanogaster}. These naturally derived chromosomes have a
long tradition in fly genetics and are frequently used as a tool to keep
recessive lethal mutations from being lost from the population. Three relevant
features allow them to perform this task: (1) they carry recessive lethal
mutations so that homozygous offspring will not live, effectively balancing the
alleles in a population. This further requires that haplotypes remain intact and
are not exchanged with their homologue, so consequently (2) balancer chromosomes
suppress recombination by disrupting homologous pairing. This is achieved by the
introduction of multiple large inversions via, for example,
\explain{X-ray mutagenesis}{Balancer chromosomes date back to the work of
    Hermann Joseph Muller, who in the 1920s studied mutagenesis through X-ray
    radiation \citep{Muller1928} and received a Nobel prize in 1948\footnotemark}.
\footnotetext{See more in a blog post from Laurence A. Moran:
    \url{https://sandwalk.blogspot.de/2008/10/hermann-muller-invented-balancer.html}}
(3) Balancer chromosomes include dominant marker alleles so that carriers can be
readily detected based on their phenotype in adult stage. Such balancer
chromosomes are nowadays available for all the major chromosomes 2, 3, and X.
Interestingly, despite their common usage only recently some balancer
chromosomes had been characterized via whole-genome sequencing \citep{Miller2016}.

We chose balancer chromosomes for chromosome 2 and 3, namely \explain{\cyo}{
    chromosome 2 balancer, derived from original work of \citet{Oster1956},
    which carries the \textit{Cy} allele, hence showing a characteristic
    phenotype of curly wings in adult flies \marginfig{dm_cyo.pdf}}
and TM3 \citep{Tinderholt1960} to create a double balancer line, named $F_1$.
See \cref{fig:balancer_chroms} for an overview of the \textit{D. melanogaster}
genome of both the wild type line and the double balancer line utilized in this
study. For both balancer chromosomes the major rearrangements had been
previously characterized by karyotyping (see
note\footnoteref{footnote:balancer_karyotype} and
\cref{fig:balancer_hiC_rearrangements}). The idea behind using two balancer
chromosomes was to effectively increase the number of genomic rearrangements to
be studied, which was an important consideration early on in the project.
Together both balancer chromosomes carry 16 breakpoints of large, partly
centromere-spanning inversions and we expected them to contain additional
sub-microscopic \acp{sv} related to their mutagenic origin. Accordingly, one of
the first step of the study was a deep characterization of the mutational
landscape  of the balancer chromosomes, which is presented in
\cref{sec:balancer_mutational_landscape}.





\subsection{Studying \texorpdfstring{\textit{cis}}{cis}-regulatory effects via
    \texorpdfstring{\acl{ase}}{allele-specific expression}}
\label{sec:balancer_ase_motivation}

To study the effects of \acp{sv} on gene regulation we wanted to compare global
gene expression of the balancer chromosomes against expression of the wild type
homologues. This could principally be done by comparing two different fly
strains, one carrying genomic rearrangements and one without. However, while
this is also not possible using balancer chromosomes that cannot be
in a homozygous state, this approach would make it difficult to distinguish
\textit{cis}-regulation, i.e. the effect of alterations of the DNA itself, from
\textit{trans}-regulation such as an altered expression of some transcription
factor. This is why another central foundation of this study was to compare wild
type chromosomes and balancer chromosomes within the same nucleus via \acf{ase}.
This is achieved by measuring gene expression separately for both alleles, where
alleles differ by naturally occurring variation, notably \acp{snv}.
Since we were able to resolve complete haplotypes in this study, we could
measure expression not only at single alleles but could assign alleles to
the balancer- or wild type chromosomes and can thus distinguish up- and
down-regulated genes of the balancer chromosomes in respect to the wild type
expression. \Crefrange{sec:balancer_maternal_rna}{sec:balancer_ase_impl} provide
more detail on the procedure.
We assume that the vast majority of \acp{snv} simply tag alleles but do not have
an effect on their regulation, or that this effect is negligible in comparison
to larger rearrangements. This might not be true in all cases and it can indeed
be difficult to assign a change in expression to the respective functional
mutation.



\figuretwocolumns[0.38]
    {balancer_chroms.pdf}{balancer_chroms}{Genome overview}{Schematic of the
        four chromosomes of (female) \textit{D. melanogaster} flies of the wild
        type line $F_0$ and of the double balancer cross $F_1$. \Ac{cyo} red,
        \ac{tm3} blue. }
    {balancer_crossing_scheme.pdf}{balancer_crossing_scheme}{Crossing scheme}{
        The fly lines used in this study are derived from a homozygous wild type
        line, denoted by \textit{+/+;+/+} or $F_0$ and a double balancer line
        (\textit{w;If/CyO;Sb/TM3,Ser}). After a first cross adult flies are
        selected for markers of both balancer chromosomes, yielding an $F_1$
        generation (a.k.a. double balancer cross). A backcross with the initial
        wild type line generates a pool of four different genotypes ($N_1$)
        of which on average 25\% of both chromosome 2 and 3 are balancer
        chromosomes.}



\subsection{Haplotype-resolved Hi-C data to study chromatin conformation}
\label{sec:balancer_study_design_hic}

Another central point of the study was to explore the three-dimensional
chromatin conformation of the highly rearranged balancer chromosomes and to link
this to observed \ase. Similarly to gene expression we also needed to measure
chromatin conformation in a haplotype-resolved manner. We hence performed a \hic
experiment and again used \acp{snv} to distinguish fragments belonging to
balancer- or wild type chromosomes. As a matter of fact the resulting
haplotype-resolved maps of contact frequency can additionally be utilized to
characterize genomic rearrangements (described in \cref{sec:balancer_hic_svs}).

Early on we made a decision to study fly embryos (instead of adult flies) for
mainly two reasons: To begin with, \textit{D. melanogaster} embryos are well
described, there is an outstanding amount of external data available
\citep{Gramates2017,Celniker2009}, and the Furlong lab has years of experience
with them (e.g. \cite{Furlong2001,Ghavi-Helm2014}). Secondly, though,
it is experimentally very difficult (if not infeasible) to extract intact
nuclei from adult flies, which is a crucial step to 3C-based experiments.
Directly collecting double balancer embryos, on the other hand, is not
realizable based on markers only expressed at an adult stage. We hence decided
to collect fly embryos from a backcross of the double balancer line ($F_1$) with
the original wild type line ($F_0$). The resulting generatio, termed $N_1$, is a
mix of genotypes as shown in \cref{fig:balancer_crossing_scheme}, in which
effectively 25\% of chromosomes 2 and 3 are balancer chromosomes.

In the subsequent sections I describe the methodology and our findings on the
effect genomic rearrangements have on chromatin organization and gene regulation.







\section{Mutational landscape of balanced chromosomes}
\label{sec:balancer_mutational_landscape}

We performed deep paired-end \wgs (approximately 100~x and 200~x, respectively)
of both the wild type line ($F_0$) and the double balancer line ($F_1$) with
read lengths of 300~bp and 200~bp on an Illumina MiSeq platform. To be able to
better resolve \acp{sv} we additionally sequenced mate pair libraries of both       \todo{explain mate pair in introduction}
samples with a read length of 2~x~100~bp and a median insert size of circa 4~kb.
I utilzied this data as well as \hic data (briefly described in
\cref{sec:balancer_cc_impl}) to characterize the mutations present on balancer
and wild type chromosomes in respect to a common reference genome, as described
subsequently.


\subsection{\Aclp{snv}}
\label{sec:balancer_snvs}

I utilized the \wgs data of both $F_0$ and $F_1$ cross simulataneously to call
\acp{snv} and small indels with \freebayes. The comparison of genotypes of the
homozygous wild type line and the heterozygous cross enabled me to assign
mutations to individual haplotypes: if, for example, a \snv is heterozygous in
the cross and homozygous alternative in the wild type sample, the variant is
located on the wild type chromosome and not on the homologous balancer
chromosome.

Of the 761,348 detected \acp{snv} on \ac{chr2} and \ac{chr3}, 38.9\% could be
assigned to the balancer chromosomes, 29.5\% to the wild type chromosomes and
29.8\% were shared between both. Only a fraction of 1.8\% of \acp{snv} did not
match one of the expected genotypes, which can mostly be attributed to regions
in the wild type chromosomes not being fully homozygous (see \cref{tab:snvs2}).
These numbers imply that, on average, there is one \snv every 210~bp that
distinguishes balancer and wild type haplotypes. This density was an important
parameter as \acp{snv} were utilized to separate sequencing data such as
RNA-seq into haplotypes later on (e.g. in \cref{sec:balancer_ase_impl}).

\begin{table}[ht]
    \centering
        %     wild type-specific    224,353
        %     balancer-specific     296,168
        %     common                226,907
        %     -----------------------------
        %chr2 wild type-specific 109654     48800648 2246.978360
        %chr2         errorneous    857     48800648   17.561242
        %chr2       heterozygous   4814     48800648   98.646231
        %chr2  balancer-specific 132594     48800648 2717.054085
        %chr2             common 114128     48800648 2338.657470
        %chr3         errorneous   1003     60189558   16.664020
        %chr3  balancer-specific 163574     60189558 2717.647470
        %chr3       heterozygous   7246     60189558  120.386330
        %chr3             common 112779     60189558 1873.730324
        %chr3 wild type-specific 114699     60189558 1905.629545
        %chrX         errorneous    134     23542271    5.691889
        %chrX       heterozygous   4148     23542271  176.193707
        %chrX  balancer-specific  22513     23542271  956.279876
        %chrX             common  52082     23542271 2212.275952
        %chrX wild type-specific  19870     23542271  844.013732
    \begin{tabu}{lrrr}
        \toprule
        Genotype & \ac{chr2} & \ac{chr3} & \ac{chrX} \\
        \midrule
        balancer-specific  & 2717.1 & 2717.6 &  956.3 \\
        wild type-specific & 2247.0 & 1905.6 &  844.0 \\
        shared             & 2338.7 & 1873.7 & 2212.3 \\
        wt heterozygous    &   98.6 &  120.4 &  176.2 \\
        errorneous         &   17.6 &   16.7 &    5.7 \\
        \bottomrule
    \end{tabu}
    \tabcap{snvs2}{Number of \acp{snv} per Megabase}{The
        majority of \snv calls are in concordance with the study design
        (balancer-specific, wild type-specific, or shared), except in short
        regions of remaining heterozygosity of the wild type chromosomes and
        a negligible number of false calls.
        Note that \ac{chrX} is not balanced but also paired with a homologue
        from a divergent line.}
\end{table}

The number of balancer-specific \acp{snv} was around 1.3 times higher than wild
type-specific ones. Moreover, when I compared our fly lines to the Drosophila
reference panel (DGRP) \citep{Mackay2012,Huang2014}, a panel of fully sequenced
inbred \textit{D. melanogaster} lines derived from a natural population, I
found that a striking majority of \acp{snv} is represented in the panel, yet
balancer-specific \acp{snv} to a lower fraction (86.1\%) than wild type-specific
ones (92.1\%). These observations are not surprising since balancer chromosomes,
due to their inability to undergo recombination, accumulate and likely tolerate
more mutations over time than normal chromosomes \citep{Araye2013}.

Moreover, I was interested in the inspecting the distribution of base
substitutions, i.e. the \explain{mutation spectrum}
    {as a summary of the total set of
    mutations I count the relative frequencies of different base substitutions
    (e.g. \texttt{C>T}) and their tri-nucleotide contexts (one base up- and
    downstream). The spectrum of somatic \acp{snv} is often analysed in cancer
    genomics in this way, in order to extract \emph{mutational signatures} that
    are linked to mutagenic processes}
 of the balancer chromosomes. The idea behind this is that balancer chromosomes
 had been derived by X-ray mutagenesis and that this could have led to a
 different distribution of base substitutions. However, after removing \acp{snv}
 shared with the DGRP I did not observe any striking differences in the mutation
 spectrum of \acp{snv} between wild type and balancers (see
 \cref{fig:signatures}; more detailed methods in \cref{sec:suppl_mutsign}).






\subsection{\texorpdfstring{\Aclp{cnv}}{Copy number variants}}
\label{sec:balancer_cnv}

\paragraph{Deletions} I called deletions on \wgs data of the $F_0$ and $F_1$
samples based on their paired-end read signature using \delly and developed an
extensive set of filters to reduce false positive predictions. For both reasons
of filtering and with validation in mind I categorized deletion calls into three
size ranges: (1) Below 50~bp -- these calls, predicted from split reads,
historically belong into the indel category and an additional set in this size
range was predicted with \freebayes. (2) 50~-~159~bp -- a size range where
predictions based on a split read signature typically yields reliable results,
yet these calls are large enough for validation with \pcr.
(3) 160~bp and larger -- in this size range I apply filters using additional
read depth signal before \pcr-based experimental validations.

For deletion predictions based on a paired-end read signature it is highly
recommended to utilize an additional read depth-based signal for filtering
(this is not exclusive to \delly, but noted also by the authors of other tools,
e.g. \cite{Layer2014}). Hence I developed an ad hoc filtering strategy
for larger (meaning 160~bp and more) deletion calls, which compares read depths
between samples (see \cref{sec:suppl_del} for detail). As a consequence I
excluded shared deletions in this size range, i.e. ones present in both balancer
and wild type haplotypes. However, this is not a limitation to this study as we
are primarily interested in genetic variation distinguishing the homologues.

\begin{table}[ht]
    \centering
    \begin{tabu}{lrrrr}
        \toprule
        Deletion subset           & \#Calls &  V &  I & Emp. \fdr\\
        \midrule
        \emph{<50~bp}             &   3,072 &  0 &  0 & \emph{unknown} \\
        50 - 159~bp               &     737 & 24 &  1 &     4\% \\
        160+~bp, low mappability  &      75 & 25 &  0 &     0\% \\
        160+~bp, high mappability &     395 & 24 &  1 &     4\% \\
        \midrule
        Weighted average          &   4,279 &    &    & 0.375\% \\
        \bottomrule
    \end{tabu}
    \tabcap{pcrresults}{\Ac{pcr} for deletion validation}{Number of calls as
        well as number number of validated (\textbf{V}) and invalidated
        (\textbf{I}) loci as determined by \pcr. The proportion of validated
        calls determines the empirical \ac{fdr} (\textbf{Emp. \fdr}).
        Calls below 160~bp were not \pcr-validated.}
\end{table}

Then \yad performed \pcr verification experiments on 75 randomly picked loci for
three subsets of deletion calls (\cref{tab:pcrresults}). A locus was considered
validated if the resulting \pcr bands of both $F_0$ and $F_1$ sample matched the
expected sizes; otherwise it was classified as invalidated. In order to be able
to compare homozygous and heterozygous results shared variants were not
submitted to \pcr. The final deletion call set based on \delly predictions
contained 4,279 calls. In order to get a single set of deletions I chose a lower
cutoff of 15~bp for deletions and merged the \delly call set with small
deletions predicted by \freebayes. This led to a total set of 8,340 calls on
chromosomes 2, 3, and X with a size distribution as shown in
\cref{fig:svsize_del}.

\figuretwocolumns
    {SVsize_DEL.pdf}{svsize_del}{Deletion size distribution}
    {The size distribution of final deletion calls (based on \delly and
    \freebayes) is shown. Common deletions (present in both balancer and wild
    type chromosomes) were excluded in the from 160~bp on. A higher number of
     balancer- over wild type-specific deletions was observed and this ratio
     increases with larger sizes.}
    {SVsize_DUP.pdf}{svsize_dup}{Duplication size distribution}
    {The final duplication set consisting of manually validated tandem
    duplication and a set of three non-tandem duplications is shown.}

\paragraph{Duplications} Next, I also predicted tandem duplications using \delly
using both \wgs and mate pair data simultaneously (see also
\cref{sec:suppl_dup}). Typically this class of variant is harder to
ascertain than deletions, which exhibit a very clear read-depth signal.
Yet also for duplications orthogonal information in addition
to the paired-end read signature can be utilized to gain confidence in
predictions. Specifically these are read depth, which should be considered in
the context of local mappability, and the bi-allelic frequency.                 \todo{Exlain bi-allelic freuqency in the INTRO}
After initial filters I generated overview plots containing such information for
352 loci on chromosomes 2 and 3. One example is shown in
\cref{fig:dup_validation}, and further examples, including a negative one, can
be found in \cref{fig:dup_validation_wt,fig:dup_validation_false}.
Upon inspection of the plots I demanded to see a change in read depth between
samples, as well as a characteristic change in the bi-allelic frequency. In a
balancer-specific duplication, for instance, the read-depth should increase only
in the $F_1$ sample and the bi-allelic frequency of \acp{snv} should switch from
50\% to either 33\% or 66\%.

In addition to tandem duplications predicted by \delly I found another three
non-tandem duplications by manual inspection of the bi-allelic frequency signal
across the genome, which were then included in the set of duplications prior
to manual validation. Some predicted loci could not be validated because they
resided within low-mappability regions, rendering both read depth and allelic
support for \acp{snv} unreliable. Nevertheless, I verified a total of 122
duplications in the size range from several hundred base pairs up to 258~kb
using this strategy (\cref{fig:svsize_dup}).

\figuretextwidth{dup_validation_example.pdf}{dup_validation} % DUP00003911
    {Signals used for duplication validation shown in an example}
    {A tandem duplication at locus \textit{chr2R:7,520,066âˆ’7,526,996} is shown,
    which was predicted based on paired-end read signature using \delly.
    The additional signals
    included for a visual validation are (1) a mappability track (fraction
    of uniquely mappable reads), (2) the bi-allelic frequency (fraction of
    reads supporting the alternative allele of an \snv), and (3)
    the total read coverage (as genomic coverage, in 100bp windows). Bi-allelic
    frequency is typically the clearest signal to discriminate true from false
    copy number variants, but reuqires the presence of \acp{snv}.}

\paragraph{Summary} The \acp{cnv} I report for balancer and wild type
chromosomes contain an abundance of short deletions starting from 15~bp and up
to 5.2~kb. The short size range of these deletions is owed to the gene-dense
character of the \textit{D. melanogaster} genome and in line with a previous
study that performed population-scale \sv calling in a population of flies,
where a median deletion size of 178~bp at a lower cutoff of 50~bp is reported
\citep{Zichner2013}.
With an experimentally estimated \fdr of 3.75\% the deletion call set can be
considered of high accuracy. Although no unexpectedly large deletions were
detected in neither wild type nor balancer chromosomes, I do note a slightly
higher rate of deletions on the balancer chromosomes. The ratios are 1.08 for
deletions in the size range below 50~bp, 1.3 for 50~-~159~bp, and 1.9 for larger
ones. Interstingly, a study in human cancer reported an increase in deletion
rate as a consequence of ionizing radiation \citep{Behjati2016}, hence the
subtle increase in deletion number (notably in the larger size spectrum) could
in fact be a fading trace of the irradiation applied during generation of the
balancer chromosomes.

Duplications, on the other hand, which underlay thorough manual validation,
where distributed evenly on balancer and wild type chromosomes. Nevertheless,
an exceptionally large duplication was found on \cyo, which effectively
increases the dosage of around 33 genes.






\section{Validation of large chromosomal rearrangements in Hi-C data}
\label{sec:balancer_hic_svs}

I further aimed at refining breakpoint junctions of the major rearrangements on
the balancer chromosomes, which had previously been known only at microscopic
resolution\footnote{\label{footnote:balancer_karyotype}
    Described in terms of cytological bands, i.e. \cyo = \textit{2Lt-22D1
    33F5-30F 50D1-58A4 42A2-34A1 22D2-30E 50C10-42A3 58B1-2Rt} and
    \ac{tm3} = \textit{3Lt-65E 85E-79E 100C-100F2 92D1-85E 65E-71C
    94D-93A 76C-71C 94F-100C 79E-76C 93A-92E1 100F3-3Rt}.
    Source: \url{https://bdsc.indiana.edu/stocks/balancers/balancer_bps.html}.}.
Based on allelic separation using \acp{snv}, \alek generated a \hic map of
contact frequencies for both balancer and wild type haplotypes separately, yet
in respect to the same reference genome (\cref{sec:balancer_cc_impl}).
Right away we identified strong signals off the diagonal with a characteristic
``bowtie'' shape as shown in \cref{fig:balancer_cc_impl}, which coincide with
gaps close to the diagonal. Undoubtedly this signal stems from the genomic
rearrangements we expected to see and by comparing them to the available
karyotype information Yad Ghavi-Helm could assign 15/16 of these breakpoints to
the respective inversions. I then searched through inversion and
translocation\footnote{Translocations are connections across chromosomes, yet
    because the left and right arms of chromosomes 2 and 3 are reported
    separately in \ac{dm6} rearrangements spanning the centromere are reported
    as translocations.}
predictions made using \delly on \wgs and mate pair sequencing data to
track down breakpoints at base pair resolution. This yielded base pair
resolution breakpoints in in 14/15 cases, which are reported in
\cref{sec:suppl_balancer_breakpoints}.

\figuretextwidth{dup_hic_biaf.png}{dup_hic_biaf}{Large duplication on
    \texorpdfstring{\cyo}{CyO} characterized by differential Hi-C data}{
    Genomic regions around the duplication \textit{chr2L:8,957,000-9,215,000}.
    The top panel displays differential unnormalized Hi-C fragment counts
    ($\textrm{log}_2 \frac{\textrm{balancer}}{\textrm{wild type}}$)
    colored from red (positive) to blue (negative). The panel below shows the
    biallelic frequency. This duplication was initially discovered
    based on the characteristic biallelic frequency signal and is validated by
    a total increase of locus-internal Hi-C contacts on the balancer chromosome,
    as seen as the red triangle. Moreover, a decrease in balancer-specific
    contacts close to the diagonal left of the locus (blue) as
    well as an increase left of the upper tip of the triangular region
    suggest the duplication was inserted to the left in inverted orientation.
    Hi-C data processing and plots were contributed by \alek.}

Reassuringly, when I compared these
breakpoint with the genomic positions stated in a recent study mapping mutations
on \ac{tm3} \citep{Miller2016} I found precise agreement for 8 of 10
breakpoints. For one of the two remaining breakpoints (20.3~Mb on \ac{tm3})
that were missed both by Miller et al. and here I could at least estimate the
breakpoint location at a resolution of 5kb and further narrowed it down to a
``best guess'' based on manual inspection of mate pair data. The other
breakpoint is outside the span of the reference genome and thus inaccessible to
read-mapping based methods. Only one of the breakpoint junctions in the balancer
chromosomes showed a precise re-ligation, whereas most of them are scarred by a
loss of genetic material (9 cases, max. 1.8~kb, median 114~bp) or small sequence
duplications (4 cases, median 9~bp, max 281~bp; see also
\cref{sec:suppl_balancer_breakpoints}). Additionally, the estimation of the
breakpoint at 20.3~Mb on \ac{tm3} appears to carry a deletion of around 17~kb.

Moreover, I utilized the haplotype-resolved contact maps further to validate and
characterize other large \acp{sv}. Notably, an inversion of approximately 40~kb
on \ac{tm3}\footnote{Precisely, at \textit{chr3L:~14.61~-~14.64~Mb}.} was
validated by a characteristic Hi-C signal (data not shown). I also inspected
aforementioned 258~kb duplication on \cyo closer, which I had predicted by a
chatacteristic bi-allelic frequency signal (described in
\cref{sec:balancer_cnv}). Interestingly, the differential \hic contact map as
displayed in \cref{fig:dup_hic_biaf} confirms the duplication (seen as a red
triangle, which stands for a total increase in contacts on the balancer
haplotype), yet also suggests that the additional copy of the locus was most
likely inserted in inverted orientation to its left end. There are two signals
providing evidence for this: A reduction in contacts of the locus to its left
neighboring region (blue), suuggesting that this end of the locus is no longer
proximal to its original left neighboring region, just as if an insertion had
happened there. And secondly, there is a striking increase in contact frequency
of the right end of the duplicated locus to the left neighboring region. Taken
together this is best explained by an inverted duplication occurring in tandem.

\figuretextplusmargin{balancer_hiC_rearrangements.png}{balancer_cc_impl}
    {Major rearrangements of the balancer chromosomes revealed by Hi-C}{
    Haplotype-resolved Hi-C contact frequency maps of chromosomes 2 and 3 in
    respect to the wild type reference genome are shown (A). Details of the data
    processing can be found in \cref{sec:suppl_hic}. The bottom triangles show
    characteristic ``bowtie-shaped'' patterns as well as gaps on the diagonal
    that demarcate breakpoint junctions of the rearranged balancer chromosomes.
    The respective sections of the reference genome are color-coded, with a grey
    circle representing centromers.
    These junctions can be followed to reconstruct the relative order of the
    balancer chromosomes such as shown in the top panel (B). Vertical ``bowties''
    represent a connection of segments in same orientation where as horizontal
    ones mark junction of segments that are inverted in respect to one another.
    The fully reconstructed order of the balancer genomes is shown in the bottom
    panel (C). This matches previous annotation from
    karyotyping\footnoteref{footnote:balancer_karyotype}.}









\FloatBarrier
\section{Global changes in gene expression}
\label{sec:balancer_ase}

In the next step we sought to measure gene expression from both haplotypes to
determine the potential impact of the observed genetic variation on the
regulation of genes. To do so we first had to overcome biases related to
maternally deposited mRNA present in early embryos, as described subsequently.
Then I implemented robust \ase detection across multiple biological replicates
(\cref{sec:balancer_ase_impl}) and applied it to the data of our study as well
as in the scope of two additional control experiments
(\cref{sec:balancer_ase_controls}).




\subsection{Controlling for maternally deposited mRNA in early embryos}
\label{sec:balancer_maternal_rna}

\figuretextwidth{ASE_histograms.pdf}{ASE_histograms}{Allelic mRNA ratio per gene}{
    Histograms of the fraction per gene of balancer RNA-seq fragments among the
    fragments that can be assigned to one of the haplotypes for three
    different RNA-seq experiments with two replicates each. The expected
    fraction of 25\% is marked by a dotted line. Counts were derived as described
    in \cref{sec:balancer_ase_impl}.}

At an early stage of the project we sequenced mRNA from 4-8h embryos of the
$N_1$ backcross (\cref{fig:balancer_crossing_scheme}) to explore the power to
detect significant \ase genes. As briefly introduced in
\cref{sec:balancer_study_design_hic}, 25\% of chromosomes 2 and 3 within the
genetic pool consist of balancer chromosomes. This means that when we separate
RNA-seq reads by haplotype we expect their distribution to be centered around a
balancer fraction of 25\%. As the left panels of \cref{fig:ASE_histograms}
display, this initial experiment was marked by a (unsatisfactory) wide
distribution. More precisely, it did not follow a binomial distribution, which
is a model applied by many \ase detection methods, but showed a large
overrepresentation of wild type mRNA instead. Only after discussions with
\garfield we figured that, since we had crossed female wild type flies with double
balancer males, the reason for this skew in the distribution could be attributed
to \explain{maternally deposited mRNA}{Before maternal-to-zygotic transition
    zygotes or early embryos do not transcribe their own genes but utilize mRNA
    of the maternal cell that was loaded into the egg during oogenesis. After
    around two hours of embryonic development a large portion of maternal mRNA
    is actively degraded, yet not for the complete transcriptome
    \citep{Tadros2009}}
present in early embryos as well as unfertilized eggs remaining in the sequenced
pool.

Accordingly, \yad initiated a new experiment with modified conditions: First, we
collected embryos only in a later, 6-8h time window when the impact of maternal
mRNA is expected to be weaker. Second, she manually sorted out unfertilized eggs
from the collection of embryos. And third, she derived two different crosses,
one with a paternal double balancer line (same way as initially), termed \Npat,
and another one with a maternal double balancer line, \Nmat. The potential
effect of maternally deposited mRNA will then cause a shift in exactly opposite
directions in both these lines, which can be accounted for during the \ase
calling. These considerations are nicely reflected in the histograms on the
right of \cref{fig:ASE_histograms}. More narrow peaks indicate a much lower
fraction of maternal transcripts, and a shift from left to right between \Npat
to \Nmat point to remaining maternal transcripts that can be identified as such.





\subsection{Allele-specific expression detection}
\label{sec:balancer_ase_impl}


More detail in \cref{sec:suppl_deseq}.

\begin{table}[ht]
    \centering
    \begin{tabu}{lrrrr}
        \toprule
        Data set   & Total reads   & Balancer   & WT   & Ambiguous \\
        \midrule
        \Npat 1. replicate & 51,560,368  & 6.151\% & 22.489\%  & 0.136\% \\
        \Npat 2. replicate & 52,618,060  & 5.762\% & 21.471\%  & 0.139\% \\
        \Nmat 1. replicate & 54,521,194  & 8.030\% & 20.879\%  & 0.142\% \\
        \Nmat 2. replicate & 48,179,068  & 7.834\% & 20.352\%  & 0.153\% \\
        \bottomrule
    \end{tabu}
    \tabcap{balancer_rnaseq}{Haplotype separation of RNA-seq data}{
        Fraction of RNA-seq read pairs that could be assigned to either one of
        the haplotypes balancer or wild type (WT) ot that contained conflicting
        variants (ambiguous). The difference to 100\% in each comes from read
        pairs that could not be assigned.}
\end{table}


\figuretextwidth{DESeq2_N1_6_8h.pdf}{DESeq}{ASE analysis based on
    \texorpdfstring{\deseq}{DESeq2}}{Average haplotype-resolved fragment count
    per gene across four replicates vs. log 2 fold change of balancer over wild
    type counts are shown. A single outlier to the right was trimmed. }


\subsection{RNA-seq control experiments}
\label{sec:balancer_ase_controls}




\figuremargin{ASE_chrX_numbers.pdf}{ase_chrX}{ASE fractions per chromosome}{ASE
    fraction detected on each of the chromosomes 2, 3, and X in a female adult
    line $F_1^{f}$. Error bars indicate 99\% confidence intervals from a binomial
    test.}



\section{The interplay between SVs and differentially expressed genes}

\subsection{Genes affected by large rearrangements}
\label{sec:balancer_ase_breakpoints}

\subsection{Positional clustering of ASE genes}
\label{sec:balancer_ase_clustering}

\figuretextwidth{ase_clustering.pdf}{ase_clustering}{Distance between
    neighboring ASE genes}{This is a histogram of distances between neighboring
    significant ASE genes (blue) as well as of the same number of random control
    genes (sampled from expressed, but non-ASE genes 500 times). Errorbars
    indicate the 5\% and 95\% quantiles of random sub-sampling.}

From the point of view of large rearrangements \cref{fig:suppl_ase_genes_around_bps}





\subsection{\texorpdfstring{\ase}{ASE} signal related to changes in copy number}
\label{sec:balancer_ase_cnvs}


% Testing this matrix for randomness:
%           no_CNV CNV_overlap
% no_ASE      4551         292
% ASE_genes    439          73
%
%         Fisher's Exact Test for Count Data
%
% p-value = 2.63e-10
% alternative hypothesis: true odds ratio is not equal to 1
% 95 percent confidence interval:
%  1.940543 3.426937
% sample estimates:
% odds ratio
%   2.591054
\figuretextwidth{ASE_cnv_overlap.pdf}{ASE_cnv_overlap}{Log fold change of
    \texorpdfstring{\ase}{ASE} genes overlapping \texorpdfstring{\acp{cnv}}{CNVs}}
    {Fold change (log 2) of 73 significant \ase genes that overlap \acp{cnv}.
    Two genes overlapped multiple different \acp{cnv} (\textit{ambiguous}). A
    positive log fold change means higher expression in the balancer haplotype
    and vice versa.}



\subsection{Mobile element insertions can give rise to strong ASE signals}
\label{sec:balancer_ase_mei}

\figuretextwidth{MEI_example.pdf}{balancer_mei_example}{Example of a
\texorpdfstring{\acs{mei}}{MEI} driving ectopic expression of a gene}{
    RNA-seq tracks along the gene \textit{Ptp52F} on chromosome 2R showing the
    total RNA-seq reads (total read coverage in absolute numbers; bottom panel)
    as well as the portion of RNA-seq reads that could be resolved to the
    wild type (top panel) and balancer (middle panel) haplotypes. Colored
    vertical lines represent \acp{snv} used for haplotype-separation. RNA-seq
    tracks were visualized using \igb.
}

List of cases in \cref{tab:meilist}

\figuretextplusmargin{MEI_impact.pdf}{balancer_mei_impact}{ASE genes associated
    to MEI}{List of significant ASE genes (x axis) ordered by log fold change
    (balancer/wild type, y axis). Genes that are very likely dis-regulated due
    to an MEI driving their chimeric expression are highlighted in blue.}





\section{Changes in chromatin conformation: an outlook}
\label{sec:balancer_cc}

\subsection{Haplotype-resolved maps of contact frequency}
\label{sec:balancer_cc_impl}

\subsection{Differences in chromatin conformation between wild type and balancer chromosomes}
\label{sec:balancer_cc_differences}



\section{Integrated visualization of single loci for data exploration}
\label{sec:balancer_visualization}


\section{Conclusions}
\label{sec:balancer_concl}
