\chapter{Effects of SVs on Gene Expression and Chromatin
Organization in \texorpdfstring{\textit{D. Melanogaster}}{D. Melanogaster}}
\label{sec:balancer}



In the project described here I teamed up with \yad, \alek and \eileen to study the functional
impact of \acp{sv}, specifically of large chromosomal rearrangements, in the model
organism \textit{Drosophila melanogaster}. Recent
sequencing technologies, here \hic, were of fundamental importance for the characterization
of \acp{sv}, which was an essential step of the project. All wet lab experiments
described below were performed by \yad with the help of \rebecca.
\alek implemented all \hic-related analyses, which are hence
only depicted briefly. All other computational analyses including all figures
are my own work if not explicitly stated differently. Supplementary information
can be found in \cref{sec:suppl_balancer}. At the time of writing a manuscript
of this study was in preparation [Ghavi-Helm, Meiers, Jankowski, et al., 2018].



\section{Background}
\label{sec:balancer_background}


\subsection{Impact of \texorpdfstring{\acsp{sv}}{SVs} in health and disease}

Population-scale studies revealed that \acp{sv} are a major contributor to
genetic variation in the human population \citep{Conrad2010} â€“ in fact they
contribute more base pair differences than \acp{snv} \citep{Sudmant2015}.
Besides their abundance in healthy individuals, \acp{sv} had earlier already
been recognized for their role in a number of diseases, as
\citet{Zhang2009,Weischenfeldt2013,Carvalho2016} review in more detail.

The consequences certain \acp{sv} imply can range from purely molecular
phenotypes, such as altered gene expression, to little severe (e.g. the efficiency of
starch digestion depending on the copy number of \textit{AMY1} \citep{Perry2007})
or more severe phenotypes (e.g. red-green blindness, caused by \acp{cnv} on
chromosome X \citep{Nathans1986}), to disease or to an increased susceptibility
towards a disease. The latter can include complex diseases such as autoimmunity
\citep{Fanciulli2007} or autism \citep{Sebat2007}. Especially \acp{cnv}, the
most-frequently studied type of \sv, have been linked to Mendelian diseases
\citep[see table 2]{Zhang2009} and are frequently involved in cancer
\citep{Beroukhim2010}. Yet \sv classes that contribute to disease are by far
not restricted to only deletions or duplications. Aneuploidy, to start with, can
play a major role in diseases such as Down syndrome, which was among the earliest
identified genetic diseases \citep{Lejeune1959}. Also inversions were identified
as the cause of several Mendelian diseases \citep{Feuk2010}. Recurrent
translocations, often creating gene fusions, are known to be driving many
cancer types \citep{Mertens2015} and, at last, transposable elements have
been noted be to play a role in cancer \citep{Burns2017}.

The mechanisms by which \acp{sv} induce these phenotypic consequences are
manifold. Breakpoints of \acp{sv} that fall into a gene may cause its loss
of function. Aforementioned gene fusions can create novel chimeras that
potentially exert fundamentally different functions from the original genes
\citep{Mertens2015}. Copy number gains or losses lead to dosage imbalance, with
critical consequences for many cancer-associated genes \citep{Fehrmann2015}, and
a heterozygous deletion or a \loh event can reveal recessive mutations
that had been rescued by a functional allele beforehand.
Furthermore, \acp{sv} may impact on the molecular level, e.g. on gene expression,
which can be studied even if not associated with an observable phenotype.
In the context of \explain{expression quantitiative trail loci}{Genomic loci
    with different alleles that influence the expression of genes (typically a
    single gene) that are not neccessarily in close proximity} analyses,
more and more such effects have lately been found \citep{Sudmant2015,Chiang2017}.

However, causative \acp{sv} need not affect the actual gene itself, but could
target regulatory sequences in proximity or elsewhere in the genome. This is
typically the case for aforementioned expression quantitative trait loci, yet
the (then very surprising) discovery was already made in 1979 in a row of
heritable blood diseases named thalassemia \citep{Fritsch1979}. These
\emph{position effects} have long been noted to have an impact in health and
disease, yet the underlying mechanism are in many cases not yet understood
\citep{Kleinjan2005}.

Over the past years the community has gained more insight into a novel mechanism
which \acp{sv} neither affects genes nor regulatory sequences themselves.
Instead, this mechanism re-models the three-dimensional organization of
chromatin, as I elaborate further below.




\subsection{Three-dimensional chromatin conformation}
\label{sec:chromatin_conformation}

\figuremargin{TAD_schematic_zaugg.jpg}{TAD_schematic}{Schematic of
\texorpdfstring{\aclp{tad}}{topologically associating domains}}{Characteristic
triangles in a mammalian \hic map (top) belong to spatial domains of DNA
inside the nucleus (bottom). Figure taken from \citetitle{Ruiz-Velasco2017review}
\citep{Ruiz-Velasco2017review} licensed under \acl{ccby4}.}

The advance of chromatin conformation capture techniques, notably \hic
(introduced in \cref{sec:ccc}), revealed a feature of the spatial organization
of chromatin named topologically associating domains (\acp{tad})
\citep{Dixon2012,Nora2012,Sexton2012,Rao2014} (see
\cref{sec:suppl_balancer_literature} for comments on the cited literature).
\Acp{tad} are physical domains of
DNA that are characterized by an increase in chromatin interactions inside them
and a relative insulation of contacts across distinct domains. These structures
show up as characteristic triangles in contact frequency maps, as illustrated in
\cref{fig:TAD_schematic}. Despite molecular differences in, for example,
the involved architectural proteins, the phenomenon of \acp{tad} was observed in
a range of species across the tree of life ranging from humans and mice to
\textit{Drosophila melanogaster} and \textit{Caenorhabditis elegans}. This suggests
that these structures are a universal feature of metazoan genomes
\citep{Dekker2015}.

\Acp{tad} have gained extraordinary attention in the field and their potential
function has been discussed and reviewed intensely in recent years
\citep[among others]{Gibcus2013,Gorkin2014,Sexton2015b,Hnisz2016a,Ruiz-Velasco2017}.
Key characteristics of \acp{tad} are that the expression of genes within
\acp{tad} tends to be orchestrated \citep[see figure 4b]{LeDily2014,Nora2012} and
that they align with epigenetic features such as histone marks \citep{Nora2012}
and DNA replication timing \citep{Pope2014,LeDily2014}. Furthermore, \tad
boundaries are associated with the insulator element-binding protein CTCF,
enriched in house-keeping genes and conserved across cell types
\citep{Dixon2012,Rao2014,Schmitt2016}.

Well-characterized long-range interactions between promoters and enhancers,
which can be revealed through 3C-based techniques, appear to be confined
within \acp{tad} (reviewed by \citet{Smallwood2013}). Additionally, units of
enhancers and promoters with correlated activity were found to align well
with them \citep{Shen2012}.
It is unclear whether the DNA loops connecting enhancers and promoters are cause
or consequence of active expression, but it was observed that many such contacts
establish long before gene activation \citep{Ghavi-Helm2014}.
Further research using functional assays supported that genes are co-regulated
within \acp{tad}. Notably, \citet{Symmons2014} inserted reporter genes at
several hundred sites of the mouse genome and observed tissue-specific activity
in spatial blocks correspond to \acp{tad}.

Together, these results suggest that \acp{tad} exert a crucial regulatory
function. A current hypothesis is that they confine promoter-enhancer
interactions inside, as \citet{Ji2016} call them, insulated neighbourhoods. As
I described in the next section, a series of recent perturbation studies
gave additional substance to this hypothesis.





\subsection{Consequences of disrupting chromatin conformation}
\label{sec:disrupting_tads}

Several studies followed up on the question what would happen when the boundaries
between \acp{tad} were disrupted. In line with the hypothesis of insulated
neighborhoods, they found that the merging of two \acp{tad} can alter the
``search space'' of an enhancer element, which can then suddenly drive
expression of another gene to non-physiological amounts. This mechanism was
termed \emph{enhancer adoption} \citep{Lettice2011} or \emph{enhancer
hijacking} \citep{Northcott2014}.

The cause for a \tad boundary disruption can be a \sv that deletes or inverts
the genomic region harboring this boundary. This was quite remarkably shown in
various cancer types including medulloblastoma \citep{Northcott2014}, lung cancer
and colorectal cancer \citep{Weischenfeldt2016} and acute lymphoblastic leukemia
\citep{Hnisz2016}. Furthermore, computational studies linked this mechanism to
genetic diseases via mining of public databases and showed, among other things,
that around 10\% of known disease-associated deletions potentially function via
an enhancer adoption mechanism \citep{Ibn-Salem2014,Li2016,Zepeda-Mendoza2017}.

Moreover, as \citet{Lupianez2016} and \citet{Krijger2016} review, a number of studies
specifically tested this hypothesis on particular genomic loci by genome
editing using CRISPR/Cas \citep{Doudna2014}. For example, \citet{Guo2015} altered
the binding sites of an architectural protein in proximity of an enhancer element
and observed looping as well as ectopic gene expression across the boundary.
Similarly, \citet{Narendra2015} observed heterochromatin spreading when they
deleted such a boundary at the \textit{Hox} locus in \textit{Drosophila}.
In fact, already \citet{Nora2012} observed this mechanisms when they studied an
additional mouse line with a \tad boundary deletion and observed a merged \tad
and ectopic gene expression instead of the two clearly separated \acp{tad} and
regulatory insulation seen in wild type mice.

Recently, \citet{Lupianez2015} impressively replicated limb abnormalities in mice
that were known from human genetic diseases. They first showed that these
abnormalities originated from misregulated gene expression in a developmental
stage, which was caused by \tad boundary deletion. When they engineered similar deletions in
mice using the CRISPR/Cas system, they observed a change in chromatin
conformation, ectopic gene expression across the boudary and indeed the same
phenotype of limb malformations that had been observed in human patients.

In conclusion, all these studies provide strong evidence for \acp{tad} to play
a central role for gene regulation. When \acp{tad} break---and fuse with another
\tad---genes encounter a new regulatory environment including enhancers, which
may lead to severe misregulation via en enhancer adopiton mechanisms.







\section{Design of the study}
\label{sec:balancer_study_design}

In this study we set out to understand how genomic aberrations such as the
disruption of \acp{tad} caused by large chromosomal rearrangements and other
\acp{sv} can affect the regulation of surrounding genes. In contrast to previous
studies (as outlined in \cref{sec:disrupting_tads}) we explicitly
focused on a phenotypically healthy system where no dominating effects, but
rather modest changes in molecular phenotypes were to be expected. We think this
work will complement previous studies, which had investigated rather
pathological situations, and open a new perspective on the functional impact of
\acp{tad}. Moreover, instead of selecting a single locus we aimed at testing
multiple rearrangements in a genome-wide fashion to gain a broader understanding
on the generality of such effects.




\subsection{Balancer chromosomes carry large rearrangements}
\label{sec:balancer_balancers}

We found a suitable model system for this task in so-called balancer chromosomes
of \textit{Drosophila melanogaster}. These naturally derived chromosomes have a
long tradition in fly genetics and are frequently used as a tool to keep
recessive lethal mutations from being lost from the population. Three relevant
features allow them to perform this task: (1) they carry recessive lethal
mutations so that homozygous offspring will not live, effectively balancing the
alleles in a population. This further requires that haplotypes remain intact,
so consequently (2) balancer chromosomes
suppress recombination by disrupting homologous pairing. This is achieved by the
introduction of multiple large inversions via, for example,
\explain{X-ray mutagenesis}{Balancer chromosomes date back to the work of
    Hermann Joseph Muller, who in the 1920s studied mutagenesis through X-ray
    radiation \citep{Muller1928} and received a Nobel prize in 1948\footnotemark}.
\footnotetext{See more in a blog post from Laurence A. Moran:
    \url{https://sandwalk.blogspot.de/2008/10/hermann-muller-invented-balancer.html}}
(3) Balancer chromosomes include dominant marker alleles so that carriers can be
readily detected based on their phenotype in adult stage. Such balancer
chromosomes are nowadays available for all the major chromosomes 2, 3, and X.
Interestingly, despite their common usage only recently some balancer
chromosomes had been characterized via whole-genome sequencing \citep{Miller2016,Miller2018}.

We chose balancer chromosomes for chromosome 2 and 3, namely \explain{\cyo}{
    Chromosome 2 balancer, derived from \citet{Oster1956} and based on work of
    \citet{Ward1923} (see \citet{Miller2018}), which carries the \textit{Cy}
    allele, hence showing a characteristic
    phenotype of curly wings in adult flies \marginfig{dm_cyo.pdf}}
and TM3 \citep{Tinderholt1960} to create a double balancer line, named $F_1$.
See \cref{fig:balancer_chroms} for an overview of the \textit{D. melanogaster}
genome of both the wild type line and the double balancer line utilized in this
study. For both balancer chromosomes the major rearrangements had been
previously characterized by karyotyping (see
note\footnoteref{footnote:balancer_karyotype} and
\cref{fig:balancer_hic_rearrangements}). With two balancer chromosomes,
we effectively increased the number of genomic rearrangements to
be studied, which was an important consideration early on in the project.
Together both balancer chromosomes carry 16 breakpoints of large, partly
centromere-spanning inversions and we expected them to contain additional
sub-microscopic \acp{sv} related to their mutagenic origin. Accordingly, one of
the first step of the study was a deep characterization of the mutational
landscape  of the balancer chromosomes, which is presented in
\cref{sec:balancer_mutational_landscape}.





\subsection{Studying \texorpdfstring{\textit{cis}}{cis}-regulation through
    allele-specific gene expression and haplotype-resolved chromatin conformation}
\label{sec:balancer_ase_motivation}

To study the effects of \acp{sv} on gene regulation we wanted to compare global
gene expression of the balancer chromosomes against their the wild type
homologues. This could principally be done by comparing two different fly
strains, one carrying genomic rearrangements and one without. However, while
this is also not possible using balancer chromosomes that cannot be
in a homozygous state, this approach would make it difficult to distinguish
\textit{cis}-regulation, i.e. the effect of alterations of the DNA itself, from
\textit{trans}-regulation such as an altered expression of some transcription
factor. This is why a central foundation of the study was to compare wild
type chromosomes and balancer chromosomes within the same fly line via \acf{ase} analysis.
This is achieved by measuring gene expression separately for both alleles, which
differ by naturally occurring variation, notably \acp{snv}.
Since we were able to resolve chromosome-long haplotypes in this study, alleles can be
aggregated across genes and expression changes can be distinguished into up- and
down-regulation of the balancer genes in respect to the wild type genes.
\Cref{sec:balancer_maternal_rna,sec:balancer_ase_impl} provide more detail on the procedure.
We assume that the vast majority of \acp{snv} simply tag alleles but do not have
an effect on their regulation, or that this effect is negligible in comparison
to larger rearrangements.

\figuretwocolumns[0.38]
    {balancer_chroms.pdf}{balancer_chroms}{Genome overview}{Schematic of the
        four chromosomes of (female) \textit{D. melanogaster} flies of the wild
        type line $F_0$ and of the double balancer cross $F_1$. \Ac{cyo} red,
        \ac{tm3} blue. }
    {balancer_crossing_scheme.pdf}{balancer_crossing_scheme}{Crossing scheme}{
        The fly lines used in this study are derived from a homozygous wild type
        line, denoted by \textit{+/+;+/+} or $F_0$ and a double balancer line
        (\textit{w;If/CyO;Sb/TM3,Ser}). After a first cross adult flies are
        selected for markers of both balancer chromosomes, yielding an $F_1$
        generation (a.k.a. double balancer cross). A backcross with the initial
        wild type line generates a pool of four different genotypes ($N_1$)
        of which on average 25\% of both chromosome 2 and 3 are balancer
        chromosomes.}

Another central point of the study was to explore the three-dimensional
chromatin conformation of the highly rearranged balancer chromosomes and to link
this to observed \ase. We hence performed a \hic
experiment and again used \acp{snv} to distinguish fragments belonging to
balancer- or wild type chromosomes. As a matter of fact the resulting
haplotype-resolved maps of contact frequency can additionally be utilized to
characterize genomic rearrangements, as I describe in \cref{sec:balancer_hic_svs}.

Early on we made a decision to study fly embryos (instead of adult flies) for
mainly two reasons: To begin with, \textit{D. melanogaster} embryos are well
described, there is an outstanding amount of external data available
\citep{Gramates2017,Celniker2009}, and the Furlong lab has years of experience
on embryogenesis \cite[among others]{Furlong2001,Ghavi-Helm2014}. Secondly,
it is experimentally very difficult (if not infeasible) to extract intact
nuclei from adult flies, which is a crucial requirement for \hic experiments.
However, collecting double balancer embryos is not directly possible
based on markers only expressed in an adult stage. We hence decided
to collect fly embryos from a backcross of the double balancer line ($F_1$) with
the original wild type line ($F_0$). The resulting generation, termed $N_1$, is a
mix of genotypes as shown in \cref{fig:balancer_crossing_scheme}, in which
effectively 25\% of chromosomes 2 and 3 are balancer chromosomes. This cross
was used both to measure \ase and chromatin conformation. In the subsequent
sections I describe the methodology and our findings on the effect genomic
rearrangements have on chromatin organization and gene regulation.









\section{Results I: Mutational landscape of balanced chromosomes}
\label{sec:balancer_mutational_landscape}

We performed deep paired-end \acf{wgs} (approximately 100~x and 200~x, respectively)
of both the wild type line ($F_0$) and the double balancer line ($F_1$) with
read lengths of 300 and 200~bp on an Illumina MiSeq platform. To be able to
better resolve \acp{sv} we additionally sequenced mate pair libraries of both
samples with a read length of 2~x~100~bp and a median insert size of circa 4~kb.
I utilzied this data as well as \hic data (experiments described in
\cref{sec:balancer_cc_impl}) to characterize the mutations present on balancer
and wild type chromosomes in respect to a common reference genome. Below, a deep
characterization of mutations on the balanced chromosomes are described.



\subsection{\Aclp{snv}}
\label{sec:balancer_snvs}

I utilized \wgs data of both $F_0$ and $F_1$ cross simulataneously to call
\acp{snv} and small indels with \freebayes. The comparison of genotypes of the
homozygous wild type line and the heterozygous cross enabled me to assign
mutations to individual haplotypes: if, for example, a \snv is heterozygous in
the cross and homozygous alternative in the wild type sample, the variant is
located on the wild type chromosome and not on the homologous balancer
chromosome.

Of the 761,348 detected \acp{snv} on chromosomes 2 and 3, 38.9\% could be
assigned to the balancer chromosomes, 29.5\% to the wild type chromosomes and
29.8\% were shared between both. Only a fraction of 1.8\% of \acp{snv} did not
match any of the expected genotypes, which can mostly be attributed to regions
in the wild type chromosomes not being fully homozygous (\cref{tab:snvs2}).
These numbers imply that, on average, there is one \snv every 210~bp that
distinguishes balancer and wild type haplotypes. This density was an important
parameter as \acp{snv} were utilized later to separate sequencing data such as
RNA-seq into haplotypes (\cref{sec:balancer_ase_impl,sec:balancer_cc_impl}).

\begin{table}[ht]
    \centering
        %     wild type-specific    224,353
        %     balancer-specific     296,168
        %     common                226,907
        %     -----------------------------
        %chr2 wild type-specific 109654     48800648 2246.978360
        %chr2         errorneous    857     48800648   17.561242
        %chr2       heterozygous   4814     48800648   98.646231
        %chr2  balancer-specific 132594     48800648 2717.054085
        %chr2             common 114128     48800648 2338.657470
        %chr3         errorneous   1003     60189558   16.664020
        %chr3  balancer-specific 163574     60189558 2717.647470
        %chr3       heterozygous   7246     60189558  120.386330
        %chr3             common 112779     60189558 1873.730324
        %chr3 wild type-specific 114699     60189558 1905.629545
        %chrX         errorneous    134     23542271    5.691889
        %chrX       heterozygous   4148     23542271  176.193707
        %chrX  balancer-specific  22513     23542271  956.279876
        %chrX             common  52082     23542271 2212.275952
        %chrX wild type-specific  19870     23542271  844.013732
    \begin{tabu}{lrrr}
        \toprule
        Genotype & \ac{chr2} & \ac{chr3} & \ac{chrX} \\
        \midrule
        balancer-specific  & 2717.1 & 2717.6 &  956.3 \\
        wild type-specific & 2247.0 & 1905.6 &  844.0 \\
        common             & 2338.7 & 1873.7 & 2212.3 \\
        wt heterozygous    &   98.6 &  120.4 &  176.2 \\
        errorneous         &   17.6 &   16.7 &    5.7 \\
        \bottomrule
    \end{tabu}
    \tabcap{snvs2}{Number of \acp{snv} per Megabase}{The
        majority of \snv calls are in concordance with the study design
        (balancer-specific, wild type-specific, or common), except in short
        regions of remaining heterozygosity of the wild type chromosomes and
        a negligible number of false calls.
        Note that \ac{chrX} is not balanced but also paired with a homologue
        from a divergent line.}
\end{table}

The number of balancer-specific \acp{snv} was around 1.3 times higher than wild
type-specific ones. Moreover, when I compared the observed variation to the \textit{Drosophila}
reference panel (DGRP) \citep{Mackay2012,Huang2014}, a panel of fully sequenced
inbred \textit{D. melanogaster} lines derived from a natural population, I
found that a striking majority of \acp{snv} is represented in the panel, yet
balancer-specific \acp{snv} to a lower fraction (86.1\%) than wild type-specific
ones (92.1\%). These observations are not surprising since balancer chromosomes,
due to their inability to undergo recombination, accumulate and likely tolerate
more mutations over time than normal chromosomes \citep{Araye2013}.

Moreover, I was had a closer look at the distribution of base
substitutions, i.e. the \explain{mutation spectrum}
    {As a summary of the total set of
    mutations I count the relative frequencies of different base substitutions
    (e.g. \texttt{C>T}) and their tri-nucleotide contexts (one base up- and
    downstream). The spectrum of somatic \acp{snv} is often analysed in cancer
    genomics in this way, in order to extract \emph{mutational signatures} that
    are linked to mutagenic processes}
of the balancer chromosomes. The idea behind this is that balancer chromosomes,
which had been derived by X-ray mutagenesis, could exhibit distinct pattern in
the distribution of base substitutions. However, after removing \acp{snv}
shared with the DGRP I did not observe any striking differences in the mutation
spectrum of \acp{snv} between wild type and balancers (see
\cref{fig:signatures}; more detailed methods in \cref{sec:suppl_mutsign}).






\subsection{\texorpdfstring{\Aclp{cnv}}{Copy number variants}}
\label{sec:balancer_cnv}

\paragraph{Deletions} I called deletions from \wgs data of the $F_0$ and $F_1$
samples using \delly, which utilizes paired-end read signatures, and developed an
extensive set of filters to reduce false positive predictions. For both reasons
of filtering and with validation in mind I categorized deletion calls into three
size ranges: (1) Below 50~bp: these calls, predicted from split reads,
historically belong into the indel category and an additional set in this size
range was predicted with \freebayes. (2) 50~-~159~bp: a size range where
predictions based on a split read signature typically yields reliable results,
yet these calls are large enough for validation with \pcr.
(3) 160~bp and larger: in this size range additional filters based on a
read depth signal are applied prior to \pcr-based experimental validations.
The latter is highly recommended for \cnv predictions based on a paired-end read
signature in general and noted also by authors of other tools than \delly
(e.g. \citet{Layer2014}). I developed an ad hoc filtering strategy
for larger ($\geq$ 160~bp) deletion calls, which compares locally normalized
read depths between of between heterozygous and homozygous samples (see
\cref{sec:suppl_del} for details). As a consequence, shared deletions are
excluded in this size range, which is not a limitation to our study, though, as
we are primarily interested in genetic variation distinguishing the homologues.

\begin{table}[ht]
    \centering
    \begin{tabu}{lrrrr}
        \toprule
        Deletion subset           & \#Calls &  V &  I & Empirical \fdr\\
        \midrule
        \emph{<50~bp}             &   3,072 &  0 &  0 & \emph{unknown} \\
        50 - 159~bp               &     737 & 24 &  1 &     4\% \\
        160+~bp, low mappability  &      75 & 25 &  0 &     0\% \\
        160+~bp, high mappability &     395 & 24 &  1 &     4\% \\
        \midrule
        Weighted average          &   4,279 &    &    & 0.375\% \\
        \bottomrule
    \end{tabu}
    \tabcap{pcrresults}{\Ac{pcr} for deletion validation}{Number of calls as
        well as number number of validated (\textbf{V}) and invalidated
        (\textbf{I}) loci as determined by \pcr. The proportion of validated
        calls determines the empirical \ac{fdr}.
        Calls below 160~bp were not \pcr-validated.}
\end{table}

After that, \yad performed \pcr verification experiments on 75 randomly picked loci for
three subsets of deletion calls (\cref{tab:pcrresults}). A locus was considered
validated if the resulting \pcr bands of both $F_0$ and $F_1$ sample matched the
expected sizes; otherwise it was classified as invalidated. Common variants were
not submitted to \pcr validations to allow comparison between homozygous and
heterozygous samples. The final deletion call set based on \delly predictions
contained 4,279 calls. In order to get a single set of deletions I chose a lower
cutoff of 15~bp for deletions and merged the \delly call set with small
deletions predicted by \freebayes. This led to a total set of 8,340 calls on
chromosomes 2, 3, and X with a size distribution as shown in
\cref{fig:svsize_del}.

\figuretwocolumns
    {SVsize_DEL.pdf}{svsize_del}{Deletion size distribution}
    {The size distribution of final deletion calls (based on \delly and
    \freebayes) is shown. Common deletions (present in both balancer and wild
    type chromosomes) were excluded in the from 160~bp on. A higher number of
     balancer- over wild type-specific deletions was observed and this ratio
     increases with larger sizes.}
    {SVsize_DUP.pdf}{svsize_dup}{Duplication size distribution}
    {The final duplication set consisting of manually validated tandem
    duplication and a set of three non-tandem duplications is shown.}

\paragraph{Duplications} Next, I predicted tandem duplications using \delly
on \wgs and mate pair data simultaneously (see
\cref{sec:suppl_dup}). Typically this class of \sv is harder to
ascertain than deletions both utilizing paired-end and read depth signals.
Yet also for duplications orthogonal information in addition can be utilized to
gain confidence in predictions. Specifically these are read depth, which should
be considered in the context of local mappability, and \baf.
After initial filters I generated overview plots containing such information for
352 loci on chromosomes 2 and 3. An example is shown in
\cref{fig:dup_validation}, and further examples, including a negative case, can
be found in \cref{fig:dup_validation_wt,fig:dup_validation_false}.
I looked for a change in read depth between samples as well as a characteristic
change in \baf to validate calls. In a
balancer-specific duplication, for instance, the read-depth should increase only
in the $F_1$ sample and the \baf of \acp{snv} should switch from
50\% to around 33\% and 66\%.

In addition to tandem duplications predicted by \delly I found another three
non-tandem duplications by manual inspection of the \baf signal
across the genome, which were then included in the set of duplications prior
to manual validation. In total I verified a total of 122
duplications in the size range from several hundred base pairs up to 258~kb
using this strategy (\cref{fig:svsize_dup}).

\figuretextwidth{dup_validation_example.pdf}{dup_validation} % DUP00003911
    {Signals used for duplication validation shown in an example}
    {A tandem duplication at locus \textit{chr2R:7,520,066âˆ’7,526,996} is shown,
    which was predicted based on paired-end read signature using \delly.
    The additional signals
    included for a visual validation are (1) a mappability track (fraction
    of uniquely mappable reads), (2) the \baf (fraction of
    reads supporting the alternative allele of an \snv), and (3)
    the total read coverage (as genomic coverage, in 100bp windows). \Ac{baf}
    is typically the clearest signal to discriminate true from false
    copy number variants, but reuqires the presence of \acp{snv}.}

\paragraph{Summary} The \acp{cnv} I report for balancer and wild type
chromosomes contain an abundance of short deletions from 15~bp up
to 5.2~kb. The short size range of these deletions is owed to the gene-dense
character of the \textit{D. melanogaster} genome and in line with a previous
study that performed population-scale \sv calling in a population of flies,
where a median deletion size of 178~bp at a lower cutoff of 50~bp is reported
\citep{Zichner2013}.
With an experimentally estimated \fdr of 3.75\% the deletion call set can be
considered of high accuracy. Although very large deletions were
detected neither in wild type nor balancer chromosomes, I do note a slightly
higher rate of deletions on the balancer chromosomes. The ratios are 1.08 for
deletions in the size range below 50~bp, 1.3 for 50~-~159~bp, and 1.9 for larger
ones. Interstingly, a study in human cancer reported an increase in deletion
rate as a consequence of ionizing radiation \citep{Behjati2016}, suggesting that the
subtle increase in deletion number (notably in the larger size spectrum) could
be a trace of the irradiation applied during generation of the
balancer chromosomes.

Duplications, on the other hand, which underlay thorough manual validation,
where distributed evenly on balancer and wild type chromosomes. An exceptionally
large duplication of 258~kb was found on \cyo, which effectively
increases the copy number of 33 genes.






\subsection{Validation of large rearrangements using Hi-C data}
\label{sec:balancer_hic_svs}

I further aimed at refining breakpoint junctions of the major rearrangements on
the balancer chromosomes, which had previously been known only at microscopic
resolution\footnote{\label{footnote:balancer_karyotype}
    Described in terms of cytological bands, i.e. \cyo = \textit{2Lt-22D1
    33F5-30F 50D1-58A4 42A2-34A1 22D2-30E 50C10-42A3 58B1-2Rt} and
    \ac{tm3} = \textit{3Lt-65E 85E-79E 100C-100F2 92D1-85E 65E-71C
    94D-93A 76C-71C 94F-100C 79E-76C 93A-92E1 100F3-3Rt}.
    Source: \url{https://bdsc.indiana.edu/stocks/balancers/balancer_bps.html}.}.
\alek generated \hic maps of contact frequencies for both balancer and wild type
haplotypes separately in respect to the same reference genome (see
\cref{sec:balancer_cc_impl}). In these maps, which are shown in
\cref{fig:balancer_hic_rearrangements}, we identified strong contact signals off
the diagonal with a characteristic ``bowtie'' shape that coincide with gaps
close to the diagonal. Undoubtedly these signals stem from genomic
rearrangements and by comparing them to the available cytogenetic information,
\yad could assign 15/16 of these breakpoints to the respective inversions.
The last breakpoint appears to lie outside the mappable reference genome and is
thus inaccessible to read-mapping based methods.
I then searched through inversion and
translocation\footnote{These are not actual translocation, they just exhibit
    translocation signatures because the left and right arms of chromosomes 2
    and 3 are reported separately in \ac{dm6}}
predictions made using \delly on \wgs and mate pair sequencing data to
track down breakpoints at base pair resolution. This yielded base pair
resolution breakpoints in 14/15 cases, which are reported in
\cref{tab:blancer_breakpoints}.

\figuretextwidth{dup_hic_biaf.png}{dup_hic_biaf}{Large duplication on
    \texorpdfstring{\cyo}{CyO} characterized by differential Hi-C data}{
    Genomic regions around the duplication \textit{chr2L:8,957,000-9,215,000}.
    The top panel displays differential unnormalized Hi-C fragment counts
    ($\textrm{log}_2 \frac{bal}{wt}$)
    colored from red (positive) to blue (negative). The panel below shows the
    biallelic frequency. This duplication was initially discovered
    based on the characteristic biallelic frequency signal and is validated by
    a total increase of locus-internal Hi-C contacts on the balancer chromosome,
    as seen as the red triangle. Moreover, a decrease in balancer-specific
    contacts close to the diagonal left of the locus (blue) as
    well as an increase left of the upper tip of the triangular region
    suggest the duplication was inserted to the left in inverted orientation.
    Hi-C data processing and plots were contributed by \alek.}

Since our (yet unpublished) \hic- and \mps-based breakpoint refinement, two
consecutive studies had mapped---for the first time---the genomic coordinates of
several balancer rearrangements including \cyo and \ac{tm3}
\citep{Miller2018,Miller2018}. Reassuringly, when I compared our
results to the positions stated inthese studies I found precise agreement for
12/12 breakpoints provided by the studies. However, utilizing read pair and
split-read signatures in paired-end \mps data followed by assembly, these
studies failed to identify three breakpoints: for two of them (namely \textit{2L:2.14~Mb}
and \textit{2L:12.71~Mb} on \cyo) estimates at 10~kb resolution were provided but they are
off from our reported breakpoints by several kilobases. Another
breakpoint (\textit{3R:20.3~Mb} on \ac{tm3}) remained fully obscure, whereas we at least
provide a ``best guess'' based on \hic information. This strongly highlights the
technological advantage in our study: through \hic we were able to track down
the genomic positions of breakpoints down to 5~kb resolution. Then,
both short- and long-insert paired sequencing data was searched for read pairs
spanning the breakpoint junction. We found, again in agreement with these previous
studies, that only one of the breakpoint junctions in the balancer chromosomes
showed a precise re-ligation, whereas most of them are scarred by a
loss of genetic material (9 cases, max. 1.8~kb, median 114~bp) or small sequence
duplications (4 cases, median 9~bp, max 281~bp; see
\cref{sec:suppl_balancer_breakpoints}). Additionally, we predict that the
breakpoint on \ac{tm3} at \textit{3L:20.3~Mb} contains a deletion of around 17~kb, which could
explain the difficulties in ascertainment by Miller et al.

Moreover, I utilized the haplotype-resolved contact maps further to validate and
characterize other large \acp{sv}. Notably, an inversion of approximately 40~kb
on \ac{tm3} (\textit{chr3L:14.61-14.64~Mb}). was
validated by a characteristic Hi-C signal (data not shown). I also inspected
aforementioned 258~kb duplication on \cyo closer, which I had predicted by a
chatacteristic \baf signal (described in
\cref{sec:balancer_cnv}). Interestingly, the differential \hic contact map as
displayed in \cref{fig:dup_hic_biaf} confirms the duplication (seen as a red
triangle, which stands for a total increase in contacts on the balancer
haplotype), yet also suggests that the additional copy of the locus was most
likely inserted in inverted orientation to its left end. There are two signals
providing evidence for this: A reduction in contacts of the locus to its left
neighboring region (blue), suuggesting that this end of the locus is no longer
proximal to its original left neighboring region, just as if an insertion had
happened there. And secondly, there is a striking increase in contact frequency
of the right end of the duplicated locus to the left neighboring region. Taken
together this is best explained by an inverted duplication occurring in tandem.

\figuretextplusmargin{balancer_hiC_rearrangements.png}{balancer_hic_rearrangements}
    {Major rearrangements of the balancer chromosomes revealed by Hi-C}{
    \subpanel{A} Haplotype-resolved Hi-C contact frequency maps of chromosomes 2 and 3 in
    respect to the wild type reference genome are shown. Details of the data
    processing can be found in \cref{sec:suppl_hic}. The bottom triangles show
    characteristic ``bowtie-shaped'' patterns as well as gaps on the diagonal
    that demarcate breakpoint junctions of the rearranged balancer chromosomes.
    The respective sections of the reference genome are color-coded, with a grey
    circle representing centromers.
    \subpanel{B} These junctions can be followed to reconstruct the relative order of the
    balancer chromosomes such as shown in the top panel. Vertical ``bowties''
    represent a connection of segments in same orientation, horizontal
    ones segments that are inverted to one another.
    \subpanel{C} The fully reconstructed order of the balancer genomes is shown in the bottom
    panel and exactly matches previous annotation from
    karyotyping\footnoteref{footnote:balancer_karyotype}.}
\FloatBarrier












\section{Results II: Global changes in gene expression}
\label{sec:balancer_ase}

In the next step we sought to measure gene expression from both haplotypes to
determine the potential impact of the observed genetic variation on the
regulation of genes. To do so we first had to overcome biases related to
maternally deposited mRNA present in early embryos, as described subsequently.
Then I implemented robust \ase detection across multiple biological replicates
(\cref{sec:balancer_ase_impl}) and applied it to the data of our study as well
as in the scope of two additional control experiments
(\cref{sec:balancer_ase_controls}).




\subsection{Controlling for maternally deposited mRNA in early embryos}
\label{sec:balancer_maternal_rna}

\figuretextwidth[t]{ASE_histograms.pdf}{ASE_histograms}{Allelic mRNA ratio per gene}{
    Histograms of the fraction per gene of balancer RNA-seq fragments among the
    fragments that can be assigned to one of the haplotypes for three
    different RNA-seq experiments with two replicates each. The expected
    fraction of 25\% is marked by a dotted line. Counts were derived as described
    in \cref{sec:balancer_ase_impl}.}

At an early stage of the project we sequenced mRNA from 4-8h embryos of the
$N_1$ backcross (\cref{fig:balancer_crossing_scheme}) to explore the power to
detect significant \ase genes. As briefly introduced in
\cref{sec:balancer_ase_motivation}, 25\% of chromosomes 2 and 3 within the
genetic pool consist of balancer chromosomes. This means that when we separate
RNA-seq reads by haplotype we expect their distribution to be centered around a
balancer fraction of 25\%. As the left panels of \cref{fig:ASE_histograms}
display, this initial experiment was marked by a (unsatisfactory) wide
distribution. More precisely, it did not follow a binomial distribution, which
is a model applied by many \ase detection methods, but showed a large
overrepresentation of wild type mRNA instead. Only after discussions with
\garfield we figured that, since we had crossed female wild type flies with double
balancer males, the reason for this skew in the distribution could be attributed
to \explain{maternally deposited mRNA}{Before maternal-to-zygotic transition
    zygotes or early embryos do not transcribe their own genes but utilize mRNA
    of the maternal cell that was loaded into the egg during oogenesis. After
    around two hours of embryonic development a large portion of maternal mRNA
    is actively degraded, yet some maternal transcipt can remain
    \citep{Tadros2009}}
present in early embryos as well as unfertilized eggs remaining in the sequenced
pool.

To solve this issue, \yad initiated a new experiment with modified conditions: First, she
collected embryos only in a later, 6-8h time window when the impact of maternal
mRNA is expected to be weaker. Secondly, she manually sorted out unfertilized eggs
from the collection of embryos. And thirdly, she derived two different crosses,
one with a paternal double balancer line (as initially), termed \Npat,
and another one with a maternal double balancer line, \Nmat. The potential
presence of maternally deposited mRNA will then cause a shift in exactly opposite
directions in both these lines, which can be accounted for during \ase
calling. These considerations are nicely reflected in the histograms on the
right of \cref{fig:ASE_histograms}. More narrow peaks indicate a much lower
fraction of maternal transcripts, and a shift from left to right between \Npat
to \Nmat point to remaining maternal transcripts that can be identified as such.






\subsection{Allele-specific expression detection}
\label{sec:balancer_ase_impl}

The basic idea of measuring \ase is to separate the RNA-seq signal for a given
gene into its two alleles and to test the resulting ratio against a null
hypothesis of both alleles being represented equally. This is typically done by
mapping RNA-seq reads to a common reference and inspecting sites of tagging
variants (i.e. \acp{snv}), although some approaches also rely on mapping to
personalized genomes to overcome mapping biases \citep{Rozowsky2014} and future
extensions of this idea will likely lead towards \explain{reference graphs}{
    Representation of different alleles or haplotypes simultaneously in a graph
    structure. A diploid genome, but also the haplotypes of a whole populations
    could be represented in such a data strucutre to avoid a reference bias.
    However, such approaches, which are being actively researched, require new
    methods for processing and interpretation of sequencing data}
\citep{Dilthey2015,Marschall2016,Novak101378}. Accurate unbiased \ase detection
requires many different considerations, as nicely elaborated by \citet{Castel2015},
and there is an abundance of software available to perform the different steps
involved \citep{Skelly2011,Mayba2014,Harvey2014,Pirinen2015,Romanel2015,VandeGeijn2015,Liu2016}.
However, due the particular requirements of our study, they are mostly not
applicable here: First, we needed to test \ase against an unusual fraction of 25\%,
whereas some models (reasonably) expect a 50\% null hypothesis. Second, we
would like to incorporate multiple replicates to reduce statistical noise, and third,
we expected the different replicates to entail different \ase ratios for some
genes (i.e. the ones with maternal mRNA), a situation that is unique to our
study. I hence implemented an approach to \ase detection that captures these
features.

As a first step I mapped RNA-seq read pairs to \ac{dm6} using \STAR. I then
separated read pairs into haplotypes using a simple Python script based on
\textsc{pysam}\footnote{Found online at \url{https://github.com/pysam-developers/pysam}}.
This approach considers all \acp{snv} that a read pair overlaps and can hence
also detect if observed alleles within the fragment are disaccordant---a fact
that we deemed especially important for the separation of \hic
data later on (\cref{sec:balancer_cc_impl}). This approach also differs
from the \snv-centric approaches of other tools \citep[for example]{Castel2015} that
yield allelic read counts at each variant. Furthermore, I preferred to derive a
single sum of fragments for each allele of a whole gene, instead of considering
different sites of variation within a gene. This comes with caveat that
it might mask potential allele-specific alternative exon usage, which, for instance, was
recently shown to occur at cites of CTCF binding motifs
\citep{Ruiz-Velasco2017} and which can be identified using profound
statistical tests \citep[for instance]{Skelly2011}. The advantage, though, is
that our method does not require an additional model to integrate across sites of
variation within a gene (such as presented by \citet{Mayba2014}) and can be
expected to be more robust to local fluctuations in coverage than single-\snv
analyses. On average we could separate 28\% of RNA-seq read pairs (read lengths
2x~144~bp) as shown in \cref{tab:balancer_rnaseq}.

\begin{table}[ht]
    \centering
    \begin{tabu}{lrrrr}
        \toprule
        Data set   & Total reads   & Balancer   & WT   & Disaccordant \\
        \midrule
        \Npat 1. replicate & 51,560,368  & 6.151\% & 22.489\%  & 0.136\% \\
        \Npat 2. replicate & 52,618,060  & 5.762\% & 21.471\%  & 0.139\% \\
        \Nmat 1. replicate & 54,521,194  & 8.030\% & 20.879\%  & 0.142\% \\
        \Nmat 2. replicate & 48,179,068  & 7.834\% & 20.352\%  & 0.153\% \\
        \bottomrule
    \end{tabu}
    \tabcap{balancer_rnaseq}{Haplotype separation of RNA-seq data}{
        Fraction of RNA-seq read pairs that could be assigned to either one of
        the haplotypes balancer or wild type (WT) ot that contained conflicting
        variants (disaccordant). The difference to 100\% in each comes from read
        pairs that could not be assigned.}
\end{table}

At last, each gene should be statistically tested against the null
hypothesis. The commonly used Binomial test is known to inflate small p-values
\citep{Harvey2014} and further cannot integrate our replicate design. I thus
used \deseq, a well-established tool for differential RNA-seq analysis, to test
for \ase. Given the haplotype-specific counts for each gene, \deseq is able to
detect divergence from the 25\% fraction across multiple replicates and can
further cope with a potential shift in \ase ratio between the replicates caused
by maternally deposited mRNA, which will increase the p-value. It also performs
multiple testing correction at a controlled \fdr (more detail in
\cref{sec:suppl_deseq}).

\figuretextwidth{DESeq2_N1_6_8h.pdf}{DESeq}{ASE analysis based on
    \texorpdfstring{\deseq}{DESeq2}}{Average haplotype-resolved fragment count
    per gene across four replicates vs. log ratio of balancer over wild
    type counts are shown. A single outlier to the right was trimmed. }

Among 5357 genes on chromosome 2 and 3 for which there was sufficient
haplotype-resolved coverage we detected 512 significant \ase genes (9.6\%) at a
\fdr of 5\%. The median fold change was 1.8 with a symmetrical distribution in
both directions (see \cref{fig:DESeq}). At a minimum fold change of at least
1.5, 343 genes remain (6.4\%). These percentages are similar to the fraction of
\ase genes in $F_1$ crosses of two distinct DGRP lines, where 5-10\% ASE genes
are reported (Furlong lab, unpublished work).






\subsection{RNA-seq control experiments}
\label{sec:balancer_ase_controls}

We carefully considered some of the assumptions made in the \ase analysis and
decided to carry out two important control experiments. A first question was
whether balanced chromosomes carry a higher fraction of \ase genes than
unbalanced chromosomes such as chromosome X.
\figuremargin{ASE_chrX_numbers.pdf}{ase_chrX}{ASE per chromosome}{
    Balancer fraction of allele-specific RNA-seq fragments on chromosomes 2, 3,
    and X in a adult, female flies from an $F_1^{f}$ line. Error bars
    indicate 99\% confidence intervals from a binomial test.}
This could not be answered in our
$N_1$ backcrosses, where chromosome X eventually became homozygous due to the
maternal wild type line. Instead, we performed RNA-seq on adult, handpicked
female flies from the $F_1$ generation, named $F_1^f$, and sequenced two
replicates. In this sample chromosome X is paired with its (non-rearranged,
but diverged) homologue at a ratio of 50\%, too, and can thus be compared to
chromosomes 2 and 3. The two chromosome X homologues contain less, but still
sufficient distinguishing variation for \ase analysis (see \cref{tab:snvs2}).
I executed \ase analysis as described in \cref{sec:balancer_ase_impl} on $F_1^f$
RNA-seq data and found 301/4940 (6.1\%) significant \ase genes on chromosomes
2 and 3 and 24/463 (5.2\%) on chromosome X. The lower fractions compared to
embryonic $N_1$ samples are of no concern, as gene expression is expected to
be highly tissue- and developmental\ stage-dependent. Importantly though, the
fraction of \ase genes on balancer chromosomes is not significantly higher than
on chromosome X (p=0.47, Fisherâ€™s exact test) as shown in \cref{fig:ase_chrX}.

Furthermore we wanted to test whether the existence of multiple different
genotypes in the $N_1$ generation could cause potential \textit{trans} effects,
e.g. whether genes on chromosome 3 would depend on the presence of \cyo
(i.e. balancer chromosome 2), and vice versa. In order to test this \yad generated
three new $F_1$ generations of adult flies with different genotypes: one line
(\Fcyo) with only chromosome 2 balanced, another one (\Ftm) with only
chromosome 3 balanced, and a last one with double balancer configuration ($F_1$).
The latter differs from $F_1^f$ by including both sexes and it was sequenced
together with \Fcyo and \Ftm to prevent batch effects. Note that
in these single-balancer lines, haplotype separation is only
possible on the balanced chromosomes themselves, for which \acp{snv} had been
previously mapped.

I then determined \ase genes on chromosome 2 using two replicates of \Fcyo
and two replicates of $F_1$ as described previously. Respectively, I did
the same for chromosome 3 using \Ftm and $F_1$.
Together I obtained 478/6356 (7.5\%) significant \ase genes across both balancer
chromosomes â€“ a number that is fairly in accordance to the results of the
previously sequenced $F_1^f$ generation. Moreover, the balancer-to-wild\ type
ratios of genes from these two samples are in high correlation (Pearsonâ€™s
$r^2$~=~0.824). Relevantly, I was then able to test the interaction term of \ase
and genetic background using \deseq: I asked, for instance, for which genes on
chromosome 2 the \ase ratio significantly changes between $F_1$ (contianing
\ac{tm3}) and \Fcyo (not containing \ac{tm3}).
Again I executed this analysis separately on both chromosomes and found a
significant interaction for 77/5981 genes (1.3\%) at an FDR of 5\% and a
minimum fold change of 1.5. In order to put this number into context, I also
estimated the differences in \ase ratio between the two adult double balancer
samples (i.e. $F_1$ vs. female-only $F_1^f$). These samples also exhibit high
correlation (Pearsonâ€™s $r^2$~=~0.848) and I detected 20/5382 (0.4\%) genes
for which the balancer/wild type ratio is significantly influenced by the
genetic background (also at an FDR of 5\% and a minimum fold change of 1.5).
Requiring a fold change of at least two, these rates decrease to 39/5981
(0.65\%) and 11/5382 (0.2\%).
The analysis is shown in \cref{fig:ASE_single_balancer_control}.

We concluded from this analysis that there is indeed a \textit{trans} effect of
the genetic background on the haplotype ratios of genes. However, this effect is
small, maybe negligible, and even between the genetically seemingly identical
lines such as $F_1$ and $F_1^f$ significant changes are observed.
We together reasoned that the analysis of a double balancer line (in contrast
to a single balancer line) adds another variable to our study design that should
be kept in mind (as it has a small, but notable effect), yet that this strategy
is still preferable due to the sheer amount of chromosomal rearrangements that
can be studied.







\section{Results III: The interplay between SVs and differentially expressed genes}
\label{sec:balancer_sv_interplay}

After characterizing \acp{sv} and detecting differentially expressed genes we
were interested in their relationship. In the upcoming section I describe how we
explored their correlation and which \acp{sv} we believe are causal to \ase
genes. Later on, \cref{sec:balancer_cc} will specifically cover the role of
chromatin conformation.



\subsection{Genes affected by large rearrangements}
\label{sec:balancer_ase_breakpoints}

During the characterization of the large chromosomal rearrangements
I observed that they often happen to disrupt genes. In fact, 11 out
of 15 breakpoints do so, affecting genes such as \textit{Src42A}
(Draper-Shark-mediated signalling immune response pathway), \textit{GlyP}
(carbohydrate metabolism pathway) and \textit{p53} (tumour suppressor). The
deletion present at breakpoint \textit{chr3R:20.3~Mb} even spans a complete
gene, which is consequently lost in the balancer chromosome. All cases are
listed in \cref{tab:blancer_breakpoints}. We assume that disrupted genes become
non-functional or at least severely truncated on the balancer chromosomes,
which is why most of them also show up in the set of \ase genes.

These frequent gene knockouts are perhaps surprising, as it is expected that
selective forces during the creation of balancer chromosomes would rule out
breakpoints disrupting genes. It can thus be assumed that these
particular genes are not dosage-dependent. Balancer chromosomes are indeed known
to tolerate recessive mutation due to the reduced influence of
natural selection \citep{Araye2013}. Yet the frequency of gene knockouts
without apparent phenotypic consequences should be acknowledged here.





\subsection{Positional clustering of ASE genes}
\label{sec:balancer_ase_clustering}

Following the intuition that \ase genes could be preferably located around
rearrangement junctions on the balancer chromosomes, I specifically
looked for an enrichment of \ase genes at breakpoints compared to random
positions places in the genome (\cref{fig:suppl_ase_genes_around_bps}).
Surprisingly, only a single breakpoint, namely \textit{chr3R:20.31~Mb},
appeared to be surrounded by more \ase genes than expected by random chance: 5
out of 6 neighbouring genes, which span a region of around $\pm$~50~kb.
By means of integrated visualization (\cref{sec:balancer_visualization}) of this
locus I noted that two genes were highly up-regulated left of the breakpoint,
another gene down-regulated right of the breakpoint, and an additional gene
disrupted by the breakpoint itself. Notably, this was the only breakpoint
that could potentially involve a chromatin-structure-related mechanism affecting
gene regulation. However, closer inspection revealed that the \ase signal of
these genes is likely caused by chimeric transcription across the breakpoint
junction. We observed such chimeric, likely non-functional expression happening
on an appreciable set of other genes, too, as I elaborate in
\cref{sec:balancer_ase_mei}.

\figuretextwidth{ase_clustering.pdf}{ase_clustering}{Distance between
    neighboring ASE genes}{This is a histogram of distances between neighboring
    significant ASE genes (blue) as well as between a random set of control
    genes (grey), which were sampled from expressed, but non-ASE genes 500 times.
    Errorbars indicate the 5\% and 95\% quantiles of random sub-sampling.}

I then turned to a more unbiased approach and searched the set of \ase
genes for positional clustering anywhere in the genome.
As \cref{fig:ase_clustering} demonstrates, \ase genes are generally located as
far away from another as random expressed genes, with the exception of
a small set of genes in a distance of approximately 3~kb to one another.
These, as the integrated visualization again quickly revealed, mostly belong to
the large duplication on \cyo (see \cref{sec:balancer_cnv,fig:dup_validation})
and not to one of the chromosomal rearrangements. We thus concluded that there
is no single spot in the balancer genomes that shows an enrichment in
significantly mis-regulated genes; instead \acl{ase} is fairly
evenly distributed across the genome. In addition, these analyses hinted at
mechanisms unrelated to chromatin conformation that could create \ase signal,
which are further explored subsequently.





\subsection{\texorpdfstring{\ase}{ASE} signal related to changes in copy number}
\label{sec:balancer_ase_cnvs}

% Testing this matrix for randomness:
%           no_CNV CNV_overlap
% no_ASE      4551         292
% ASE_genes    439          73
%
%         Fisher's Exact Test for Count Data
%
% p-value = 2.63e-10
% alternative hypothesis: true odds ratio is not equal to 1
% 95 percent confidence interval:
%  1.940543 3.426937
% sample estimates:
% odds ratio
%   2.591054
\figuretextwidth{ASE_cnv_overlap.pdf}{ASE_cnv_overlap}{Log fold change of
    \texorpdfstring{\ase}{ASE} genes overlapping \texorpdfstring{\acp{cnv}}{CNVs}}
    {Expression level log fold change of 73 significant \ase genes that overlap
    \acp{cnv}. Two genes overlap multiple different \acp{cnv} (\textit{ambiguous}).
    A positive log fold change means higher expression in the balancer haplotype
    and vice versa.}

Next, I tried to understand how much of the \ase signal is caused by \acp{cnv}.
I therefor inspected significant \ase genes that have at least one of their
exons (incl. $3^\prime$ or $5^\prime$ untranslated regions) affected by a \cnv.
This overlap yielded 73 genes, which is a significantly higher fraction
among the \ase genes (14\%) than among expressed, but non-\ase genes (292 cases,
6\%; p-value $<10^-9$, Fisher's exact test). \Cref{fig:ASE_cnv_overlap} shows
the effect \acp{cnv} have on the expression of \ase genes. As expectable, a
clear trend towards a dosage effect can be seen, i.e. that a deletion within a
balancer gene decreases and a duplication increases the balancer
expression (seen by a positive log fold change), and vice versa for
wild-type-specific \acp{cnv}. Duplications, which are typically much larger in
size than deletions (\cref{fig:svsize_del,fig:svsize_dup}), show a
clearer signal of differential expression as they often duplicate whole genes.
Yet it cannot be expected that a higher copy number neccessarily leads to an
increase in transcribed RNA. For example, a partial duplication could disrupt a
gene in a way that its expression decreases. Accordingly, the impact on
expression is less clear for deletions that typically affect only single exons.
The specific explanation for each gene could, if need to, be unravelled
in case-by-case fashion.

We derive as a conclusion from this analysis that up to 14\% of significant \ase
genes could be explained directly by \acp{cnv} affecting their exons---ergo,
these genes are unlikely to be involved in chromatin\ conformation-related
mechansisms.





\subsection{Mobile element insertions can give rise to strong ASE signals}
\label{sec:balancer_ase_mei}

\figuretextwidth{MEI_example.pdf}{balancer_mei_example}{Example of a
\texorpdfstring{\acs{mei}}{MEI} driving ectopic expression of a gene}{
    RNA-seq tracks along the gene \textit{Ptp52F} on chromosome 2R showing the
    total RNA-seq reads (total read coverage in absolute numbers; bottom panel)
    as well as the portion of RNA-seq reads that could be resolved to the
    wild type (top panel) and balancer (middle panel) haplotypes. Colored
    vertical lines represent \acp{snv} used for haplotype-separation. RNA-seq
    tracks were visualized using \igb.}

In a thorough manual data inspection \yad found a sizeable number of genes with
an RNA-seq signal that did not start at their expected transcription start site.
We further found that this expression was often only present on one of the two
haplotypes, and that the RNA-seq coverage started at a seemingly random
position. Interestingly, despite starting in an intronic or even intergeneic
region, this signal would often pick up a characteristic splicing pattern
(high signal within exons, no or low signal along introns) from the end of the
first traversed exon. \Cref{fig:balancer_mei_example} depicts an example gene
with strong balancer-specific gene expression.

We speculated that this might be driven by the insertion of mobile elements in
the positions where the RNA-seq signal started. Hence I developed a computational
pipeline that extracts DNA-sequencing reads of a given genomic region,
computes an assembly of these reads using \spades and compares the assembled
contigs to a database of common transposable elements in \textit{Drosophila}.
The comparison is done via read mapping and confirmed visually using the dotplot
functionality of \maze developed during previous work (\cref{sec:maze}). Using this
approach I found transposable element insertions in 42 loci. Based on the
position and orientation of the mobile element (i.e. mobile elements contain
their own transcription start site) I evaluated whether the aberrant \ase
signal of the gene could be caused by chimeric transcription from the mobile
elementâ€™s promoter. This is the case in 37 loci (\cref{tab:meilist}),
of which 25 were expressed high enough to be tested for \ase.

\figuretextplusmargin{MEI_impact.pdf}{balancer_mei_impact}{ASE genes associated
    to MEI}{List of significant ASE genes (x axis) ordered by log fold change
    (balancer/wild type, y axis). Genes that are very likely dis-regulated due
    to an MEI driving their chimeric expression are highlighted in blue.}

Astonishingly, all 25 genes were also called to show significant \ase, partially
with extreme fold changes reported as shown in \cref{fig:balancer_mei_impact}.
In the aftermath, this is
little surprising given that an \mei drives expression of a gene that, even if
weak, can be orders of magnitudes larger than the second copy which is only
weakly expressed, if at all. \Acp{mei} could in principle also cause chimeric
expression of higher expressed genes, but in such cases the additional
mRNA is probably negligible compared to the active transcription of both
haplotypes. At last, it should be kept in mind that this analysis centered on
differentially expressed genes with aberrant transcription start sites, hence
the full impact of \mei is still under-explored.









\section{Results IV: Changes in chromatin conformation}
\label{sec:balancer_cc}

Here, I summarize our findings in respect to the chromatin conformation of
balancer and wild type chromosomes. Most analyses described below were performed
by \alek unless clearly marked by ``I''. These investigations were ongoing at
the time of writing and might hence be subject to change prior to final
publication.

\subsection{Differences in chromatin conformation between wild type and balancer chromosomes}
\label{sec:balancer_cc_impl}

To study chromatin organization, we performed \hic on nuclei from 4-8~h embryos
of the $N_1$ generation in two biological replicates. We sequenced the \hic
library in as many as 24 lanes on an Illumina machine to obtain high resolution
contact frequency maps. We then annotated read pairs by haplotype in the same
way RNA-seq data was annotated beforehand (\cref{sec:balancer_ase_impl}).
Moreover, a strict filter procedure was designed following recommendations from
\citet{Ramirez2018}, which is described in the appendix (see \cref{sec:suppl_hic}
and \cref{tab:hic_fragments} for details on the effect of different filters).
Finally, 117 million pairwise contacts remained on the wild type chromosomes
and 35 million on the balancer chromosomes to build \hic maps in 5~kb
resolution. It is interesting to note that more \hic read pairs could be
separated into haplotypes relative to RNA-seq read pairs, despite shorter read
lengths (38\% vs. 28\%, approximately). This is a consequence of the large
``insert sizes'' of \hic data, which make it more likely for a pair to bridge
genomic regions with low \snv density \citep{Edge2017}.

At first we visually compared the contact maps of matching genomic regions of
balancer and wild type chromosomes using, among others, the visualization tools
described in \cref{sec:balancer_visualization}. Except for in proximity of the
breakpoints of large rearrangements, at which contact frequencies of the two
differently arranged chromosomes cannot simply be compared, we observed a
striking similarity between the balancer and wild type chromosomes. As it
appears, the rearrangement on the balancer chromosomes affects the
three-dimensional chromatin organization only locally around breakpoints and
anywhere else it remains mainly unaffected. In order to quantify this we
calculated the \tad separation score as a one-dimensional feature of the local
compactness \citep{Ramirez2015,Ramirez2018}. In fact we calculated score
profiles for seven window sizes\footnote{50~kb, 60~kb, 80~kb, 100~kb, 130~kb,
160~kb, and 195~kb} and averaged them at each genomic position. We found a high
correlation ($r^2 = 0.93$) between balancer and wild type haplotypes,
confirming our visual impression.

Despite the similarity on the large scale, single pairwise contacts can in fact
differ between the haplotypes. We hence tested for significant differential
pairwise contacts using \deseq by providing haplotype-resolved fragment counts
in 5~kb bins utilizing both \hic replicates. Among 509,217 tested pairwise
connections that had sufficient read support we found a total of 48,206
significant differences. Despite attempts to normalize for distances in the
altered genomic order, 1,388 (2.9\%) of the detected significant differences
spanned across a breakpoint of the large chromosomal rearrangements---these
pairs were excluded from the further analysis.

\figuretextwidth[t]{HiC_ASE_genes_Alek.pdf}{HiC_ASE_genes}{Differential \hic
    contact around genes}{This shows differential \hic contacts between gene
    promoters (at 0~kb; all genes are aligned and oriented with the gene and
    downstream sequence to the right) and all other 5~kb bins within $\pm$100~kb.
    On the y-axis, the fraction of significantly differential \hic contacts among
    all \hic contacts is shown. The signal is averaged across \ase and non-\ase
    genes and plotted here including 90\% confidence interavls.}

We were then interested in whether changes in chromatin architecture are
associated with differential gene expression. Except for breakpoint-spanning
contacts. To study a potential relationship between \ase and differential
contacts, we centered \hic contacts at gene promoters (i.e. at the 5~kb bin
surrounding the promoter) and averaged across many genes. In
\cref{fig:HiC_ASE_genes}, genes are aligned and oriented according to their
transcription start site and the average fraction of differential contacts among
all contacts are shown on the y-axis. Interestingly, we found that on average
there were more differential \hic contacts reaching to the promoters of
\ase genes than to promoters of non-ASE genes. This difference is pronounced
within 50~kb around the gene and notably vanishes at a distance of 100~kb.
This data suggests that chromatin conformation might indeed play a role in
regulating gene expression. However, it is unclear whether the conformational
changes are cause or consequence of altered gene expression and how much this
association varies between single loci.





\subsection{TAD structure around breakpoints}
\label{sec:balancer_tads_at_bp}

One of the crucial questions of the study was how chromatin conformation,
especially \tad structure, would change around breakpoints of large chromosomal
rearrangements. Based on \hic data \acp{tad} can be predicted using one of
several available algorithms â€“ however, given the differences in the predictions
from these methods, this is still a major challenge in the field
\citep[notably figure 3]{Forcato2017}. We utilized \tad calling based on the
\tad separation score \citep{Ramirez2018} to predict \acp{tad} only on the
contact map of the wild type chromosomes. This resulted in 880 domain calls with
a median size of 125~kb, which is a slightly lower segmentation than reported by
\citet{Sexton2012} (1,170 domains with a median size of 62~kb).

We then inspected the \acp{tad} around the 15 breakpoints of large chromosomal
rearrangements. Most \tad callers, including the method we applied here, report
\tad boundaries as infinitesimal points instead of larger intervals, which we
think does not capture the reality well. In order to distinguish intra-\tad
space from boundaries, I thus required a minimum distance of 20\% (of \tad size)
to both ends of the interval in order to consider a \tad disruption.
According to this computational analysis, 12/15 breakpoints fall into \acp{tad}
rather than into boundaries (\cref{tab:tads_at_bp}). Since \tad calls did not
always match our visual impression, \cref{tab:tads_at_bp} also includes a manual
annotation, in which we classified 10/15 breakpoints as breaking \acp{tad}.

\begin{table}[ht]
    \centering
    \begin{tabu}{lrrrrl}
        \toprule
        \multicolumn3{c}{Breakpoint coordinates} & \acs{tad} & Distance & Evaluation\\
        \midrule
        chr2L  & 2,137,067   & 2,137,075   &  180~kb &       22,925  &  $+$   \\
        chr2L  & 12,704,649  & 12,704,657  &  335~kb & \sout{14,649} &  $-$   \\
        chr2L  & 9,805,567   & 9,805,575   &  200~kb &       94,425  &  $++$  \\
        chr2R  & 14,067,771  & 14,067,782  &  215~kb &      107,218  &  $++$  \\
        chr2R  & 6,012,459   & 6,012,739   &   40~kb &        7,261  &  $--$  \\
        chr2R  & 21,971,918  & 21,972,072  &  205~kb &      101,918  &  $+$   \\
        chr3L  & 6,925,034   & 6,926,125   &  230~kb &       53,875  &  $++$  \\
        chr3R  & 9,943,831   & 9,944,040   &  115~kb &       40,960  &  $+$   \\
        chr3L  & 15,150,269  & 15,150,272  &  120~kb &  \sout{5,269} &  $--$  \\
        chr3R  & 23,050,763  & 23,050,764  &  380~kb &       79,236  &  $+$   \\
        chr3L  & 19,386,273  & 19,388,151  &  310~kb &      126,273  &  $+$   \\
        chr3R  & 20,637,930  & 20,637,930  &  275~kb &       82,930  &  $-$   \\
        chr3L  & 22,637,876  & 22,637,952  &  310~kb &       82,048  &  $+$   \\
        chr3R  & 31,653,695  & 31,653,707  &  135~kb &       43,695  &  $++$  \\
        chr3R  & 20,308,200  & 20,325,700  &  260~kb & \sout{13,200} &  $--$  \\
        \bottomrule
    \end{tabu}
    \tabcap{tads_at_bp}{Overview of \texorpdfstring{\tad}{TAD} calls at
    breakpoint positions}{  \Ac{tad} annotation at breakpoints of large
    chromosomal rearrangements. \Acp{tad} were called only based on wild type
    \hic data using the approach of \citet{Ramirez2018}. Column \emph{\acs{tad}}
    contains the size of the overlapping \tad call and \emph{Distance} the
    distance of breakpoints to the \tad boundaries. Only in three cases (marked
    by \sout{Distance}) breakpoints were closer to \tad boundaries by more than
    20\% of \tad size---these cases supposedly do not interrupt \tad structure.
    The \emph{Evaluation} column contains a manual assessment
    of \acp{tad} based on inspection of \hic maps. $+$ means that a \tad is
    likely interrupted, $++$ it is clearly interrupted, and $-$
    ($--$) that breakpoints likely (clearly) fall into \tad boundaries.}
\end{table}

Aforementioned computational criteria can also be applied to \tad calls from
balancer \hic data in respect to the balancer genomic order, in order to
evaluate formation of new \acp{tad} around the junction points. According to
computational analysis (again using the 20\%-distance criterion), 8/15 of these
junction points appear to reside within newly formed \acp{tad}. A manual
inspection also yielded 8/15 cases of \tad formation, yet in two examples my
classification differed from the computational analysis.

Despite inaccuracies in computationally defining \acp{tad}, the numbers reported
here make clear that breakpoints in fact disrupt \tad structure of the wild type
chromosomes. Moreover, in the new genomic order of the balancer chromosome, new
\acp{tad} have formed across the junctions in many cases. This is for example the
case at locus \textit{chr2R:14.0~Mb}, which is visualized in \cref{fig:ggbio1}
(and explained in more detail subsequently). Domains that were not affected by
breakpoints, on the other hand, did not seem to change at all
(\cref{sec:balancer_cc_impl}).
Interestingly, when we inspected newly formed \acp{tad} spanning breakpoint
junctions, we observed that they often expanded up to neighboring domains,
effectively re-using previous \tad boundaries. This has not been formally tested,
though. However, as already noted earlier (\cref{sec:balancer_ase_clustering})
\tad disruptions or formations appear not to have a notable effect on gene
expression. \Ac{ase} genes are not enriched around breakpoints, but rather
spread evenly across the genome, and no evidence for the presence of an
enhancer\ hijacking-like mechanism could be found in this study.






\section{Integrated visualization of genomic loci}
\label{sec:balancer_visualization}

At last, this section covers a methodological aspect that was essential in
eriving many of the aforementioned results, especially of section
\cref{sec:balancer_tads_at_bp}.

With a variety of data available in our study including RNA-seq and \hic, we
faced the challenge of how to explore these datasets around loci of interest.
Genome Browsers, which were specifically designed for this task, can show
multiple tracks of different data in respect to the same genomic coordinates
\citep{Freese2016,Thorvaldsdottir2013,Gramates2017}. In our particular case
existing solutions were not satisfying, though. First of all, we would like to
include \hic maps into our figures. Second, we needed all data to be
represented in the two different genome assemblies resulting from balancer and
wild type chromosomes. Only recently some tools (published or at pre-print
stage) addressed the first point \citep{Ramirez2018,Kerpedjiev2017}, yet not
allowing the flexibility we required. This is why \alek and I utilized the
R-package \textsc{ggBio} \citep{Yin2012} to create our own tailored
visualization.

\figuretextplusmargin{ggbio_dm6_2R_14Mb.jpg}{ggbio1}{Integrated visualization
    around breakpoint \textit{2R:14.1~Mb}}{
     visualized via custom plot functions based on
    \textsc{ggBio} \citep{Yin2012}. The figure is explained in the main text.}

\Cref{fig:ggbio1} shows an excerpt of one of the plots we generated. The top
panel of the figure highlights \circled{a} the position of the genomic locus
within the respective chromosomal assembly (here the wild type reference genome
\textsc{dm6}) with colors that roughly correspond to the initial illustration of
balancer chromosomes (\cref{fig:balancer_hic_rearrangements}). Below, we
incorporated different \hic maps. IN \cref{fig:ggbio1}, there are is one \hic
map created only on \circled{b} wild type data and one only on \circled{c}
balancer \hic data. Around the breakpoint of one of the large rearrangements,
for example at \textit{chr2R:14.07~Mb} as shown here, these haplotype-resolved
\hic maps quite remarkably display the missing connections in the balancer
chromosome. If these signals were plotted in respect to the genomic order of
the balancer chromosome, this gap would be visible in the wild type track
respectively (see \cref{fig:ggbio2,fig:ggbio3} for exactly this).
Optionally, the plot can include additional \hic maps, for
example generated from the complete data or showing differential contacts,
as in \cref{fig:dup_hic_biaf}.

The middle part of the figure may contain arbitrary one-dimensional genomic
features. The example figure includes \circled{d} deletion calls on balancer and
wild type haplotype, \circled{e} DNase-I hypersensitivity tracks (Furlong lab,
unpublished data), and strand-specific RNA-seq tracks including \circled{f} all
RNA-seq reads, only \circled{g} wild\ type-specific RNA-seq reads and only
\circled{h} balancer-specific RNA-seq reads (positive: coverage on plus strand;
negative: minus strand). Other signals that could be displayed
are \snv positions, other \sv classes, insulation scores, or predicted
\tad boundaries.

The bottom part contains \circled{i} gene annotations and maps the
genomic coordinates of genes to a table listing additional information about
these genes, including their names. Shown here are the \circled{k} total
expression level, the \circled{j} log fold change of RNA levels in balancer
compared to wild type haplotypes, and whether this change is significant
according to our ASE analysis (orange color for significance).
Such a breakpoint-centered plot is accompanied by two additional
plots that represent the same locus in the genomic order of the balancer
chromosomes, which can be found in the appendix (\cref{fig:ggbio2,fig:ggbio3}).







\section{Conclusions}
\label{sec:balancer_concl}

In this study, we aimed at testing the influence of \acp{sv} on chromatin
organization and gene expression. Particularly, we wanted to assess whether an
enhancer hijacking mechanism, which affects gene regulation by restructuring
chromatin in 3D, occurs in a phenotypically healthy organism and how frequent it
might be. Based on the study design and experiments of \yad, I deeply
characterized balancer chromosomes of \textit{Drosophila melanogaster} in terms
of variation, gene expression and chromatin conformation. I unraveled the exact
positions of the cytogenetically annotated rearrangements, which were partly
found concurrently by two studies \citep{Miller2016,Miller2018} and which are in
perfect agreement to the results stated therein. I further characterized the
balancer and wild type chromosomes in respect to \acp{snv}, small indels,
deletions and duplications, the latter two of which were validated
experimentally or computationally using independent signatures. Based on
haplotype-tagging \acp{snv}, I developed a methodology to detect \acl{ase}
across four biological replicates and including a correction for maternally
deposited mRNA. I find that genes with significant differential expression
between both haplotypes are distributed rather equally across the genome and
not enriched at breakpoints.

Utilizing \hic data, \alek found that global chromatin structure, notably the
patterning into \aclp{tad}, is generally not different between wild type and
balancer chromosomes, despite their tremendously different genomic order. This
supports the current notion in the field that \acp{tad} are established by their
boundaries and are hence a local feature of chromatin. Nevertheless, exactly at
the breakpoints local chromatin structure necessarily changes, as is exemplified
for one locus in \cref{fig:ggbio1}.

Based on computational \tad predictions (\alek) and manual inspection, I found
that in 10-12 out of 15 cases \acp{tad} of the wild type chromosome were
disrupted and in approximately 8 cases, new \acp{tad} were formed within the
balancer chromosomes. Since only few \ase genes are found within these \acp{tad}
and we generally could not detect any enrichment for \ase genes close to
breakpoints, we do not think that these changes in chromatin structure affect
gene expression in our model. This is in stark contrast to previous studies
(\cref{sec:disrupting_tads}), where typically a single \tad disruption is
reported to ectopically express genes in non-physiological amounts. A possible
explanation for this apparent discrepancy is natural selection. Just as in our
study \tad disruptions had undergone selection to guarantee viability of the
organism, the alterations in chromatin structure analyzed in many of these
aforementioned studies had been selected for a pathological phenotype such as
cancer or Mendelian diseases.

Selective pressure might indeed have prevented misregulation via enhancer
hijacking-like mechanisms. However, at the same time a great amount of
variability with consequences on gene expression was tolerated: For instance,
\acl{cnv}, one of which duplicates 33 genes at once, are abundant and believed
to alter expression of 73 genes. Rearrangement breakpoints directly disrupt
genes, including \textit{p53}, in 11/15 cases. And while these changes typically
alter expression ratios in the range of not more than two- to three-fold, there
is a relevant number of \aclp{mei} that drive expression of at least 25 genes.
This typically leads to drastic \ase signals in the order of dozens to hundred
times fold change. However, the resulting chimeric transcripts lack exons or
contain intergenic sequence, so it is unclear whether they are functional.

Eventually, hundreds of genes remain that are significantly differentially
expressed but for which we have no explanation at hand. By correlating these
genes with significant differences in three-dimensional chromatin interactions,
we found that indeed chromatin structure appears to be preferably altered around
\ase genes (\cref{fig:HiC_ASE_genes}) and might consequently play a role in
mediating transcriptional regulation. Recently, \citet{Li2018} described a clear
correlation between changes in intra-\tad density and up- or down-regulation of
gene expression during differentiation of leukemia cell lines. It was suggested
that both arise simultaneously as consequence of an altered epigenetic state of
the \tad, as for example \citet{LeDily2014} observed. This could potentially
explain our observations, too. It is different from enhancer hijacking though,
but epigenetic mechanisms could principally also play a role in suppressing
enhancer hijacking. For example, it is easily conceivable that among the eight
described examples of neo-\tad formation some
active enhancer elements were juxtaopsed to potential target genes.
Such enhancers could be hindered from driving ectopic expression through
repressive histone modifications, for example.

In conclusion, this study finds that genomic rearrangements can vastly alter
chromatin architecture, but that this does not necessarily translates to
functional consequences. We do not observe any signs of enhancer hijacking when
\acp{tad} are broken, which could be a consequence of selective pressure and
which demonstrates the extraordinary robustness of biological systems. Subtle
changes in gene expression, to which our approach is sensitive to, were not
found to be enriched around disrupted TADs either, suggesting that enhancer
hijacking-related mechanisms operate on a all-or-nothing basis. Additional
biological mechanism such as epigenetics might play a role in buffering the
effects of rearrangements, yet analyzing this was beyond the scope of this
study. Future research will be necessary to fully understand the impact of
chromatin architecture on gene regulation.

% Discussion (leave out in interest of time)
% ==========================================
% Potential caveats of this study
% -------------------------------
%  - Read mapping bias / ASE analysis
%    We think we have long-enough reads (Degner2009, Stevenson2013)
%  - 1:3 ratio vs. single balancer flies
%  - ASE analysis:
%    Average across gene maybe cancel out such effects
%  - which variants cause effects? (e.g. nonsense-mediated decay??)
%  - TAD calling difficult
