\chapter{General Introduction}
\label{sec:intro}

n his famous ``On the origin of species [...]'' from 1859 Charles Darwin noted
the outstanding variety of individuals within a species. Especially domesticated
speices, as he describes, show remarkable differences---imagine a bloodhound,
terrier, spaniel, and bull-dog next to one another. But also in nature such
differences occur. They can be passed on to offspring over generation and
accumulate to such an extent that a separate species is formed. These
observations as well as a great amount of preceding research by other led
Darwin to the formulation of his theory of natural selection.

More than 150 years later, we have understood many of the molecular mechanisms
behind this principle. We know that genetic information is encoded in DNA and
that it is subject to mutations, which are partly inheritable. These changes in
the DNA create the variability of the genetic material that is so essential to
evolution: the evolutionary principle thus consists of both these parts,
mutation and selection. Today we are able to study the genetic differences
between human individuals (but also within other species) and can investigate
their consequences. Many fundamental associations between the presence of a
genetic variants and certain traits have been found since, for example why
red-green color blindness affects more males than females \citep{Nathans1986}.
Other traits, such as the expected height of a person, are less easily conceived
as they appear to be affected by the combination of many genetic variants
\citep{Wood2014,Marouli2017}. Furthermore, we today understand the causal role
of genetic variation in many diseases, including Mendelian disorders and cancer
\citep{Stankiewicz2010}.

Necessarily, in order to study the consequences of genetic variants we require
methods to accurately detect them. Nowadays, this has been largely enabled with
the advance of DNA sequencing technologies. Based on these methods we have
initially charted the average genome of many species, including humans
\citep{Lander2001,Venter2001}, and identified functional units such as genes and
regulatory elements in great detail \citep{Dunham2012}. We then gained further
insight into the variability within the human population through population
genetics studies \citep{Auton2015,Sudmant2015}. At last, the identified variants
could then be analyzed for a potential effect on phenotypes or disease using,
for example, genome-wide association studies \citep{Ott2015,MacArthur2017}.

Variant detection has become a standard procedure in genomics research. However,
not all types of genetic variants have been studied equally comprehensively.
Especially \emph{\acf{sv}}, which are introduced in \cref{sec:sv}, remained
difficult to ascertain using available assays, which is why their functional
role is still not as well explored as for other types of genetic variation. The
limitations of currently standard techniques had long been noticed and
encouraged further method development. Over the last couple of years, new
techniques for DNA sequencing (or based on standard DNA sequencing) have become
available. In this work, I utilize such emerging technologies to characterize
structural variants beyond what was previously possible.

In the rest of this chapter, I explain some general terminology
(\cref{sec:variation}), the biology behind structural variants
(\cref{sec:sv}), previous and emerging DNA sequencing technologies
(\cref{sec:sequencing}), as well as methods for SV detection
(\cref{sec:sv_detection}). I continue to elaborate the current limitations
of structural variant detection in \cref{sec:limitations}. Then, in
\cref{sec:motivation}, I formulate the goals of my research and provide an
overview of the studies covered in this dissertation.






\section{Terminology around genetic variation}
\label{sec:variation}

Alterations in the DNA of an organism, or of a cell, can arise spontaneously
via chemical or biological processes. If not repaired faithfully, they leave
\emph{variants} (or \emph{mutation}) of the nucleotide sequence. Some variants
only affect a single nucleotide in the double-stranded DNA---these variants are
referred to as \emph{\acfp{snv}}. Others, termed \emph{\acf{sv}} or sometimes
just \emph{genomic rearrangements}, affect larger portions of DNA and are the
central topic of the research presented here.

Most metazoan organisms are diploid, which means they contain two non-identical
copies of each chromosome: one of maternal and one of paternal origin. Any new
variant (within a cell or organism) typically arises only on one of the two
homologous chromosomes. Such a variant is said to be \emph{heterozygous}. The
genomic locus harboring this variant exists in two different versions, which we
call \emph{alleles}. Specifically it harbors a \emph{reference allele}, which is
in agreement with the average individual encoded in the reference assembly, and
an \emph{alternative allele} that describes the non-reference variant form. A
site with exactly two alleles seen across a population is termed
\emph{bi-allelic}, but there are also sites that contain multiple different
alleles and are thus \emph{multi-allelic}. Chromosomes can contain many variants
and depending on the detection strategies, it is often unclear which variants
reside on the same homologue. If this is known, we refer to the ensemble of
variants present along a single homologue as a haploid genotype, or short
\emph{haplotype}. Alleles that are present in the germ line, i.e. the cells
carrying inheritable genetic material, can be propagated to offspring. This way,
an individual can end up carrying the same variant of a genomic locus on both
homologues, which makes it a \emph{homozygous} variant. Variants that are seen
more often in a population, specifically in at least 1\% of the homologoues,
are also called \emph{polymorphisms}.

When a variant is present in an individual, but not in their parents, we call it
a \emph{de novo} mutation. The variant could either have occurred within the
zygote during the first few cell divisions, or already beforehand in the
parental germ line. The latter can sometimes be inferred if other offspring of
those parents carries the same variant. Variants that occur not in the germ line
but in cells of the non-inheritable part of an organism are \emph{somatic}
variants. When an affected cell undergoes repeated divisions, a somatic variant
can be present in a relevant fraction of the total cells of an individual---this
is called \emph{somatic mosaicism}. Depending on how early in development the
variant occurred and on natural selection it can be present in all cells of the
same lineage \citep{Youssoufian2002}. Somatic variants are also the underlying
mechanism of cancer \citep{Campbell2007}.


\section{Structural Variation}
\label{sec:sv}
\input{svs}


\section{DNA sequencing technologies}
\label{sec:sequencing}
\input{seq_tech}


\section{Structural variant detection}
\label{sec:sv_detection}
\input{sv_detection}


\section{Research goals and thesis overview}
\label{sec:motivation}

Studies of structural variation are still fundamentally hampered by the
shortcomings of current \sv detection methods. This especially affects studies
of balanced or complex rearrangements, classes that mostly remained cryptic in
previous studies. In this dissertation, I aim at uncovering and further examine
\acp{sv} that had been difficult to ascertain beforehand. In order to do so, I
explore the potential of emerging DNA sequencing technologies---namely the
techniques introduced in \crefrange{sec:long_read_seq}{sec:strandseq}.
My work is structured into three separate research projects, in which I explore
one of the techniques, each. Below, the specific aims of each project are
outlined. A common theme throughout these projects is the design of new
bioinformatics methods for data analysis and visualization. I believe that the
full potential of novel technology can only be unlocked by the simultaneous
development of new computational approaches and I hope to advance the state of
the art of current methodology by building---and making available---software
tools that future research could profit from.

In \cref{sec:complex_invs}, I present my work for the final phase of the
1000 Genomes Project, in which my role was to validate inversion predictions in
the human population. These inversion predictions had initially remained
inconclusive in \pcr-based validation experiments. We then utilized targeted
long-read sequencing on both \pacbio and \ont MinION platforms to examine the
respective loci more closely. My immediate goal in this project was to verify
inversion calls based on long-read information. Moreover, the subsequent goal
was to further characterize these loci and investigate why previous validations
had failed. Afterwards, I was further interested in investigating the validated
inversion cases deeper to understand the biological mechanisms that created
them.

In \cref{sec:mosaicatcher}, my collaborators and I explored the potential of
Strand-seq to discover \acp{sv} in single cells. While this had been previously
done for inversions, we aimed at extending this principle---for the very first
time---to at least six different \sv classes and towards mosaic (or subclonal)
variants.  Here, I present the current state of this ongoing endeavor. The first
goal in this study was to develop a general concept of how different \sv classes
can be revealed based on the signals available in single-cell Strand-seq data.
Then, the next goal was to prove feasibility of this concept by implementing the
approach and applying it to available as well as newly sequenced cell line data
harboring structural variation. A third important aspect was to test the
limitations of our approach in a controlled environment. The last goal was hence
to establish a simulation framework in which the capabilities and boundaries of
our \sv detection method can be determined.

At last, \cref{sec:conclusions} concludes on the achievements resulting from my
work. This includes the specific questions of the three projects as well as a
more general view on how emerging technologies improved \sv characterization in
the three different scenarios. I then review recent developments by others in
the community that occurred within the time span of my research. This gives an
impression of how emerging technologies are about to change studies on
structural variation and places my work within the broader context of recent
advances in the field.

