\chapter{Complex inversions in the human genome}
\label{sec:complex_invs}


In 2014 and 2015 I was invited to collaborate with a large consortium of
scientists on the 1000 Genomes Project. My supervisor Jan Korbel was the
co-leader of the structural variation subgroup and together with my colleagues
\tobias, \adrian, \benjamin, \markus, and \andreas
I was involved in the validation and characterization of inversions.
This chapter covers my work for the 1000 Genomes Project, which not only turned
out to solve an interesting mystery but also resulted in a co-authorship on
\cite{Sudmant2015}. I continue by describing subsequent work, including a side
project on sequence match visualization that came into being from collaboration
with \markus (\cref{sec:maze}), as well as an analysis of inversion breakpoints
(\cref{sec:breakpoints}). These results have never been published in a journal,
but were presented partially in form of a poster at the German Conference for
Bioinformatics 2016 in Berlin. There is supplementary information to this
chapter enclosed in the appendix (\cref{sec:suppl_inversions}).




\section{Motivation}
\subsection{The 1000 Genomes Project}
\label{sec:1000G}

The 1000 Genomes Project was the effort of a large international panel of
scientists to capture the genetic diversity of the human population as
comprehensively as possible. The project can be regarded as the hitherto final
step in a series of projects that unraveled and characterize the human genome:
The Human Genome Project, which, founded in 1990, was the first large-scale
collaborative work in biology, as well as their private competitor Celera
Genomics (from 1998) aimed at completing the first human genome sequence.
These tremendous efforts, published simultaneously in early
2001\citep{Lander2001,Venter2001}, had both used DNA of a few individuals (among
them Celeraâ€™s then-CEO Craig Venter) to determine an average human genome. The
focus was shifted to the variation present in the human population with the
initiation of the International HapMap Project in 2002. By performing
whole-genome \snv genotyping on hundreds of individuals from multiple continents,
including family-offspring trios, the project generated a first haplotype map of
the human population\citep{InternationalHapMapConsortium2005,Frazer2007,Altshuler2010}.
Over the years the HapMap Project was scaled up to more than 1,000 individuals,
to discover rare \acp{snv}  (minor allele frequency < 5\%) and to call copy number
variants. The samples originally collected for the HapMap Project, which are
still available today as immortalized lymphoblastoid cell lines, were also used
by the 1000 Genomoes Project. Starting long before my own participation, with
the pilot phase publication dating back to 2010\citep{Durbin2010}, the project
even extended this set of samples by multiple other
populations\citep{1000GenomesProjectConsortium2012}. However, the fundamental
novelty was to apply low-coverage whole-genome sequencing which allows to assess
a much broader spectrum of variants in a much larger portion of the genome.

In the third and final phase the project had performed mean 7.4x-coverage
sequencing on 2,504 samples and additionally expanded the palette of
computational methods that were applied as well as added new types of variation
to be studied\citep{Auton2015}. The final data set consists of approximately
88 million phased variants, which was estimated to include more than 99\% of
non-rare \acp{snv} in the studied populations. Besides novel insides on, for example,
seemingly dispensable genes, the primary achievement of the project was to make
an unprecedented resource of genetic variation available to the public. This
came along with technical advances in methods and standardizations such as the
nowadays commonly used file format \vcf\footnote{Established for the
1000 Genomes Project, the format is nowadays maintained by the Global Alliance
for Genomics \& Health (\url{https://www.ga4gh.org}}.
The 1000 Genomes data has since been utilized in many research studies, for
example to refine human population history\citep{Veeramah2014}, to
impute\citep{Howie2012} missing \acp{sv} in many genome-wide association studies
(e.g. \cite{Wood2014}), or to discriminate somatic from germline mutation in
cancer studies\citep{Hiltemann2015}. At last, since the start of the 1000
Genomes Project a considerable number of similarly large sequencing studies has
popped up that focus on specific populations\citep{UK10K,Sulem2015,Telenti2016}
or disease\citep{Campbell2017} and which eventually scale up the catalogue of
known variation in the human population.





\subsection{Predicted inversions and the validation problem}

The Structural Variation subgroup of the 1000 Genomes Project focused on the
identification of various classes of genomic rearrangements. With a total of 83
scientists involved and led by eight principle investigators with different
expertise, the group assembled the to date most comprehensive resource of
structural variation in the human population. Copy number variants, inversions,
mobile element insertions and other types of \sv were predicted from
low-coverage whole-genome sequencing data, statistically phased and
validated using a number of different approaches.

Among his many contributions to this project, \tobias had predicted inversions
using the paired-end signature-based \sv detection tool \delly. This set of
inversions, which was strictly filtered according to validation results and
population genetics aspects, contains a total of 786 inversion calls in the
range of 500~bp~-~50~kb (median size 1.70~kb) and ended up being the only
inversion call set used in the project. In comparison to phase one of the 1000
Genomes Project, this is a more than 15-fold increase in the number of
inversions.

Before its final release, this call set (just like any other call set in the
project) had to undergo cycles of validation and filtering. \adrian and
\benjamin were in charge of running \pcr-based validation experiments of these
inversions. In a nutshell they applied a \explain{Four primer strategy}
{\marginfig{four_primer_strategy.pdf} Four separate PCRs
are run using primer pairs FR, fr, Ff and Rr. In samples
not carrying the inversion only FR and fr will yield bands,
in homozygous carriers only Ff and Rr, and in heterozygous
carriers all four reactions will yield bands.}
on inversion loci, which is capable of distinguishing the genotype of inversions
based on the combination of bands resulting from the \acp{pcr}. However,
this requires that the breakpoints of the inversion were accurately predicted
and also lie in regions accessible to targeted \pcr amplification. The latter
was initially thought to be a problem, as inversions are known to often be
flanked by inverted repetitive sequence due to their origination through \nahr
(see \cref{sec:mechanisms}\todo{explain NAHR in this section!}).

When they tested \pcr verification experiments on a subset of 96 predicted
inversion loci they were puzzled: for the vast majority of loci the received
band patterns did not match the expectations, neither clearly validating nor
invalidating the locus. When counted as invalidations, these results would have
given rise to an \fdr of up to 80\%. Despite multiple trials to improve \pcr
conditions and primer locations, the experimental verification of the inversion
call set remained unsuccessful for a long time and posed a scientific mystery.






\section{Long-read sequencing of inversion loci unravels unexpected levels of complexity}
\label{sec:complex_invs_validation}

By the time I became involved in the inversion validation project my colleagues
had already decided for a new strategy. They selected a subset of small
inversions in a range of up to 10~kb including the surrounding genomic regions
via amplification through long-range \pcr. The resulting amplicons were then
submitted to long-read sequencing in order to uncover the true genomic structure
underlying the predicted inversions. At the same time collaborators within the
1000 Genomes Project had performed independent inversion validation experiments
that will not be addressed in detail here. Just to mention them briefly: 35 loci
were inserted into fosmid libraries, clonally expanded and then submitted to
\pacbio sequencing; another set of loci was genotyped in the cell line CHM1,
which had recently been sequenced with \pacbio \citep{Chaisson2014} and then
validated using the publicly available long read data; another 29 loci present
in the individual NA12878 were evaluated in additional available whole-genome
\pacbio sequencing data for this individual \citep{Pendleton2015}; at last, small
set of inversion breakpoints was submitted to Sanger sequencing
\citep[supplementary methods]{Sudmant2015}.


% QUOTE FROM Sudmant et al, 2015:
% To further explore inversion sequence complexity, we performed a battery of
% targeted analyses, leveraging PacBio resequencing of fos- mids (targeting 34
% loci), sequencing by Oxford Nanopore Minion (60 loci) and PacBio (206 loci)
% of long-range PCR amplicons, and data for 13 loci from another sample (CHM1)
% sequenced by high- coverage PacBio WGS14. Altogether we verified and further
% characterized 229 inversion sites, 208 using long-read data and 21 by PCR
% (Supplementary Table 15), increasing the number of known validated inversions
% by .2.5-fold.
%
% SUMMARY OF SUPPL. TABLE 15
% sheet A. (sheet B contains 21 additional (!) PCR-validated loci)
%
%           pacbio
% validation   N   Y
%   negative   7  36
%   positive  38 170
%
%           minion
% validation   N   Y
%   negative  37   6
%   positive 154  54
%
% class
% Complex    Inv and Del   Inverted Dup MultiDel with Inv    Simple Inv
%      11             28            113                14            42
%
\begin{table}[ht]
    \begin{adjustbox}{inner,minipage=[b]{\textplusmargin}}
            \begin{minipage}[b]{0.54\textplusmargin}
                \centering
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xrrr}
                    \toprule
                    Method      & \#Total & \#Val. & \#Inval. \\
                    \midrule
                    \pcr        &  96 &  27 &   2 \\
                    PacBio      & 308 & 186 &  37 \\
                    ONT MinION  &  96 &  54 &   6 \\
                    Fosmid      &  34 &  28 &   4 \\
                    Total       & 308 & 229 &  47 \\
                    \bottomrule
                \end{tabu}
                }
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[b]{0.42\textplusmargin}
                \centering
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xr}
                    \toprule
                    Class & Count \\
                    \midrule
                    Simple inversion     &  42 \\
                    Inverted duplication & 110 \\
                    Inv. and del.        &  29 \\
                    Inv. and multi-del.  &  14 \\
                    Highly complex       &  11 \\
                    \bottomrule
                \end{tabu}
                }
            \end{minipage}\\[-\baselineskip]%
            \begin{minipage}[t]{0.54\textplusmargin}
                \centering
                \tabcap{inv_validations}{Inversion validations}{Number of
                predicted inversions analysed (\#Total), validated (\#Val.) and
                invalidated (\#Inval.) using one of several techniques performed
                at EMBL or by collaborators.}
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[t]{0.42\textplusmargin}
                \centering
                \tabcap{inv_classes}{Inversion classes}{Summary of classes of
                    inversions found at 206 long read-resolved loci
                    (\emph{Inv. and del.}:
                    Inversion flanked by deletion, \emph{Inv. and multi-del.}:
                    Inversion flanked by multiple deletions)}
            \end{minipage}
        \end{adjustbox}
\end{table}
\todo{fix left-shifting caused (most likely) by adjustbox}


The validation experiments carried out at EMBL were three-fold: First, a small
subset of inversion calls, mostly loci with two-sided support from \delly calls,
could be validated using \pcr and aforementioned four primer strategy. For the
majority of predictions however, \pcr validations had remained inconclusive so
that long-read sequencing was applied. So secondly, amplicons were submitted to
\pacbio single-molecule sequencing\footnote{\label{footnote:xbpIhd}Details on
library preparation and sequencing can be found in the supplementary methods of
\cite{Sudmant2015}} at Baylor College of Medicine and at MPI for Plant Breeding
Research, Cologne\footnote{Data is publicly available through the European
Nucleotide Archive via accession numbers SAMEA3299513, SAMEA3257441, and
SAMEA3257407}. Thirdly, a smaller number of loci were sequenced on a
\minion device\footnoteref{footnote:xbpIhd} at EMBL. These numbers are
summarized in \cref{tab:inv_validations}.

I received both the raw sequencing data of the amplicons in multiple stages. At
first I explored the differences between raw \pacbio reads and the circular
consensus sequences provided by \pacbioâ€™s internal software. In virtually all
cases the circular consensus sequence showed the same information as its raw
subreads, yet with less sequencing errors, which is why I preferred the
consensus sequences from that point on. After mapping to the human reference
genome using \blasr, I selected reads mapping their respective locus for
visualization. I then generated dotplots of each long read against its reference
locus using a custom script, which was developed by \markus and me and which is
further described in \cref{sec:maze}. I used a very similar approach to analyze
\minion data: I obtained both 1D reads as well as 2D reads from multiple
different sequencing runs. I then utilized \last to map reads to an artificial
reference only containing the loci of interest. Again I selected mapping reads
for visualization through dotplots.

Finally, I scanned the approximately 30,000 resulting plots together with
\adrian for signs of inversions at the predicted loci. We categorized loci based
on our visual inspection into the three classes: ``validated'', ``invalidated'',
and ``inconclusive''.  We required at least five independent reads supporting
the alternative allele to count an inversion-type variant as validated. On the
other hand, we deemed a locus as invalidated only if there were at least 30
reads supporting the reference allele and not a single one with an inversion
signal. This asymmetric measure is motivated by an allelic imbalance yielding
from the \pcr, as I briefly describe in \cref{sec:complex_invs_artifacts}. The
majority of validations were successful through support from \pacbio reads,
which outperformed \minion data in both quality and quantity in our
setup.\footnote{Note that I do not carry out a detailed comparison of both
technologies. Instead, we primarily used \pacbio and additionally explored a
very early version of \minion, which turned out to be helpful for a subset of
loci}. In 20 cases \minion was instrumental in validating loci that could not be
resolved by \pacbio due to a lack of coverage. Reassuringly, among the set of 37
loci that were both informative in \pacbio and \minion data, 35 loci agreed in
showing an inversion-type variant and the other two agreed in showing the
absence thereof. The final \fdr for inversions in the 1000 Genomes project was
estimated to be 17\%, although this estimate might be conservative because of
potential allelic dropouts (as explained in \cref{sec:complex_invs_artifacts}).

\figuretextwidth{complex_invs.pdf}
    {complex_invs}
    {Complex inversion types revealed by PacBio and ONT MinION reads}
    {Examplatory loci were selected to represent the most commonly seen
    classes of complex inverted variants (listed in \cref{tab:inv_classes}).
    Green lines denote inversions predicted by \delly. PacBio circular
    consesus sequences and raw ONT MinION reads (vertical) are compared to their
    reference locus (horizontal) via dotplots using maximal unique matches
    of lengths at least 10 (PacBio) or 8 (MinION). Axes denote sequence lengths
    in base pairs. Original read names are included for later reference. See
    \cref{sec:dotplot} for an explanation of the concept of dotplots.}

Most interestingly, we observed that only a minority of cases turned out to be
simple inversions in the sense that a contiguous stretch of DNA is re-oriented
within its original locus. Instead, most positive loci harbored additional
complexity, i.e. other types of variation mixed with the inversion. We
identified five major classes among the inversion-positive loci, which are
presented in \cref{fig:complex_invs}: simple inversions, inverted duplications,
inversion flanked by deletions, inversion flanked by multiple deletions, and
loci with even higher complexity. In summary, a mere 20\% of inversions
validated by long reads are simple; the majority, on the other hand, are
inverted duplications (see \cref{tab:inv_classes}). Precisely, this is the case
for the set of inversions with a single breakpoint predicted by \delly. There is
a smaller subset of 21 validated inversion predictions with double-sided support
from read pairs that are all simple inversions. The estimated \fdr for
inversions with double-sided support is 9\%.

After the completion of the 1000 Genomes Project, \tobias adapted \delly to be
able to distinguish double-sided (hence likely simple) inversion calls from
potentially complex ones. To do so, overlapping paired-end signatures are
combined and tested for the possibility of the locus showing an inverted
duplication or an inversion flanked by a deletion. Nevertheless, this approach
would most likely not have been successful in the scope of the 1000 Genomes
Project due to the low average coverage of 7.5x, since without sufficient
coverage it becomes less likely to find enough breakpoint-spanning read pairs
for both ends of the complex rearrangement.

In the aftermath, when the high level of complexity is seen, it is clear why
\pcr validations were bound to fail. In fact, for simple inversions that were
predicted as such (due to double-sided support from read pairs) \pcr
verification experiments worked about as good as expected. Nevertheless, complex
loci were only visible to long read sequencing. Now that we know how
these types of variants look like, paired-end mapping-based tools can be trained
to capture these events and \delly is one of the first tools capable of identifying
such classes of complex \acp{sv}. At last, the abundance of complexity was higher
than anticipated; yet the total number of identified complex inversions is still
small compared to other, e.g. copy-number variants, and their functional role
remains to be shown.





\subsection{Artifacts in amplicon sequencing}
\label{sec:complex_invs_artifacts}


\figuretextwidth{complex_invs_artifacts.pdf}
    {complex_invs_artifacts}
    {Loci showing three alleles, likely resulting from PCR artifacts}
    {Example locus showing more than two alleles in a single diploid sample,
    plotted the same way as in \cref{fig:complex_invs}.}



\section{Assembly of PacBio reads yields sequence information at basepair resolution}
\label{sec:breakpoints}




\figuretwocolumns{inv_assemblies_qualities.pdf}
    {inv_assemblies_qualities}
    {Quality of assembled loci}
    {Overview of length and quality of the 210 loci (listed in
    \cref{tab:inversionlocilist}) that were assessed using
    any of three technologies. Quality is approximated by number of mismatches
    to the reference genome, taking into account known variantion of the sample.
    Note that unlike Illumina-assembled reads, PacBio assemblies are restricted
    to the size of the amplicons.}
        {inv_assemblies_sizes.pdf}
    {inv_assemblies_sizes}
    {Sizes of rearranged sequences}
    {Sizes were determined from assembled loci using \acs{last}-split as
    described in section XXX. Note that the overall size of the amplicons is
    limited, which potentially explains shorter sizes for the rearranged part in
    loci with higher complexity.}

\figuretwocolumns{inv_assemblies_del_sizes.pdf}
    {inv_assemblies_del_sizes}
    {Size of flanking deletions}
    {Size of deletions (detected via \acs{last}-split, see section XXX) next to
    inverted sequences in all assembled loci that were classified as
    ``simple inversion'', ``inversion flanked by deletion'', or ``inversion
    flanked by multiple deletions''.}
        {inv_assemblies_microhomology.pdf}
    {inv_assembles_microhomology}
    {Breakpoint characteristics of complex loci vs. deletions}
    {Micro-deletions or -duplications (a.k.a. micro-homology) as identified by
    \last-split are reported for assembled loci with exactly two breakpoints
    (199/210) as well as for a set of 12,206 single-breakpoint deletions (REF needed).
    TODO: information on tests performed}



\section{Maze: A tool for match visualization and breakpoint inpsection}
\label{sec:maze}


\subsection{Interactive match visualization in the web browser}
\label{sec:dotplot}


\figuretextwidth{maze_screenshot.pdf}{maze}
    {Screenshot from \maze}
    {On the left, \maze displays a dotplot of two sequences (here inversion
    assembly, vertically, vs. its reference locus, horizontally) and highlights
    sequence alignments (red). On the right, the sequence alignment around the
    breakpoints is highlighted and classified into cases of micro-insertions,
    blunt ends or micro-homology.}



...\acs{mummer}'s dotplot function\footnote{More precisely, by the \texttt{mummerplot} command.
Examples can be found online at \url{http://mummer.sourceforge.net/examples/}.}.



\subsection{Breakpoint identification and characterization}



\section{Conclusions}









Papers:
\begin{itemize}
\item my co-authorship: \citep{Sudmant2015}
\item HGP and Celera: \citep{Lander2001,Venter2001}
\item HapMap: \citep{InternationalHapMapConsortium2005,Frazer2007,Altshuler2010}
\item 1000G: \citep{Durbin2010,1000GenomesProjectConsortium2012,Auton2015}
\item PacBio papers: \citep{Chaisson2014,Pendleton2015}
\item Papers finding complex inversions: Talkowski paper (2017)
    \url{https://www.ncbi.nlm.nih.gov/pubmed/28260531}, where they use jumping
    libraries to find even more complex classes. Ashley's Strand-seq inversion
    paper which at least refers to complex inversions
    \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5088599/}. Another study
    (\url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4490614/}) found exactly
    one complex inversion, shown in figure 7
\item On the

\item Tools: \blasr, \last, \mummer, \bwamem, \quiver, \celeraassembler, \maze
\item Dotplots: \citep{Fitch1969,Gibbs1970}
\end{itemize}
