\chapter{Complex inversions in the human genome}
\label{sec:complex_invs}

See suppl. info in \cref{sec:suppl_inversions}.

Papers:
\begin{itemize}
\item my co-authorship: \citep{Sudmant2015}
\item HGP and Celera: \citep{Lander2001,Venter2001}
\item HapMap: \citep{InternationalHapMapConsortium2005,Frazer2007,Altshuler2010}
\item 1000G: \citep{Durbin2010,1000GenomesProjectConsortium2012,Auton2015}
\item PacBio papers: \citep{Chaisson2014,Pendleton2015}
\item Papers finding complex inversions: Talkowski paper (2017)
    \url{https://www.ncbi.nlm.nih.gov/pubmed/28260531}, where they use jumping
    libraries to find even more complex classes. Ashley's Strand-seq inversion
    paper which at least refers to complex inversions
    \url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5088599/}. Another study
    (\url{https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4490614/}) found exactly
    one complex inversion, shown in figure 7
\item On the

\item Tools: \blasr, \last, \mummer, \bwamem, \quiver, \celeraassembler, \maze
\item Dotplots: \citep{Fitch1969,Gibbs1970}
\end{itemize}



\section{Motivation}
\subsection{The 1000 Genomes Project}
\subsection{Predicted inversions and the validation problem}

In the motivation to this chapter I mention Adrian's \explain{Four-primer
strategy}{\marginfig{four_primer_strategy.pdf} Four separate PCRs are run using
primer pairs FR, fr, Ff and Rr.
In samples not carrying the inversion only FR and fr will yield bands, in
homozygous carriers only Ff and Rr, and in heterozygous carriers all four
reactions will yield bands.} and I explain it quickly in the margin.


\section{Long-read sequencing of inversion loci unravels unexpected levels of complexity}


% QUOTE FROM Sudmant et al, 2015:
% To further explore inversion sequence complexity, we performed a battery of
% targeted analyses, leveraging PacBio resequencing of fos- mids (targeting 34
% loci), sequencing by Oxford Nanopore Minion (60 loci) and PacBio (206 loci)
% of long-range PCR amplicons, and data for 13 loci from another sample (CHM1)
% sequenced by high- coverage PacBio WGS14. Altogether we verified and further
% characterized 229 inversion sites, 208 using long-read data and 21 by PCR
% (Supplementary Table 15), increasing the number of known validated inversions
% by .2.5-fold.
%
% SUMMARY OF SUPPL. TABLE 15
% sheet A. (sheet B contains 21 additional (!) PCR-validated loci)
%
%           pacbio
% validation   N   Y
%   negative   7  36
%   positive  38 170
%
%           minion
% validation   N   Y
%   negative  37   6
%   positive 154  54
%
% class
% Complex    Inv and Del   Inverted Dup MultiDel with Inv    Simple Inv
%      11             28            113                14            42


\begin{table}[ht]
    \begin{adjustbox}{inner,minipage=[b]{\textplusmargin}}
            \begin{minipage}[b]{0.54\textplusmargin}
                \centering
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xrrr}
                    \toprule
                    Method      & \#Total & \#Val. & \#Inval. \\
                    \midrule
                    \pcr        & 96  & 21  &   - \\
                    PacBio      & ??? & ??? & ??? \\
                    ONT MinION  & 60  &  35 &   2 \\
                    \bottomrule
                \end{tabu}
                }
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[b]{0.42\textplusmargin}
                \centering
                %\begin{adjustbox}{tabular=lr,valign=b}
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xr}
                    \toprule
                    Class & Count \\
                    \midrule
                    Simple inversion     &  42 \\
                    Inverted duplication & 110 \\
                    Inv. and del.        &  29 \\
                    Inv. and multi-del.  &  14 \\
                    Highly complex       &  11 \\
                    \bottomrule
                \end{tabu}
                }
                %\end{adjustbox}
            \end{minipage}\\[-\baselineskip]%
            \begin{minipage}[t]{0.54\textplusmargin}
                \centering
                \tabcap{inv_validations}{Inversion validations}{Number of
                predicted inversions analysed (\#Total), validated (\#Val.) and
                invalidated (\#Inval.) using one of several techniques.
                Not included are orthorgonal validation methods performed by
                collaborators.}
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[t]{0.42\textplusmargin}
                \centering
                \tabcap{inv_classes}{Inversion classes}{Summary of classes of
                    inversions found at the 206 resolved loci, which are also
                    illustrated in \cref{fig:complex_invs} (\emph{Inv. and del.}:
                    Inversion flanked by deletion, \emph{Inv. and multi-del.}:
                    Inversion flanked by multiple deletions)}
            \end{minipage}
        \end{adjustbox}
\end{table}
\todo{fix left-shifting caused (most likely) by adjustbox}





\figuretextwidth{complex_invs.pdf}
    {complex_invs}
    {Complex inversion types revealed by PacBio and ONT MinION reads}
    {Examplatory loci were selected to represent the most commonly seen
    classes of complex inverted variants (listed in \cref{tab:inv_classes}).
    Green lines denote inversions predicted by \delly. PacBio circular
    consesus sequences and raw ONT MinION reads (vertical) are compared to their
    reference locus (horizontal) via dotplots using maximal unique matches
    of lengths at least 10 (PacBio) or 8 (MinION). Axes denote sequence lengths
    in base pairs. Original read names are included for later reference. See
    \cref{sec:dotplot} for an explanation of the concept of dotplots.}


\subsection{Artifacts in amplicon sequencing}
\label{sec:complex_invs_artifacts}


\figuretextwidth{complex_invs_artifacts.pdf}
    {complex_invs_artifacts}
    {Loci showing three alleles, likely resulting from PCR artifacts}
    {Example locus showing more than two alleles in a single diploid sample,
    plotted the same way as in \cref{fig:complex_invs}.}



\section{Inversion breakpoint analysis}
\subsection{Assembly of PacBio reads yields sequence information at basepair resolution}



\figuretwocolumns{inv_assemblies_qualities.pdf}
    {inv_assemblies_qualities}
    {Quality of assembled loci}
    {Overview of length and quality of the 210 loci (listed in
    \cref{tab:inversionlocilist}) that were assessed using
    any of three technologies. Quality is approximated by number of mismatches
    to the reference genome, taking into account known variantion of the sample.
    Note that unlike Illumina-assembled reads, PacBio assemblies are restricted
    to the size of the amplicons.}
        {inv_assemblies_sizes.pdf}
    {inv_assemblies_sizes}
    {Sizes of rearranged sequences}
    {Sizes were determined from assembled loci using \acs{last}-split as
    described in section XXX. Note that the overall size of the amplicons is
    limited, which potentially explains shorter sizes for the rearranged part in
    loci with higher complexity.}

\figuretwocolumns{inv_assemblies_del_sizes.pdf}
    {inv_assemblies_del_sizes}
    {Size of flanking deletions}
    {Size of deletions (detected via \acs{last}-split, see section XXX) next to
    inverted sequences in all assembled loci that were classified as
    ``simple inversion'', ``inversion flanked by deletion'', or ``inversion
    flanked by multiple deletions''.}
        {inv_assemblies_microhomology.pdf}
    {inv_assembles_microhomology}
    {Breakpoint characteristics of complex loci vs. deletions}
    {Micro-deletions or -duplications (a.k.a. micro-homology) as identified by
    \last-split are reported for assembled loci with exactly two breakpoints
    (199/210) as well as for a set of 12,206 single-breakpoint deletions (REF needed).
    TODO: information on tests performed}

\section{Maze: A tool for match visualization and breakpoint inpsection}
\subsection{Interactive match visualization in the web browser}
\label{sec:dotplot}


\figuretextwidth{maze_screenshot.pdf}{maze}
    {Screenshot from \maze}
    {On the left, \maze displays a dotplot of two sequences (here inversion
    assembly, vertically, vs. its reference locus, horizontally) and highlights
    sequence alignments (red). On the right, the sequence alignment around the
    breakpoints is highlighted and classified into cases of micro-insertions,
    blunt ends or micro-homology.}



...\acs{mummer}'s dotplot function\footnote{More precisely, by the \texttt{mummerplot} command.
Examples can be found online at \url{http://mummer.sourceforge.net/examples/}.}.



\subsection{Breakpoint identification and characterization}



\section{Conclusions}
