\chapter{Complex inversions in the human genome}
\label{sec:complex_invs}


In 2014 and 2015 I was invited to collaborate with a large consortium of
scientists on the 1000 Genomes Project. My supervisor Jan Korbel was the
co-leader of the structural variation subgroup and together with my colleagues
\tobias, \adrian, \benjamin, \markus, and \andreas
I was involved in the validation and characterization of inversions.
This chapter covers my work for the 1000 Genomes Project, which not only turned
out to solve an interesting mystery but also resulted in a co-authorship on
\cite{Sudmant2015}. I continue by describing subsequent work, including a side
project on sequence match visualization that came into being from collaboration
with \markus (\cref{sec:maze}), as well as an analysis of inversion breakpoints
(\cref{sec:breakpoints}). These results were presented partially in form of a
poster at the German Conference for
Bioinformatics 2016 in Berlin. There is supplementary information to this
chapter enclosed in the appendix (\cref{sec:suppl_inversions}).





\section{Motivation}
\subsection{The 1000 Genomes Project}
\label{sec:1000G}

The 1000 Genomes Project was the effort of a large international panel of
scientists to capture the genetic diversity of the human population as
comprehensively as possible. The project can be regarded as the hitherto final
step in a series of projects that unraveled and characterize the human genome:
The Human Genome Project, which, founded in 1990, was the first large-scale
collaborative work in biology, as well as their private competitor Celera
Genomics (from 1998) aimed at completing the first human genome sequence.
These tremendous efforts, published simultaneously in early
2001\citep{Lander2001,Venter2001}, had both used DNA of a few individuals (among
them Celera’s then-CEO Craig Venter) to determine an average human genome. The
focus was shifted to the variation present in the human population with the
initiation of the International HapMap Project in 2002. By performing
whole-genome \snv genotyping on hundreds of individuals from multiple continents,
including family-offspring trios, the project generated a first haplotype map of
the human population\citep{InternationalHapMapConsortium2005,Frazer2007,Altshuler2010}.
Over the years the HapMap Project was scaled up to more than 1,000 individuals,
to discover rare \acp{snv}  (minor allele frequency < 5\%) and to call copy number
variants. The samples originally collected for the HapMap Project, which are
still available today as immortalized lymphoblastoid cell lines, were also used
by the 1000 Genomoes Project. Starting long before my own participation, with
the pilot phase publication dating back to 2010\citep{Durbin2010}, the project
even extended this set of samples by multiple other
populations\citep{1000GenomesProjectConsortium2012}. However, the fundamental
novelty was to apply low-coverage whole-genome sequencing which allows to assess
a much broader spectrum of variants in a much larger portion of the genome.

In the third and final phase the project had performed mean 7.4x-coverage
sequencing on 2,504 samples and additionally expanded the palette of
computational methods that were applied as well as added new types of variation
to be studied\citep{Auton2015}. The final data set consists of approximately
88 million phased variants, which was estimated to include more than 99\% of
non-rare \acp{snv} in the studied populations. Besides novel insides on, for example,
seemingly dispensable genes, the primary achievement of the project was to make
an unprecedented resource of genetic variation available to the public. This
came along with technical advances in methods and standardizations such as the
nowadays commonly used file format \vcf\footnote{Established for the
1000 Genomes Project, the format is nowadays maintained by the Global Alliance
for Genomics \& Health (\url{https://www.ga4gh.org}}.
The 1000 Genomes data has since been utilized in many research studies, for
example to refine human population history\citep{Veeramah2014}, to
impute\citep{Howie2012} missing \acp{sv} in many genome-wide association studies
(e.g. \cite{Wood2014}), or to discriminate somatic from germline mutation in
cancer studies\citep{Hiltemann2015}. At last, since the start of the 1000
Genomes Project a considerable number of similarly large sequencing studies has
popped up that focus on specific populations\citep{UK10K,Sulem2015,Telenti2016}
or disease\citep{Campbell2017} and which eventually scale up the catalogue of
known variation in the human population.




\subsection{Predicted inversions and the validation problem}

The Structural Variation subgroup of the 1000 Genomes Project focused on the
identification of various classes of genomic rearrangements. With a total of 83
scientists involved and led by eight principle investigators with different
expertise, the group assembled the to date most comprehensive resource of
structural variation in the human population. Copy number variants, inversions,
mobile element insertions and other types of \sv were predicted from
low-coverage whole-genome sequencing data, statistically phased and
validated using a number of different approaches.

Among his many contributions to this project, \tobias had predicted inversions
using the paired-end signature-based \sv detection tool \delly. This set of
inversions, which was strictly filtered according to validation results and
population genetics aspects, contains a total of 786 inversion calls in the
range of 500~bp~-~50~kb (median size 1.70~kb) and ended up being the only
inversion call set used in the project. In comparison to phase one of the 1000
Genomes Project, this is a more than 15-fold increase in the number of
inversions.

Before its final release, this call set (just like any other call set in the
project) had to undergo cycles of validation and filtering. \adrian and
\benjamin were in charge of running \pcr-based validation experiments of these
inversions. In a nutshell they applied a \explain{Four primer strategy}
{\marginfig{four_primer_strategy.pdf} Four separate PCRs
are run using primer pairs FR, fr, Ff and Rr. In samples
not carrying the inversion only FR and fr will yield bands,
in homozygous carriers only Ff and Rr, and in heterozygous
carriers all four reactions will yield bands.}
on inversion loci, which is capable of distinguishing the genotype of inversions
based on the combination of bands resulting from the \acp{pcr}. However,
this requires that the breakpoints of the inversion were accurately predicted
and also lie in regions accessible to targeted \pcr amplification. The latter
was initially thought to be a problem, as inversions are known to often be
flanked by inverted repetitive sequence due to their origination through \nahr
(see \cref{sec:mechanisms}).

When they tested \pcr verification experiments on a subset of 96 predicted
inversion loci they were puzzled: for the vast majority of loci the received
band patterns did not match the expectations, neither clearly validating nor
invalidating the locus. When counted as invalidations, these results would have
given rise to an \fdr of up to 80\%. Despite multiple trials to improve \pcr
conditions and primer locations, the experimental verification of the inversion
call set remained unsuccessful for a long time and posed a scientific mystery.






\section{Long-read sequencing of inversion loci unravels unexpected levels of complexity}
\label{sec:complex_invs_validation}

By the time I became involved in the inversion validation project my colleagues
had already decided for a new strategy. They selected a subset of small
inversions in a range of up to 10~kb including the surrounding genomic regions
via amplification through long-range \pcr. The resulting amplicons were then
submitted to long-read sequencing in order to uncover the true genomic structure
underlying the predicted inversions. At the same time collaborators within the
1000 Genomes Project had performed independent inversion validation experiments
that will not be addressed in detail here. Just to mention them briefly: 35 loci
were inserted into fosmid libraries, clonally expanded and then submitted to
\pacbio sequencing; another set of loci was genotyped in the cell line CHM1,
which had recently been sequenced with \pacbio \citep{Chaisson2014} and then
validated using the publicly available long read data; another 29 loci present
in the individual NA12878 were evaluated in additional available whole-genome
\pacbio sequencing data for this individual \citep{Pendleton2015}; at last, small
set of inversion breakpoints was submitted to Sanger sequencing
\citep[supplementary methods]{Sudmant2015}.


% QUOTE FROM Sudmant et al, 2015:
% To further explore inversion sequence complexity, we performed a battery of
% targeted analyses, leveraging PacBio resequencing of fos- mids (targeting 34
% loci), sequencing by Oxford Nanopore Minion (60 loci) and PacBio (206 loci)
% of long-range PCR amplicons, and data for 13 loci from another sample (CHM1)
% sequenced by high- coverage PacBio WGS14. Altogether we verified and further
% characterized 229 inversion sites, 208 using long-read data and 21 by PCR
% (Supplementary Table 15), increasing the number of known validated inversions
% by .2.5-fold.
%
% SUMMARY OF SUPPL. TABLE 15
% sheet A. (sheet B contains 21 additional (!) PCR-validated loci)
%
%           pacbio
% validation   N   Y
%   negative   7  36
%   positive  38 170
%
%           minion
% validation   N   Y
%   negative  37   6
%   positive 154  54
%
% class
% Complex    Inv and Del   Inverted Dup MultiDel with Inv    Simple Inv
%      11             28            113                14            42
%
\begin{table}[ht]
    \begin{adjustbox}{inner,minipage=[b]{\textplusmargin}}
            \begin{minipage}[b]{0.54\textplusmargin}
                \centering
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xrrr}
                    \toprule
                    Method      & \#Total & \#Val. & \#Inval. \\
                    \midrule
                    \pcr        &  96 &  27 &   2 \\
                    \pacbio      & 308 & 186 &  37 \\
                    ONT MinION  &  96 &  54 &   6 \\
                    Fosmid      &  34 &  28 &   4 \\
                    Total       & 308 & 229 &  47 \\
                    \bottomrule
                \end{tabu}
                }
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[b]{0.42\textplusmargin}
                \centering
                \adjustbox{valign=b,padding=0ex,margin=0ex}{
                \begin{tabu}{Xr}
                    \toprule
                    Class & Count \\
                    \midrule
                    Simple inversion     &  42 \\
                    Inverted duplication & 110 \\
                    Inv. and del.        &  29 \\
                    Inv. and multi-del.  &  14 \\
                    Highly complex       &  11 \\
                    \bottomrule
                \end{tabu}
                }
            \end{minipage}\\[-\baselineskip]%
            \begin{minipage}[t]{0.54\textplusmargin}
                \centering
                \tabcap{inv_validations}{Inversion validations}{Number of
                predicted inversions analysed (\#Total), validated (\#Val.) and
                invalidated (\#Inval.) using one of several techniques performed
                at EMBL or by collaborators.}
            \end{minipage}
            \hspace{0.04\textplusmargin}
            \begin{minipage}[t]{0.42\textplusmargin}
                \centering
                \tabcap{inv_classes}{Inversion classes}{Summary of classes of
                    inversions found at 206 long read-resolved loci
                    (\emph{Inv. and del.}:
                    Inversion flanked by deletion, \emph{Inv. and multi-del.}:
                    Inversion flanked by multiple deletions)}
            \end{minipage}
        \end{adjustbox}
\end{table}
\todo{fix left-shifting caused (most likely) by adjustbox}


The validation experiments carried out at EMBL were three-fold: First, a small
subset of inversion calls, mostly loci with two-sided support from \delly calls,
could be validated using \pcr and aforementioned four primer strategy. For the
majority of predictions however, \pcr validations had remained inconclusive so
that long-read sequencing was applied. So secondly, amplicons were submitted to
\pacbio single-molecule sequencing\footnote{\label{footnote:xbpIhd}Details on
library preparation and sequencing can be found in the supplementary methods of
\cite{Sudmant2015}} at Baylor College of Medicine and at MPI for Plant Breeding
Research, Cologne\footnote{Data is publicly available through the European
Nucleotide Archive via accession numbers SAMEA3299513, SAMEA3257441, and
SAMEA3257407}. Thirdly, a smaller number of loci were sequenced on a
\minion device\footnoteref{footnote:xbpIhd} at EMBL. These numbers are
summarized in \cref{tab:inv_validations}.

I received both the raw sequencing data of the amplicons in multiple stages. At
first I explored the differences between raw \pacbio reads and the circular
consensus sequences provided by \pacbio’s internal software. In virtually all
cases the circular consensus sequence showed the same information as its raw
subreads, yet with less sequencing errors, which is why I preferred the
consensus sequences from that point on. After mapping to the human reference
genome using \blasr, I selected reads mapping their respective locus for
visualization. I then generated dotplots of each long read against its reference
locus using a custom script, which was developed by \markus and me and which is
further described in \cref{sec:maze}. I used a very similar approach to analyze
\minion data: I obtained both 1D reads as well as 2D reads from multiple
different sequencing runs. I then utilized \last to map reads to an artificial
reference only containing the loci of interest. Again I selected mapping reads
for visualization through dotplots.

Finally, I scanned the approximately 30,000 resulting plots together with
\adrian for signs of inversions at the predicted loci. We categorized loci based
on our visual inspection into the three classes: ``validated'', ``invalidated'',
and ``inconclusive''.  We required at least five independent reads supporting
the alternative allele to count an inversion-type variant as validated. On the
other hand, we deemed a locus as invalidated only if there were at least 30
reads supporting the reference allele and not a single one with an inversion
signal. This asymmetric measure is motivated by an allelic imbalance yielding
from the \pcr, as I briefly describe in \cref{sec:complex_invs_artifacts}. The
majority of validations were successful through support from \pacbio reads,
which outperformed \minion data in both quality and quantity in our
setup.\footnote{Note that I do not carry out a detailed comparison of both
technologies. Instead, we primarily used \pacbio and additionally explored a
very early version of \minion, which turned out to be helpful for a subset of
loci}. In 20 cases \minion was instrumental in validating loci that could not be
resolved by \pacbio due to a lack of coverage. Reassuringly, among the set of 37
loci that were both informative in \pacbio and \minion data, 35 loci agreed in
showing an inversion-type variant and the other two agreed in showing the
absence thereof. The final \fdr for inversions in the 1000 Genomes project was
estimated to be 17\%, although this estimate might be conservative because of
potential allelic dropouts (as explained in \cref{sec:complex_invs_artifacts}).

\figuretextwidth{complex_invs.pdf}
    {complex_invs}
    {Complex inversion types revealed by \pacbio and ONT MinION reads}
    {Examplatory loci were selected to represent the most commonly seen
    classes of complex inverted variants (listed in \cref{tab:inv_classes}).
    Green lines denote inversions predicted by \delly. \pacbio circular
    consesus sequences and raw ONT MinION reads (vertical) are compared to their
    reference locus (horizontal) via dotplots using maximal unique matches
    of lengths at least 10 (\pacbio) or 8 (MinION). Axes denote sequence lengths
    in base pairs. Original read names are included for later reference. See
    \cref{sec:maze_dotplot} for an explanation of the concept of dotplots.}

Most interestingly, we observed that only a minority of cases turned out to be
simple inversions in the sense that a contiguous stretch of DNA is re-oriented
within its original locus. Instead, most positive loci harbored additional
complexity, i.e. other types of variation mixed with the inversion. We
identified five major classes among the inversion-positive loci, which are
presented in \cref{fig:complex_invs}: simple inversions, inverted duplications,
inversion flanked by deletions, inversion flanked by multiple deletions, and
loci with even higher complexity. In summary, a mere 20\% of inversions
validated by long reads are simple; the majority, on the other hand, are
inverted duplications (see \cref{tab:inv_classes}). Precisely, this is the case
for the set of inversions with a single breakpoint predicted by \delly. There is
a smaller subset of 21 validated inversion predictions with double-sided support
from read pairs that are all simple inversions. The estimated \fdr for
inversions with double-sided support is 9\%.

After the completion of the 1000 Genomes Project, \tobias adapted \delly to be
able to distinguish double-sided (hence likely simple) inversion calls from
potentially complex ones. To do so, overlapping paired-end signatures are
combined and tested for the possibility of the locus showing an inverted
duplication or an inversion flanked by a deletion. Nevertheless, this approach
would most likely not have been successful in the scope of the 1000 Genomes
Project due to the low average coverage of 7.5x, since without sufficient
coverage it becomes less likely to find enough breakpoint-spanning read pairs
for both ends of the complex rearrangement.

In the aftermath, when the high level of complexity is seen, it is clear why
\pcr validations were bound to fail. In fact, for simple inversions that were
predicted as such (due to double-sided support from read pairs) \pcr
verification experiments worked about as good as expected. Nevertheless, complex
loci were only visible to long read sequencing. Now that we know how
these types of variants look like, paired-end mapping-based tools can be trained
to capture these events and \delly is one of the first tools capable of identifying
such classes of complex \acp{sv}. At last, the abundance of complexity was higher
than anticipated; yet the total number of identified complex inversions is still
small compared to other, e.g. copy-number variants, and their functional role
remains to be shown.



\FloatBarrier
\subsection{Artifacts in amplicon sequencing}
\label{sec:complex_invs_artifacts}

During the analysis of amplicons I made a number of interesting observations
that can most likely be attributed to errors occurring in \pcr or long-read
sequencing. Since these observations led to confusion (or at least to
discussions) during data analysis I describe them below and propose possible
explanations for them.

A first type of artifacts is linked to \pacbio's circular consensus sequences.
As mentioned in \cref{sec:pacbio}, the quality of a \pacbio read can be
increased by sequencing the same DNA fragment multiple times in a ring-like
structure. \pacbio’s software resolves this repetitive signal by splitting it at
the motif of the adapter sequence used to close the ring and by forming a
consensus of the resulting sub-reads. In a few cases this software reported
reads that consisted of an amplicon locus concatenated to a reverse copy of
itself. A possible explanation of these cases would be that the software misses
an adapter sequence, e.g. due to mutations occurring in early \pcr cycles or due
to insufficient sequencing quality. Yet also the possibility of a \pcr artifact
cannot be excluded. Eventually this signature was readily identified and did not
pose a problem in the analysis.

\figuretextwidth{complex_invs_artifacts.pdf}
    {complex_invs_artifacts}
    {Loci showing three alleles, likely resulting from PCR artifacts}
    {Example locus showing more than two alleles in a single diploid sample,
    plotted the same way as in \cref{fig:complex_invs}.}

A more severe artifact was allelic imbalance among the \pcr amplicons. I
observed up to 100-fold ratios in the number of reads supporting alternative and
reference allele. Intriguingly there was a clear trend towards the shorter
allele being more abundant. For example in an inverted duplication, the
reference allele was usually more abundant than the inversion allele, whereas in
the case of an inversion flanked by a deletion, the alternative allele was seen
more frequently. This is, however, not suprising as it is a known effect of \pcr
to prefer shorter fragments and one of the reasons why many DNA protocols (such
as standard Illumina sequencing for example) contain a size selection step. This
is also why we required at least 30 reads of the reference allele to be seen
before declaring a locus invalidated. Yet it could mean that some of the
invalidated loci are in fact true inversion-type alleles but the alternative
allele dropped out. This was not futher explored so that \fdr estimates for the
amplicon-based inversion validation can be considered conservative.

A last, initially very puzzling observation is also linked to long-range \pcr
and demonstrated in \cref{fig:complex_invs_artifacts}. Up on inspection of
dotplots of long reads I identified at least six loci that seemed to carry three
different alleles. In all cases the same two alternative alleles were present in
addition to a reference allele: an inverted duplication and a deletion. At first
I verified that these observations are also present in \minion data to exclude
a \pacbio sequencing artifact (also \cref{fig:complex_invs_artifacts}).
Next, I began investigating the possibility of the whole genomic locus being
duplicated within the genome, which could explain the presence of more than two
alleles. I did not find such evidence based on read mapping and neither did the
duplication call set created for the 1000 Genomes Project contain any
information supporting this theory. The possibility remains that a rather recent
event duplicated the same locus in the analyzed cell lines, yet a much simpler
explanation is on hand. We have reason to believe that the deletion was caused
during \pcr by the presence of the inverted duplication. Such \pcr artifacts
have been previously described \citep{Ji1994,Hommelsheim2015} and this theory is
further supported by the fact that the deletion breakpoints in all cases roughly
align with the insert position of the inverted copy. Even when this replication
error, caused by the polymerase jumping to an annealing homologous region,
occurred only once during the amplification process the then-shorter allele
could have been favored during subsequent \pcr cycles. We decided to count these
loci as inverted duplications, hence also as validated.






\section{Analysis of inversion breakpoints}
\label{sec:breakpoints}

After completion of phase 3 of the 1000 Genomes Project, which includes the
results described in \cref{sec:complex_invs_validation}, I was further
interested in gaining a deeper understanding of this unexpectedly abundance of
complexity in the inversion call set. These inversions are present in the
germline of at least one of 2,504 individuals, representing an estimated allele
frequency of approximately 0.05\%, yet some of them are present much more
frequently. The origin of such inherited variation, which might have occurred
thousands of generations ago, is not accessible to functional assays. However,
as various different studies revealed in the past (reviewed in
\cite{Onishi-Seebacher2011,Hastings2009} and partly explained in
\cref{sec:mechanisms}), the mechanisms generating \acp{sv} leave stochastic but
specific traces in the genome that can later be revealed and potentially
tracked. In more detail, these traces are found around the breakpoint junctions
of an \sv. As \cite{Hastings2009} explain, \nahr is known to induce inversions
between two inverted homologous regions of at least several hundred base pairs.
Hence inversions that are found between such flanking inverted repeats are
likely to be generated by \nahr. On the other hand, homology-independent
mechanisms such as \nhej do not require annealing stretches of DNA. \acs{nhej}
is associated with very small (typically not more than a few base pairs)
deletions or, more rarely, the insertion of free DNA. Moreover, multiple
mechanisms are further associated to micro-homology, a feature that is
demarcated by short stretches of identical DNA on both ends of the breakpoint.
The sequence around breakpoints of \acp{sv} can thus be informative about the
operating mechanisms.

In order to obtain nucleotide resolution sequence data I developed a
computational pipeline to perform accurate assembly of \pacbio reads of the
amplicon loci described above. This pipeline consists of five computational
steps and a final semi-manual validation stage and is described subsequently.
With high-quality assemblies of the complex inversions at hand I could then
search for characteristic traces of, for example, micro-homology around their
breakpoints. I further expanded the list of analyzed loci by another type of \sv
and by additional \pacbio-independent data sets. In the following section I
describe they way I obtained high-resolution sequences and the findings
resulting from those. In \cref{sec:maze} I further describe the visualization
method that resulted as a side products of this work is.




\subsection{Assembly of \texorpdfstring{\pacbio}{PacBio} reads achieves nucleotide resolution sequence information}
\label{sec:pacbio_assemly}

The five computational steps towards \pacbio assembly cover both steps performed
by external software as well as by own tools specifically designed for this
task. At first, I extracted \pacbio reads mapping a specific locus and depleted
them for reads of the reference allele. This depletion relies on the orientation
of sequence alignments computed by \last. Next, I fed the remaining reads into
\celeraassembler version 8.2 to perform the actual assembly step per locus. The
outcome is a list of \explain{contigs}{are \emph{contigous} stretches of DNA
derived from, in this case, stitching together sequencing reads based on their
overlap. The order and orientation of these contigs to one another can often not
be determined by the assembly software},
of which I hope at least one to represent the true genomic structure of the
inversion-type allele. Then I mapped \pacbio reads back onto these contigs using
\blasr and implemented several quality filtering steps, including trimming of
low-coverage regions and a rudimentary detection of artifacts. Finally I
``polish'' the assembly using \pacbio’s tool \quiver. This last step is the only
point in the pipeline that utilizes the full potential of the quality metrics
reported by the \pacbio sequencers alongside the sequence reads, and turned out
to be crucial in achieving near-perfect quality.


\figuretwocolumns{inv_assemblies_qualities.pdf}
    {inv_assemblies_qualities}
    {Quality of assembled loci}
    {Overview of length and quality of the 210 loci (listed in
    \cref{tab:inversionlocilist}) that were assessed using
    any of three technologies. Quality is approximated by number of mismatches
    to the reference genome, taking into account known variantion of the sample.
    Note that unlike Illumina-assembled reads, \pacbio assemblies are restricted
    to the size of the amplicons.}
        {inv_assemblies_sizes.pdf}
    {inv_assemblies_sizes}
    {Sizes of rearranged sequences}
    {Sizes were determined from assembled loci using \acs{last}-split as
    described in \cref{sec:size_of_assemblies}. Note that the overall size of
    the amplicons is limited, which potentially explains shorter sizes for the
    rearranged part in loci with higher complexity.}

Afterwards I prepared the contigs for visualization-guided manual validation. I
inserted all contigs into a database and used an early version of \maze (see
\cref{sec:maze}) to prepare dotplots. The goal was to decide whether they (a)
indeed show a correct inversion signal and not the reference allele or a
spurious contig, and (b) which class of inversion they represent
(\cref{fig:complex_invs}). This type of quality control is rather difficult to
automatize completely, yet it is an easy task for the human eye. This is why I
designed a simple, web browser-based user interface featuring keyboard shortcuts
to be able to quickly annotate a large number of assemblies. Out of 17,000
initial contigs generated using this approach I managed to identify high quality
assemblies for 153 complex loci, which were then used for breakpoint analysis.

Once I achieved satisfying quality with \pacbio assemblies I complemented the
dataset by other types of data that had become available: Sanger sequencing of
65 breakpoint loci that were generated by \adrian, and Illumina short read
assemblies of multiple loci generated by \tobias. Furthermore I included another
class of \sv, namely proximal duplications, which were predicted by \tobias and
which closely resemble inverted duplications yet without the copy being
inverted.

As a measure of quality I determined the number of mismatches of the assembly to
the reference locus. This comparison was performed within each mapping segment
separately (e.g. within the inverted part and the flanking parts separately) and
known \acp{snv} were not counted. Eventually I gathered high quality,
basepair-resolution sequence information for 210 loci (and 365 sample) of
different classes of variants. The sizes and qualities of the best sequence per
locus are shown in \cref{fig:inv_assemblies_qualities} and are the technology
yielding the best data per locus is explicitely listed in
\cref{tab:inversionlocilist}.




\subsection{Sizes of rearranged sequences suggest origin through a common mechanism}
\label{sec:size_of_assemblies}

I used sequence alignments calculated by the split-mode of \last, as implemented
in \maze, to determine each mapping segment and to resolve breakpoints (see
\cref{sec:maze_breakpoints} for more detail). This way I resolved the sizes of
the rearranged sequence of all different classes of \sv, as shown in
\cref{fig:inv_assemblies_sizes}.

\figuretwocolumns{inv_assemblies_del_sizes.pdf}
    {inv_assemblies_del_sizes}
    {Size of flanking deletions}
    {Size of deletions (detected via \acs{last}-split, see method in
    \cref{sec:maze_breakpoints}) next to inverted sequences in all assembled
    loci that were classified as ``simple inversion'', ``inversion flanked by
    deletion'', or ``inversion flanked by multiple deletions''.}
        {inv_assemblies_microhomology.pdf}
    {inv_assemblies_microhomology}
    {Breakpoint characteristics of complex loci vs. deletions}
    {Micro-deletions or -duplications (a.k.a. micro-homology) as identified by
    \last-split are reported for assembled loci with exactly two breakpoints
    (199/210) as well as for a set of 12,206 single-breakpoint deletions from
    \cite{Sudmant2015}.}

Interestingly, an immediate observation was that inverted duplications and
proximal duplications show the same very tight size distribution of
approximately 200~-~400~bp (\cref{fig:inv_assemblies_sizes}). Furthermore, not
only the size of the
additional copy but also the distance to the insertion point is highly similar,
between 0.5 and 3 kb. While these sizes could be limited due to the amplicon
approach, which is dependent on the capabilities of long-range \pcr, the
predictions made by \delly suggest not. The paired-end calling by \delly does
not penalize the detection of larger versions of the same types of variants but
reported only few, which suggests that these classes of variation indeed follow
such a tight size distribution. Of course germline variants underlie selective
forces; it is thus unclear whether the mechanism of origin exclusively creates
this short size spectrum or whether larger inverted or promximal duplications
are strongly selected against. Nevertheless, the similarities between inverted
and proximal duplication suggest both to be generate by a similar, possibly
replication-based mechanism.

Moreover, I also observed an interesting feature of the deletions flanking some
inversions. The classification into ``simple inversion'', ``inversion flanked by
deletion'', and ``inversion flanked by multiple deletions'' relies on a
rule-of-thumb estimate during visual inspection. As
\cref{fig:inv_assemblies_del_sizes} suggests, deletions flanking inversions
display a continuous size spectrum instead of falling into clear categories
(deletion vs. no deletion). This again hints at a common mechanism generating
these three classes of inversions, which sloppily handles the ends of the
to-be-inverted sequence and hence creates more or less large deletions at one
or both ends.



\subsection{Breakpoints analysis suggests mechanisms other than \texorpdfstring{\acs{nahr}}{NAHR} operating}

In order to further explore the mechanism creating complex inversions I searched
for homologous sequence around the breakpoints that would be a clear sign for
\nahr. Except for a few low-complexity repeats no true homology could be found
around the breakpoints. This is, however, not surprising, as the paired-end
read-based detection strategy by \delly is known to have lower power in
repetitive (hence not uniquely mappable) regions.

I continued the investigation of mechanism by inspecting breakpoints for the
presence of micro-homology. I selected 199/210 loci with exactly two breakpoints
(the remaining ones being ``highly complex'') and analyzed the breakpoints of
the sequenced alleles in terms of their matches to the reference sequence. This
analysis can unravel small insertions of non-templated sequence, exact
breakpoints or the presence of micro-homology (see \cref{fig:maze} for example).
The definition of such base-level features is highly susceptible to alignment
methods and –parameters. To make statements about the breakpoint characteristics
of the set of complex \acp{sv}\footnote{I refer to the set of 199 loci as
``complex SVs'' from now on although technically it contains also simple
(non-complex) inversions} I compared them to another set of \acp{sv} using the
exact same methodology. I employed a data set of short read assemblies of
deletion breakpoints available from the 1000 Genomes Project \citep{Sudmant2015},
parts of which had been analyzed beforehands \citep{Abyzov2015}. I filtered to
this data set down to 12,206 single-breakpoint events without additional
complexity and compared the breakpoint characteristics of complex \acp{sv} and
deletions as shown in \cref{fig:inv_assemblies_microhomology}. Interstingly,
deletions and complex \acp{sv} did not differ strongly in the amount and length
of micro-homology. However, the distributions in general differed significantly
(p-value < 0.001, Kolmogorov–Smirnov test ) with complex \acp{sv} harboring less
exact breakpoints than deletions and slightly longer stretches of non-templated
insertion of DNA (median 1bp for deletions vs. median 4bp for complex \acp{sv},
p-value < 0.001, Wilcoxon-Mann-Whitney test).

In summary I identified multiple features of the set of complex \acp{sv} that
provide subtle hints towards the mechanisms they originate from. Clearly they
arise independent of homology, a fact that is tightly coupled to the detection
method applied. Among the non-homology dependent mechanisms there is no evidence
of a micro-homology-based mechanism such as \fostes/\mmbir being favored.
Nevertheless, there are significant differences compared to the class of
deletions, which acts as a proxy for non-complex SVs, namely in the number and
length of non-templated sequence insertions. Notably, I described multiple
different classes of \acp{sv}, of which, as the data suggests, many appear to be
generated by the same mechanisms: inverted and non-inverted, proximal
duplications follow exactly the same tight size distribution, and the deletions
that sometimes flank inversions span a continuous size range rather than showing
a binary pattern (present or absent). At last, the insertion of a few base pairs
of DNA at the breakpoints of complex \acp{sv} could be traces of \nhej or \bir
mechanisms operating \citep{Hastings2009,Carvalho2016}, yet I deemded the singal
not strong enough to make these distinctions.







\section{\texorpdfstring{\maze}{Maze}: A tool for match visualization and breakpoint inpsection}
\label{sec:maze}

A common theme of this thesis is how the development of computational methods
for analysis and, notably, visualization of biological data are instrumental to
accelerate scientific research. This chapter shall not fall behind in that
respect. Together with \markus I developed a tool to easily visualize the
alignment of two DNA sequences and to inspect the breakpoints of potential
matches between them. Termed \maze, this piece of software resulted from the
necessities I faced during the analyses described in
\crefrange{sec:complex_invs_validation}{sec:breakpoints}. It is available online at
\url{https://github.com/dellytools/maze} or, in a version with reduced
functionality, at \url{https://gear.embl.de/maze/}. In this section I describe
the method, design decisions, and its use cases.




\subsection{Interactive match visualization in the web browser}
\label{sec:maze_dotplot}

\maze is designed to quickly compare two nucleotide sequences against another.
As an easy-to-grasp, visual comparison method we chose dotplots, which have a
long-standing tradition in sequence analysis \citep{Fitch1969,Gibbs1970}. The
principle of dotplots is to graphically highlight regional identity of two
different sequences or of the same sequence to itself in a 2-dimensional matrix.
The top panel of \cref{fig:complex_invs_artifacts} shows such a graphical
comparison of two similar sequences, and \cref{fig:complex_invs} shows examples
of sequences carrying different types of inversion compared to a non-inverted
sequence. In contrast to a sequence alignment, which generally aims at computing
the constellation of two sequences having the closest distance, a dotplot
simultaneously compares all substrings of both sequences to one another.

This idea has recurred many times in the history of bioinformatics with
different goals in mind. For example, some dotplots originally designed for
peptide comparisons color matches based on exact, single-character matches.
Dotplots for DNA comparison, on the other hand, tend to require longer matches
or calculate an average of the identity of multiple characters within a sliding
window. This is to reduce the background noise that naturally occurs when
comparing sequences with a small alphabet, such as DNA\footnote{For instance, a
single-character comparison of two random DNA sequences would fill as much as
one quarter of the matrix}. Furthermore, dotplots have been optimized for
differently scaled inputs. A genome-scale comparison\footnote{See for example
\url{http://last.cbrc.jp/hg18-mm9.png} for a human vs. mouse comparison}
likely requires a different design and different data structures to find matches
than the comparison of many small sequences.

The dotplots generated by \maze display \explain{maximal unique match}{
consecutive stretch of exact matches that is not included in any other such
stretch, hence cannot be expanded to either side. A maximal \emph{unique} match
additionally require the match to occur exactly once in both sequences}es or,
alternatively, just maximal matches). Those can be fast computed by \mummer,
which internally uses an efficient suffix tree data structure. They are then
filtered to a minimum length and displayed in a web browser using Javascript.
This is basically the same as offered by \acs{mummer}'s dotplot function\footnote{
More precisely, by the \emph{mummerplot} command. Examples can be found online
at \url{http://mummer.sourceforge.net/examples/}}. Although maximal unique
matches have many of the aforementioned advantages for DNA comparisons, they are
bad at handling small insertions or deletions as they show only exact matches.
This is why the minimal length offers a trade-off between sensitivity
(shorter matches) and reduction of background noise (longer matches).

\figuretextwidth{maze_screenshot.pdf}{maze}
    {Screenshot from \maze}
    {On the left, \maze displays a dotplot of two sequences (here inversion
    assembly, vertically, vs. its reference locus, horizontally) and highlights
    sequence alignments (red). On the right, the sequence alignment around the
    breakpoints is highlighted and classified into cases of micro-insertions,
    blunt ends or micro-homology.}

The selling point of \maze is its ease of use tailored towards quick inspection
of long-read sequencing data of, for example, amplicons. The sequences can be
entered as standard FASTA or FASTQ files through a user interface in the web
browser. Dotplots are calculated on the fly and displayed as in \cref{fig:maze}
(left panel). \Ac{maze} also computes matches to the reverse complement sequence
and displays those in red. Besides a one-vs.-one mode, \maze can compare many
sequences to a single one, just as I needed in \cref{sec:complex_invs_validation}
to inspect long reads of amplicon sequencing. Moreover, Maze can also compare
multiple pairs of sequences, a scenario that became relevant once optimal
assemblies for diferent loci were available in \cref{sec:breakpoints}. In those
sections an early version of \maze was used and plots were written as PDF files
instead of displayed in the web browser.

The current version of \maze uses the popular d3 library\footnote{Found online
at \url{https://d3js.org/}} to display dotplots. A user can readily adapt the
minimum length and type of matches to be calculated via in-app settings. The
dotplot can be panned and zoomed-in using the computer mouse, saved as PDF, and
by using keyboard shortcuts a user can quickly browse through multiple dotplots.
At last, after I successfully utilized \maze’s dotplots to solve the inversion
validation mystery of \cref{sec:complex_invs_validation}, I extended its
functionality by an analysis module dedicated to breakpoints of sequence
matches, as I describe in the upcoming section.





\subsection{Breakpoint identification and characterization}
\label{sec:maze_breakpoints}

Any analysis of \sv breakpoints depends a reasonable definition of what
breakpoints actually are and how to determine them. I found a convincing
solution in an algorithm called \last-split \citep{Frith2015}: given a sequence
and a reference genome, it finds the optimal alignments that maximally cover the
query sequence without allowing overlaps.  This idea treats the contiguous
sequence (from assembled reads, in this case) as the true allele and tries to
split it into consecutive parts that can be assigned to their most likely
origins in the reference genome. This also means that while the sequence is
split into consecutive stretches, matches on the reference are unconstrained and
can potentially overlap, e.g. in the case of a duplication. This method was
specifically designed with inversions and complex rearrangements in mind
and therefor determines breakpoints accurately in the given scenario.

The breakpoint module of \maze runs \last-split on the two given sequences to
determine breakpoints. First of all, the aligning parts are highlighted inside
the dotplot as rectangles that become visible when a user crosses the respective
entry with the computer mouse (\cref{fig:maze}, on the left). In addition to
that, a separate field on the right displays the exact sequence alignment in
form of text (not shown in \cref{fig:maze}). Given the breakpoints from
\last-split, \maze can compute the characteristics of that breakpoint in respect
to the reference sequence. For example, if the two consecutive parts of the
sequence match to regions of the reference genome that slightly overlap, the
exact position of the breakpoint is ambiguous. This is what is commonly
understood as micro-homology (see \cref{sec:mechanisms} and compare to
figure 1 in \cite{Hastings2009}) and can be evidence for a
micro-homology-dependent mechanism forming the \sv. If the two regions are not
exactly contiguous in the sequence because no according match to the reference
is found, an insertion of other, possibly non-templated DNA occurred. \Ac{maze}
graphically arranges the alignments around the breakpoints, detects such
features and reports them in a separate panel on the right, as can be seen in
\cref{fig:maze} (right panel). The respective matches are highlighted within the
alignment to direct the user’s attention to the right position belonging to the
breakpoint. A current shortcoming is that the extension of this alignment does
not allow gaps. In order to do so, another alignment would have to be computed,
yet of a region which was deemed irrelevant by the \last aligner. Additionally
to the breakpoints on the query sequence, a third panel focuses on the
breakpoints of the reference sequence (not shown in \cref{fig:maze}).
Neighboring reference matches also can overlap, meaning the affected bases are
duplicated in the sequence, or can be distinct with a small gap in between.
These micro-duplications and micro-deletions are similarly displayed up to a
size of 100~bp by \maze.

I used an early version of this breakpoint analysis tool to obtain the results
in \cref{sec:breakpoints}. Together with custom scripts the analysis was
automated to process a large number of loci sequentially, allowing me to
generate \crefrange{fig:inv_assemblies_del_sizes}{fig:inv_assemblies_microhomology}.
In the current available version\footnote{\url{https://github.com/dellytools/maze/commit/106803d350719188181ce39cbb5c832822b724da}}
of \maze this functionality was abandoned for the sake of simplicity. I utilized
expandable fields to support a user in showing or hiding certain information and
adopted the modern look and feel of the main window of Maze that was designed by
\markus.

In summary, \maze is a user-friendly and easy-to-install software specifically
designed for inspecting long reads or assemblies in the range up to 50~kb. It
allows a user to visually browse through their sequence data to (1) judge the
quality (compare, for example, \pacbio to \minion data in \cref{fig:complex_invs}),
(2) distinguish successful sequencing reads from artifacts and (3) discover
interesting alleles such as complex rearrangements. The additional breakpoint
module of \maze comes into play whenever a user is interested in characteristics
of the breakpoint junctions of \acp{sv}. \Ac{maze} uses current web browser-based
technology and a modern design to offer a responsive user experience and was
instrumental in gaining the scientific insights described in
\crefrange{sec:complex_invs_validation}{sec:breakpoints}.








\section{Conclusions}

The third and final phase of the 1000 Genomes Project created an unprecedented
resource of variation in the human population \citep{Auton2015}. One of the
major achievements was the integrated map of structural variation
\citep{Sudmant2015}, for which various classes of \acp{sv} were detected and
genotyped in 2,504 individuals. My colleagues and I were responsible for the
set of inversions, which are notoriously difficult to ascertain as also other
population-scale studies note \citep{Kidd2008,Hehir-Kwa2016}. With 768 loci it
is to date the most comprehensive high-accuracy inversion-type call set of the
human population.

These inversions had been predicted from low-coverage paired-end sequencing data,
but \pcr-based validation experiments proved difficult. For a while it was a
major mystery why inversions that seemed to be accurately predicted could not be
experimentally validated. This mystery was solved with the help of long-read
sequencing, notably \pacbio and \minion sequencing. A subset of 208 predicted
loci (all below 10~kb in size) were amplified together with surrounding genomic
DNA, sequenced and then visualized via dotplots. This way, more than 80\% of
analyzed loci could be verified, exemplifying the potential of third-generation
sequencing.

Strikingly, we found that the vast majority of loci turned out to be complex
instead of simple. This realization explained the initial dilemma of
\pcr-based verification experiments and it revealed a previously unexpected
amount of complexity. Besides simple inversions, we discovered inverted
duplications as the most abundant class of complex \acp{sv} with inversion
signature, but also inversions flanked by deletions on one or both sides and
variants with even higher complexity (shown in \cref{fig:complex_invs}). The
existence and amount of such complex variants has since been confirmed and even
extended to more complex classes by independent studies (notably
\cite{Collins2017}, but also \cite{Sanders2016} and \cite[fig. 7]{English2015}).

A presumably abundant source of inversions is \acl{nahr}, which creates
inversions embedded in inverted repeats \citep{Carvalho2016}. For example, up to
12\% of the human genome were estimated to be susceptible to \nahr-mediated
inversions \citep{Dittwald2013}. However, paired-end mapping strategies are
known to have limited power when \sv breakpoints align with repetitive sequence
because of their inability to correctly map reads in those regions together with
their short insert sizes, which typically cannot span these repetitive elements.
\citep{Korbel2007} note a mere 14\% of SVs, which they detected using paired-end
mapping, to \nahr and the SV calls, notably inversions, of the 1000 Genomes
Project underlie a similar detection bias. In contrast, studies using alternative
technologies reveal higher levels of \nahr-mediated inversions: Kidd et al.
report 67\% of inversions identified to be flanked by inverted repeats using
fosmid libraries (Kidd2008), Chaisson find et al 68\% of 34 inversions detected
via \pacbio assemblies \citep{Chaisson2014}, and also optical mapping reportedly
has a greater ability to detect \acp{sv} embedded in repetitive DNA
\citep[fig. 6]{Teague2010}. Furthermore, studies using breakpoint-independent
approaches report inversions in a much larger size range (i.e. a minimum of 200kb
in \cite{Bansal2007}; a median size of 176kb in \cite{Sanders2016} using
Strand-seq).

It can thus be noted that the inversions predicted by the 1000 Genomes Project
are complementary to previously revealed inversions, both in terms of size and
in the high level of complexity. While generally the functional relevance of
this class of complex SVs remains to be shown, a few variants overlap genes as
for example \textit{Ras} homologue family member H \textit{(RHOH)}
\citep{Sudmant2015}. \Ac{nahr} appears not to be relevant in creating these
\acp{sv}, but the effective mechanisms still remain obsure. A higher number of
micro-insertions, as I found through detailed breakpoint analysis based on
nucleotide-resolution assembly sequences, could hint at \nhej or a
replication-based mechanism (e.g. \bir) operating. \cite{Collins2017}, who did a
more extensive analysis of complex rearrangements, suggest the synchronous repair
of simultaneous double-strand breaks as seen in cases of chromothripsis might
play a role here.

Finally we learned that complex classes of inversions are present in the human
genome and that this complexity should be taken into consideration in future
efforts of \sv discovery. \Ac{delly} is one of the first tool that is capable of
doing so, yet manual inspection of complex \acp{sv} is still highly recommended
(also reported for other tools \citep{Collins2017}). \maze, which resulted as a
side product from this work, is a handy tool to inspect complex \acp{sv}
sequenced with long reads and can be used in the future to further investigate
complex loci.





