\chapter{Complex inversions in the human genome}
\label{sec:inversions}

See suppl. info in \cref{sec:suppl_inversions}.

In the motivation to this chapter I mention Adrian's \explain{Four-primer
strategy}{\marginfig{four_primer_strategy.pdf} Four separate PCRs are run using
primer pairs FR, fr, Ff and Rr.
In samples not carrying the inversion only FR and fr will yield bands, in
homozygous carriers only Ff and Rr, and in heterozygous carriers all four
reactions will yield bands.} and I explain it quickly in the margin.


Tools: \blasr, \last, \mummer, \bwamem, \quiver


% SUMMARY OF SUPPL. TABLE 15
% sheet A. (sheet B contains 21 additional (!) PCR-validated loci)
%
%           pacbio
% validation   N   Y
%   negative   7  36
%   positive  38 170
%
%           minion
% validation   N   Y
%   negative  37   6
%   positive 154  54
%
% class
% Complex    Inv and Del   Inverted Dup MultiDel with Inv    Simple Inv
%      11             28            113                14            42

\figuretextwidth{complex_invs.pdf}
    {complex_invs}
    {Complex inversion types revealed by PacBio and ONT MinION reads}
    {Examplatory loci were selected to represent the most most commonly seen
    classes of complex inverted variants (listed in table XXX).
    Green lines denote inversions predicted by \delly. PacBio circular
    consesus sequences and raw ONT MinION reads are compared to their reference
    locus via dotplots using maximal unique matches (REF) of
    lengths at least 10 (PacBio) or 8 (MinION). Axes denote sequence lengths in
    base pairs. Original read names are included for later reference.}

\figuretextwidth{complex_invs_artifacts.pdf}
    {complex_invs_artifacts}
    {Loci showing three alleles, likely resulting from PCR artifacts}
    {Example locus showing more than two alleles in a single diploid sample,
    plotted the same way as in \cref{fig:complex_invs}.}
