\chapter{SV Detection in Single Cells}
\label{sec:mosaicatcher}

Here, I present the current state of a project with the objective to develop a
computational method for \sv detection in Strand-seq data. This method is being
developed together with \jan and \ashley as well as with our collaborators
\david, \maryam, and \marschall from the Max-Planck Institute for Informatics,
Saarbr√ºcken. \ashley and \jan conceived the general concept of \sv discovery
based on the signals provided by Strand-seq data. \marschall and \david
contributed analyses related to phasing and and improvements to the overall work
flow. \maryam developed the Bayesian classification approach described below.
Also work from \venla on \acl{sce} events was included into this method.
The main implementation and all other analyses, including figures, are my own.
At last, I would like to thank \balca, who provided the cell lines utilized for
demonstration purposes, and \landsdorp's lab, who carried out the Strand-seq
experiments on this cell line.





\section{Structural variants in the context of somatic mosaicism}
\label{sec:mosaic_mosaicism}

\todo{Write this section}
text text text text text text text text text text text text text text text
text text text text text text text text text text text text text text text
text text text text text text text text text text text text text text text





\section{\textsc{Mosaicatcher}: A novel approach for comprehensive SV detection in single-cell Strand-seq data}
\label{sec:mosaic_mc}

Strand-seq generates sequencing reads that are all of the same directionality
when they stem from the same homologue. As explained in \cref{sec:strandseq},
this is achieved by labeling and degrading the non-template strand of
actively replicating cells. When the resulting sequencing reads are mapped to a
reference assembly, they map either to the Watson (W) or Crick (C) strand. The
presence of two homologous chromosomes then leads to the presence WW, CC, or WC
chromosomes, which we call their \emph{inherited strand states}.
Here, I show how these unique characteristics of Strand-seq reveal the presence
of seven different \sv classes and demonstrate our computational method called
\mc that implements this idea.

%\afterpage{%
%    \clearpage%
%    \ifodd\value{page} \expandafter\afterpage \fi {%
    \begin{figure}[t!]                                             % cell BM160815_WT_007p1 from BM160815_WT.200000.pdf
        \captionsetup{type=figure}
        \includegraphics[width=\textplusmargin,inner]{ss_lib_rpewt.pdf}
        \figcap{ss_library}{Example of a single cell Strand-seq library}{
            Here, a single cell Strand-seq library of an \rpe-I wild type cell
            line is displayed using the plot function of \mc.
            Each vertical panel shows binned read counts of one chromosome, with
            the Watson strand on the left, in orange, and the Crick strand on
            the right in blue. In the cell shown here, each 200~kb-bin contains
            a median of 76 total reads, which is depicted by the
            dotted lines. Some regions, e.g. the centromere of chromosome 1 or
            the sub-telomeric regions of chromosomes 13--15 are not coverded by
            reads because of mappability issues; these bins are excluded from
            further analyses. The estimated strand inheritance state of each bin
            is color-coded in the background: blue for CC, orange for WW and yellow
            for WC. Chromosomes 5 and 11 carry \acp{sce}, which are visible by a
            change in the strand inheritance state that continues to the end of
            the chromosomes.}%
    \end{figure}
%    }}

\Cref{fig:ss_library} depicts Strand-seq data from a single cell of an \acf{rpe}
wild type cell line (courtesy by \balca and \landsdorp). This cell-wide overview
plot was generated using \mc and is typically the starting point for
an analysis of Strand-seq data. The procedure leading to this plot is outlined
briefly in \cref{sec:mosaic_method}.

An overview plot allows an initial judgment on the success, quality, and depth
of a Strand-seq library. In the cell shown here, reads aligning to either W or C
strand were binned at 200~kb resolution. The cell was sequenced at ample depth
(median 76 reads per bin), shows the Strand-seq characteristic strand
inheritance patterns (WW, CC, and WC chromosomes) and is of high quality, which
can be judged from the low number of reads on the opposite strand in WW or CC
cells and a fairly even coverage distribution. Experimental parameters
influencing the quality of Strand-seq libraries are discussed in detail by
\cite{Sanders2017}.

\Cref{fig:ss_library} further gives a first impression of genomic rearrangements
in this cell. For example, the region of WW reads at around 120~Mb on chromosome
1 reveals an inversion of that locus in respect to the reference assembly. The
most prominent alterations are visible on chromosomes 5 and 11, though. Here,
the strand inheritance states change at one position and remain consistent to
the end of the chromosome from there. These positions mark \acf{sce} events,
which are reciprocal exchanges between two identical sister chromatids and
which can be specifically measured using Strand-seq \citep{Falconer2012}.
\Acp{sce} occur randomly across the genome and must not be mistaken with other
classes of \acp{sv} for our purpose.





\subsection{Three signals within Strand-seq data are distinctive of SVs}
\label{sec:mosaic_concept}

Strand-seq libraries reveal the presence of large \Acl{sv}. This had been shown
for inversions in the past by \cite{Sanders2016}. Here, we determined seven
different \sv classes that can be revealed in Strand-seq data, five of which are
principally discernible within a single cell (deletion, duplication, inversion,
inverted duplication, \loh), and two that become apparent across a population of
cells (aneuploidy and translocation). These \sv classes can be distinguished
using three independent signals: (1) the normalized read coverage of a locus,
which corresponds to a diploid state (2N) in a non-affected locus.
(2) The strand ratio, i.e. the number of W reads over the number of C reads.
Alternatively, a fraction can be used here, for example the Watson fraction
W/(W+C). (3) Haplotype information. When whole-chromosome haplotypes are known
at sites of \acp{snv}, these alleles can be used to reason about potential
rearrangements. Sequencing reads overlapping such variants can be queried for
both their original homologue \emph{and} their strand direction.
Below I explain how we can utilize the combination of these signals to detect,
genotype and phase \acp{sv}. \Cref{fig:mosaic_examples} contains examples of
several \sv classes that we identified in \rpe cell lines.

\FloatBarrier
\begin{table}[t]%
\newcommand\ccb{\cellcolor{blue!10}}
\newcommand\cco{\cellcolor{orange!10}}
    \begin{adjustbox}{width=\textplusmargin,inner}%
    \begin{tabu} to \textplusmargin {X[l] c c c c l c c c c}
        \toprule
          & \multicolumn4{c}{WC chromosome} & & \multicolumn4{c}{WW chromosome} \\
         \cmidrule{2-5} \cmidrule{7-10}
          & & & \multicolumn2{c}{Haplotype} & & & & \multicolumn2{c}{Haplotype} \\
        \cmidrule(lr){4-5} \cmidrule(lr){9-10}
         & Cov. & W.f. & \emph{W} & \emph{C} & & Cov. & W.f. & \emph W & \emph{C} \\
        \midrule
        Reference allele             &      2N &      50\% & $h_1$   & $h_2$     & &      2N &      100\% & $h_1 + h_2$     & - \\
        Deletion of $h_1$            &      1N &       0\% & -       & $h_2$     & & \ccb 1N & \ccb 100\% & $h_2$           & - \\
        Deletion (homozygous)        &      0N &         - & -       & -         & &      0N &          - & -               & - \\
        Duplication of $h_1$         &      3N &      66\% & $2 h_1$ & $h_2$     & & \ccb 3N & \ccb 100\% & $2 h_1 + h_2$   & - \\
        Duplication (homozygous)     &      4N &      50\% & $2 h_1$ & $2 h_2$   & &      4N &      100\% & $2 h_1 + 2 h_2$ & - \\
        Inversion of $h_1$           &      2N &       0\% & -     & $h_1 + h_2$ & & \ccb 2N & \ccb  50\% & $h_2$ & $h_1$       \\
        Inversion (homozygous)       & \cco 2N & \cco 50\% & $h_2$   & $h_1$     & &      2N &        0\% & -     & $h_1 + h_2$ \\
        Inverted duplication ($h_1$) & \cco 3N & \cco 33\% & $h_1$ & $h_1 + h_2$ & & \ccb 3N & \ccb  66\% & $h_1 + h_2$ & $h_1$ \\
        \bottomrule
    \end{tabu}
    \end{adjustbox}
    \tabcap{mosaic_sv_signals}{Distinct signatures of focal SVs in Strand-seq
    data}{SVs can be identified based on three separate signatures of Strand-seq
    data: the total read coverage (\emph{Cov.}), the strand ratio---here shown
    as Watson fraction (\emph{W.f.}), and the presenece of haplotype-tagging
    \acp{snv} on each strand (\emph{Haplotpye}). In this table I show how the
    various focal \sv types can be inferred from these signals. This is different
    for WC chromosomes than for WW or CC chromosomes. For the sake of simpliciy,
    the table only shows the WW case and assumes heterozygous variants to affect
    haplotype $h_1$. Entries in orange are \sv classes that cannot be
    distinguished unambigously using only coverage and Watson fraction.
    Specifically, a homozygous inversions remain hidden in WC cells and an
    inverted duplication of $h_1$ cannot be distinguished from a duplication of
    $h_2$. Entries marked in blue cannot be phased based on coverage and Watson
    fraction alone. However, all these cases can be disentangled when
    haplotype-resolved \acp{snv} are available or by integrating information
    across several cells that share an \sv.}
\end{table}

\paragraph{Deletions and duplications}
\Acp{cnv} alter the total read coverage of an affected locus. A deletion
decreases the coverage from 2N to 1N, i.e. by a factor of two, and is hence
typically easier to detect---the same has been observed for other read
depth-based \sv callers. Duplications, which increase copy number to 3N, alter
the read depth only by a factor of 1.5 compared to the reference state.
Homozygous duplications increase the copy number even to 4N. Homozygous
deletions are marked by a complete absence of reads. They thus provide the
strongest change in read depth, with the only caveat that they resemble regions
of low mappability such as the centromere on chromosome 1. Homozygous deletions
are hence best studied in the presence of a control sample that does not carry
the deletion.

In contrast to classic read depth analysis, Strand-seq additionally provides
strand information. For example, a heterozygous deletion in a WC chromosome will
only lack reads on one of the strands. Similarly, a heterozygous duplication
will increase coverage only on one strand. In a WW or CC cell, strand
information does not add supportive evidence, but when phased \acp{snv} are
available, only one homologue will be deleted/duplicated.
\Cref{tab:mosaic_sv_signals} summarizes in detail how these signals
allow to differentiate the focal \sv classes that I cover here.






\subsection{Automatd SV detecting}
\label{sec:mosaic_method}



\subsection{Revealing potential SV breakpoints via multivariate total variation denoising}
\label{sec:mosaic_segmentation}


\FloatBarrier
\section{Simulation of Strand-seq data to explore the limits of \textsc{Mosaicatcher}}
\label{sec:mosaic_simul}

\subsection{Development of a versatile simulation framework}

\subsection{Assessing the performance of the segmentation algorithm}

\section{Conclusions and outlook}
\label{sec:mosaic_conclusion}


