DNA sequencing refers to the process of deciphering the order of the four bases
(adenine, cytosine, guanine, and thymine, abbreviated by A, C, G, and T) that
constitute a DNA molecule. In the 1990s, sequencing the human DNA was a
decade-long, multi-million dollar effort but it led to the successful production
of a reference genome of humans and many other organisms
\citep{Lander2001,Venter2001}. Today, thanks to technological improvements,
DNA sequencing has become a standard technique applied on a daily basis in
genomics research. The advance of sequencing technologies has truly
revolutionized genetic research and brought unforeseen capabilities also to
studies of structural variation. In fact, these capabilities are not yet completely satisfactory
(as described in \cref{sec:sv_detection}) and ongoing
development of new technologies and protocols is continuously pushing the
boundaries of what is possible. Since DNA sequencing takes such a prominent
position in my research---virtually every experiment in this thesis includes
sequencing---the major techniques shall be introduced here.




\subsection{Massively parallel sequencing}
\label{sec:mps}

The foundation of modern DNA sequencing technologies was laid in 1977 by Frederick Sanger
and his ``chain termination'' technique \citep{Sanger1977}. Despite not being
the first DNA sequencing method, the chain termination method brought
unprecedented ease of use and accuracy \citep{Heather2016}. It is based on DNA
polymer extension via DNA-dependent DNA polymerase (i.e. replication) and the
incorporation of dideoxynucleotides, which stop polymerization. With subsequent
electrophoresis, partially replicated DNA fragments can be ordered by length and
nucleotides identified based on radioactive or fluorescent labels.
Sanger sequencing was instrumental in the Human Genome Project and, owing to the
accuracy and length (around 1~kb) of sequenced fragments, it is still used
today for validation purposes. It sequences single DNA fragments, though, and
is thus laborious to apply in a larger scale.

The throughput could be dramatically increased with the advance of \acf{mps},
which is also referred to as short-read sequencing, next-generation sequencing,
2\textsuperscript{nd}-generation sequencing, or high-throughput sequencing.
Different commercial techniques were brought forward in the first decade of this
millennium, including Pyrosequencing by 454 Life Sciences, Sequencing by
Oligonucleotide Ligation and Detection (SOLiD) by Applied Biosystems,
Nanoball Sequencing by Complete Genomics, Helicos Single Molecule Fluorescent
Sequencing by Helicos BioSciences, and the Reversible Terminator Chemistry by
Solexa \citep{wikiDNAseq2018}. Today, the market for \mps technologies is vastly
dominated by Illumina, who acquired Solexa and their technology in 2007. Here,
the core principles of Illumina DNA sequencing shall be described representative
for \mps in general.

\paragraph{Key principles of parallel DNA sequencing}
Like the Sanger technique, Solexa/Illuminaâ€™s approach relies on the concept of
sequencing by synthesis, i.e. by replication through a DNA polymerase. It in
fact also utilizes the incorporation of fluorescently labeled dideoxynucleotides,
which initially terminate the polymerization. A major novelty, though, is that
the fluorescent label can be removed and the $3^\prime$ hydroxyl group of the
dideoxynucleotide chemically restored. This technique is widely known as
reversible terminator chemistry \citep{Turcatti2008}. DNA is then replicated
step by step, in each of which the incorporated nucleotides are detected using
fluorescent imaging. This concept of cyclic DNA synthesis followed by
fluorescent detection is shared by multiple of the aforementioned techniques,
which use slighly different molecular mechanisms \citep{Shendure2008}.

Another key concept of \mps technologies is a step of clonal amplification of
DNA fragments in order to enhance the fluorescent signal detection. DNA
fragments are initially ligated to adapter sequences and then, in case of
Solexa/Illumina, immobilized on the flow cell and amplified via
\explain{\acf{pcr}}{A method for amplification of DNA fragments. Did you know
    that \acs{pcr} was invented only after Sanger
    sequencing, in 1983? See \citet{Mullis1990} for a brief history}
    \citep{Mullis1990}, which they call bridge amplification.
The aspect of parallelism comes into play when many (up to millions of) local
clusters, each with a clonally amplified DNA fragment, are observed
simultaneously during nucleotide incorporation. This again was driven by
technological advances in high-resolution cameras, notably based on
charge-coupled devices \citep{Barbe1975,Shendure2008}. Due to clonal
amplificaiton and high-resolution imaging, Illumina machines can sequence DNA
with extremely high accuracy, with a per-base error rate is in the order
of 0.1\% \citep{Fox2014}.

\paragraph{Applications}

In contrast to Sanger sequencing, which targets a single locus, \mps can be
applied to perform \emph{\acf{wgs}}. During \wgs, DNA is highly fragmented prior
to the construction of a sequencing \emph{library}, which is then sequenced via \mps to
yield a large set of \emph{sequencing reads} from
all over the genome. Due to the random fragmentation this approach is also
commonly known as shotgun-sequencing \citep{Weber1997}. In a typical
\emph{re-sequencing} experiment, where a species with available reference genome
is sequenced, these reads are then mapped to the reference for further analysis
such as, for example, variant detection.

Apart from \wgs, a large number of
other sequencing protocols exist that utilize \mps to study different molecular
characteristics\footnote{see \citet{Pachter2018} for a list of such protocols}.
A prominent example is RNA-seq \citep{Morin2008,Wang2009},
which makes the mRNA present in cells available to sequencing by
reverse-transcription into cDNA. We used this technique in \cref{sec:balancer}.
\Cref{sec:ccc,sec:strandseq} cover two other protocols based on \mps that are
of particular interest in this work.

\paragraph{Paired-end sequencing}

\figurepagewidth[t]{reads.pdf}{reads}{MPS sequencing reads}{Sequencing reads
    from \mps are typically short (for example 100~bp), are sequenced from
    their $5^\prime$ ends and can be single-ended or paired-end. In
    paired-end or mate pair libraries, the lengths of the original DNA fragment
    (which can be estimated after read mapping) is called insert size and
    usually much larger in mate pair experiments than in paired-end experiments.}

Modern sequencing machines offer the possibility to sequence a DNA fragment
from both ends. In case of Illumina/Solexa, this is achieved by a special step
of bridge amplification that anneals the free end of all fragments in a clonal
cluster to the surface and then frees the initially attached ends. Afterwards,
sequencing continues in opposite direction to capture the other end of the
DNA fragments. This approach is called \emph{paired-end} sequencing or
paired-end tag sequencing and was used early on to study structural variants
\citep{Campbell2008} (see also \cref{sec:sv_detection_ngs}). Using pairs, more
bases can be sequenced at high quality than could be with a single read.
Typically, the DNA fragments subject ot paired-end sequencing have a length of
up to 500~bp. Larger fragments (typically around 3~kb, but up to 10~kb) can be
achieved by creating \emph{mate pair} libraries (a.k.a jumping libraries). In the
approach of mate pair sequencing, a longer DNA fragment is first circularized before
the connection of both ends is sequenced either single-ended or in paired-end mode
\citep{Korbel2007}. The size of the underlying DNA fragment is called
\emph{insert size}, and the number of sequenced bases \emph{read length}
(\cref{fig:reads}). Paired-end and mate pair sequencing have played a pivotal
role in \sv detection, which is elaborated later in this introduction.

\paragraph{Sequence analysis}
After DNA sequencing, the computational analysis of the obtained sequencing
reads begins. Naturally, this analysis may be very different depending on which
protocol was used. For a \wgs resequencing experiment, a very common first step
is to assign the short reads to their most likely origin and
\explain{read orientation}{Only one DNA strand (e.g. the $5^\prime$ strand) is
    encoded in a reference genome, but fragments from both strands are sequenced.
    Thus, also the reverse complement sequence of each read must be mapped---we
    say they are mapped to the \emph{minus} strand (\cref{fig:reads})} within the reference
genome. This process is called \emph{read mapping} or \emph{read alignment}, and
software tools to perform this task are abundant
\citep{Li2009,Weese2009,Langmead2009,Alkan2009,Li2013}. The intricacies that
hamper read mapping are sequencing errors and the repetitive nature of large
genomes, which do not allow unique placement of reads in many regions. These
regions are said to have low \emph{mappability} and are difficult to deal with---often they are simply neglected.
Subsequent to read mapping, downstream analyses can be carried out such as \sv
detection, which I describe in \cref{sec:sv_detection}.

The very popular paired-end sequencing allows the mapping of a read to be
informed by the placement of the second read. For example, when one read maps
ambigously, it can somtimes be ``anchored'' by the second, uniquely mapping read.
Paired-end reads, which are sequenced from their $5^\prime$ ends towards one
another, align in convergent orientation to the reference genome, whereas mate
pairs align in divergent orientation (\cref{fig:reads}).

An alternative to read mapping is \emph{de novo} assembly, which exploits the
relationship of sequencing reads to one another (i.e. a common subsequence) to
restore the sequenced genome independent of a reference. These methods, however,
typically fail to produce long consecutive sequences \citep{Alkan2011_assembly}.










\subsection{Emerging long read sequencing technologies}
\label{sec:long_read_seq}

Over the last years, new sequencing technologies have been developed which are
commonly referred to as ``3rd-generation sequencing'' in the community. These
technologies are fundamentally different to \mps technologies in that they avoid
clonal amplification of DNA fragments, but sequence single molecules instead.
Although single molecule sequencing had been feasible already earlier
\citep{Braslavsky2003}, later commercialized by Helicos BioSciences,
third-generation sequencing is commonly associated only with the techniques of
\acf{pacbio} and \acf{ont}. In contrast to the Helicos platform, these
techniques achieve significantly longer read lengths (up to more than 100~kb)
at usually decreased accuracy. Unlike Sanger's technology, they still sequence
many molecules in parallel.

\paragraph{\acl{pacbio}}
Like Illumina \mps, The technology of \acl{pacbio} relies on the concept of sequencing by synthesis using
fluorescently labeled deoxynucleotides. However, the sequencing
occurs in real time on single molecules and was hence termed
\emph{Single-Molecule Real-Time sequencing (SMRT)} \citep{Eid2009}. Fluorescent
image detection occurs not in cycles, but continuously (acquiring \emph{movies} instead of images).
Besides deciphering the order of nucleotides, this principle also measures
kinetics of the polymerase, allowing the detection of modified bases such as
methylated cytosine \citep{Flusberg2010}.

The major challenge in the development of SMRT sequencing
was the detection of the fluorescence signal from a single nucleotide upon
ligation. To this end, researchers had engineered microplates with so-called
zero-mode wave guides \citep{Uemura2010}. The laser for excitation of
fluorophores only illuminates the bottom of these nanowells, in which the
polymerases are deposited. This way, only the nucleotides that are actively
being incorporated by the polymerase can be excited and detected
against a background of fluorophores outside the well \citep{Heather2016} (see
\cref{fig:third_gen} A). The base calling in single molecules is still noisier than
detection in clonally amplified sequences, though, leading to reported per-base-error
rates of 11-15\% \citep{Rhoads2015}.

To improve accuracy, \pacbio researchers
promoted a technique called \emph{circular consensus sequence (CCS)}
\citep{Travers2010}. Here, a double-stranded DNA fragment ligates to hairpin
adapters on both ends to form a circular DNA molecule (the sugar phosphate
backbone is one covalently bound ring), which still preserves
its double-stranded structure. The polymerase can then pass this ring of DNA
repeatedly, effectively sequencing the same fragment multiple times. This long
read is computationally divided into sub-reads and a consensus is formed with
reported accuracies of up to more than 99\% \citep{Rhoads2015}.

\figuretextwidth[t]{third_gen_seq_Heather2016.png}{third_gen}{Concepts of
    \texorpdfstring{\acs{pacbio}}{PacBio} and \texorpdfstring{\acs{ont}}{ONT}
    sequencing}{
    \subpanel{A} DNA polymerase inside a zero-mode waveguide processes a DNA
    molecule. The fluorescently labeled nucleotides are preferably illuminated
    at the bottom of the well, distinguishing their signal from the pool of
    nucleotides in the solution.
    \subpanel{B} Nanopore sequencing by passaging of a single-stranded DNA
    polymer through a pore in a non-conductive layer. Movement of the DNA
    molecule is facilitated by a voltage across the layer and decelerated by a
    processive enzyme (upper ellipse), e.g. a helicase that unwinds the double
    stranded DNA.
    Figure taken from \citetitle{Heather2016} \citep{Heather2016} licensed under
    \acl{ccby4}.}


\paragraph{\acl{ont}}
\Acl{ont} utilize a fundamentally different approach to sequence single
molecules. Driven by electrophoresis, a single-stranded DNA passes through a
tiny pore that can detect changes in the ionic current specific to the type of
nucleotide passing through. The development of this technique spanned three
decades and is based on findings from multiple labs, as \citet{Deamer2016}
nicely portrayed. One major step towards the current technology was to find an
appropriate pore that is just wide enough for a single-stranded DNA molecule to
pass through: Initially \textit{Î±-hemolysin} channels from
\textit{Staphylococcus aureus} had been used, which were later replaced by a
genetically modified version of \textit{MspA}, a porin from
\textit{Mycobacterium smegmatis} that allowed a much better signal-to-noise
ratio \citep{Butler2008}. A second crucial step was to decelerate the passage of
each single nucleotide through the pore in order to allow accurate measurements
of currents. While DNA polymers would naturally pass a pore at a rate of $10^{-13}$ sec
per nucleotide even for the smallest voltages, researchers had the idea to place
processive enzymes in front of the pore that slightly pause the passage of each
nucleotide \citep{Deamer2016} (\cref{fig:third_gen}, B).

\Ac{ont} offered the first commercially availabe
nanopore sequeqncing devices, named MinION, in an early access program from 2014.
Their technology utilizes an ATP-dependent helicase enzyme that unwinds double-stranded DNA
before passaging the pores, with detection happening simultaneously at up to
512 pores \citep{Jain2015}. A hairpin adapter is ligated to DNA fragments, so
that after a single strand has passed the pore the complementary strand
(linked covalently via the hairpin adapter) follows. This sequences the same
fragment twice and is utilized to improve sequencing quality. These corrected reads
are called \emph{2D reads} and were reported to have an improved error rate of
15\% \citep{Jain2015}. Through the MinION early acces program our laboratory
gained access to MinION devices, enabling us to apply the technique in the scope
of the research project describe in \cref{sec:complex_invs}.






\subsection{Chromatin conformation capture sequencing}
\label{sec:ccc}

Studies of the three-dimensional structure of \textit{in vivo} chromatin have for a long
time been limited to imaging-based approaches. \citet{Dekker2002}, however,
proposed a new technique named \emph{chromatin conformation capture (3C)} that
probes the three-dimensional distance of loci using genomics methods. The basic
idea is to crosslink DNA with itself in regions that are in close spatial
proximity. The results of such experiments give insight into the \emph{contact
frequency} between two loci relative to other loci, which can be interpreted as
an average three-dimensional distance between the loci observed across many
nuclei.

The original 3C protocol relies on targeted \pcr amplification and is
capable of testing the interaction between exactly two loci. However, multiple
protocols based on 3C have recently been shown to increase the level of
parallelism, including 4C \citep{Zhao2006,Simonis2006} and 5C \citep{Dostie2006}.
Eventually, the combination of chromatin conformation capture and \mps resulted
in \acf{hic} \citep{Lieberman-Aiden2009}. The core principle of \hic, as shown
in \cref{fig:ccc}, involves cross-linking and digestion of DNA and then
submitting cross-linked loci to paired-end \mps. After filtering of data (e.g.
removing fragments ligated to themselves) both reads of a pair represent two
loci that were in close three-dimensional proximity within the nucleus.
Principally, the unbiased contact frequency of all loci against all other loci
can be measured in this way. In practice, the resolution of these two-dimensional
contact maps strongly depends on sequencing coverage. For example, to achieve
1~kb resolution---the densest maps known to date---\citet{Rao2014} required 4.9
billion pairwise contacts in a human cell line.

\figuretextwidth[t]{chrom_conf_cap_Li2014_CC.jpg}{ccc}{Chromatin conformation
    capture technologies}{
    Chromatin conformation capture relies on cross-linking of DNA, enzymatic
    digestion and subsequent ligation of cross-linked loci. In the 3C-protocol,
    targeted interactions can be analyzed via \pcr. The \hic protocol enriches
    for cross-linked fragments based on biotin labels before these fragments
    are paired-end sequneced. Figure modified from \citetitle{Li2014}
    \citep{Li2014}  licensed under \acl{ccby4}.}

\hic was designed and used to study chromatin conformation and specific
three-dimensional features, as for example DNA loops forming between enhancers and
promoters. It further revealed previously unknown structural features of the
genome, which are discussed in more detail in \cref{sec:balancer_hic_svs}.
However, the characteristics of \hic contact maps predestinate them for at least
two additional use cases: Intra-chromosomal interactions are more frequent than
inter-chromosomal interactions, which is in concordance with the theory of
chromosomal territories \citep{Cremer2001}. Thus, \hic data can by utilized to
cluster genomic loci by chromosome, which is especially relevant for \textit{de
novo} assembly \citep{Burton2013}. Another observation is that the contact
frequency between loci typically decays quickly with increasing genomic
distance, meaning that the highest signal is detected between loci that are also
close in linear genomic proximity. Because of that, larger genomic
rearrangements become prominent in contact maps---this is what I utilized
to characterize genomic rearrangements in \cref{sec:balancer}.





\subsection{Strand-seq}
\label{sec:strandseq}

Strand-seq is a single-cell DNA sequencing protocol that preserves the identity
of homologues by sequencing only the template strand of each chromosome
\citep{Falconer2012,Sanders2017}. The readout are sequencing reads, obtained via
\mps (e.g. on an Illumina platform) in either paired-end or single-ended mode,
which all map in the same orientation (plus strand or minus strand) to a
reference genome if they originated from the same homologue\footnote{In paired-end
sequencing, the first reads of all pairs map in the same orientation.}. In cells where
homologues are inherited on opposite strands, which we call \emph{Watson (W)} and \emph{Crick (C)}
strands, the original homologue for each read (including potential variants
captured by this read) can be determined simply based
on its mapping orientation. Strand-seq can thus reliably phase (i.e. distinguish
haplotypes) chromosomes in their full length \citep{Porubsky2016}. Moreover,
given the consistent directionality of sequencing reads across a homologue,
Strand-seq allows to detect (large) inversions in respect to the reference
genome \citep{Sanders2016}. Strand-seq was futher used to study \acl{sce} events
(introduced in \cref{sec:mosaic_mc}) \citep{Falconer2012}.

\figuretextwidth[t]{strand_seq.pdf}{strand_seq}{Strand-seq principle}{
    \subpanel{A} Diploid cells (here schematically represented only for a single pair of
    chromosomes) contain maternal (M) and paternal (P) homologous chromosomes,
    each of which is a double-stranded DNA molecule. Watson (W) and Crick (C)
    are highlighted in orange and green. After replication in the presence of
    BrdU instead of Thymidine (1), cells contain two sister chromatids of each
    homologous chromosome, each with a different strand labeled (dotted line).
    After cell division, photolytic nicking of BrdU sites and library
    preparation (2), only sequencing reads from the non-labeled strand remain.
    \subpanel{B} A daughter cell inherits the two homologues in one of
    four different constellations, i.e. both as W (top left), both as C (top
    right), or both as different strands (maternal W, bottom left, or paternal
    W, bottom right). After read mapping cells appear in WW, CC, or WC
    configuration (circles). Figure modified from \citetitle{Sanders2016}
    \citep{Sanders2016} licensed under \acl{ccby4}.}

Strand-seq requires actively replicating cells, which are grown for a single
round of replication in a $5$-Bromo-$2^\prime$-deoxyuridine (BrdU) medium.
The incorporation of this thymidine analog into the newly synthesized DNA strand
is the basis for obtaining stranded sequencing libraries, because after cell
division each daughter cell will have only one strand labeled with BrdU
(\cref{fig:strand_seq}). After cytokinesis, the DNA of a daughter cell is
enzymatically digested into fragments using a micrococcal nuclease enzyme
(MNase). These fragments are ligated to sequencing adaptors. BrdU-containing
fragments are then degraded via photolytic cleavage, so that
subsequently only non-BrdU-containing fragments are amplified via \pcr.
Fragments from each single cell are tagged with cell-specific barcodes and are
finally sequenced simultaneously, typically 96 cells together in one Illumina
lane \citep{Sanders2017}.

In \cref{sec:mosaicatcher}, I go into more depth on the workings of Strand-seq,
notably on the computational analysis of Strand-seq data. I then present a novel
approach that utilizes Strand-seq libraries across multiple single cells to
detect mosaic \acp{sv}.


