\chapter{Supplementary information to \texorpdfstring{\cref{sec:balancer}}{the balancer project}}
\label{sec:suppl_balancer}


The content of this supplementary chapter is largly taken from the supplementary
methods in \textbf{our paper}\todo{cite our paper} and partly adapted when
neccessary.

\section{SNV calling}
\label{sec:suppl_snv}

Both \wgs and mate pair sequencing data was mapped to
\ac{dm6} using \bwamem version 0.7.15. \Ac{snv} and short indel calling was
performed using \freebayes version v0.9.21-19 with disabled population priors on
the \wgs data of both $F_0$ and $F_1$ samples simultaneously. The results were
filtered using \vcflib based on a quality value of at least 30,
a minimum of at least two reads carrying the allele to the right and to the left
end, and on the fact that the allele was seen on at least two reads mapping in
each direction. We further normalized variants, removed mutli-allelic variants,
and decomposed multi-nucleotide substitutions (which are reported as haplotype
blocks by \freebayes) into \acp{snv} using \vt (the sub-command \textsc{decompose\_blocksub}
was used for decomposition). We finally remove contigs other than chromosome 2,
3, and X and obtained a total of 860,095 \acp{snv} and small indels.



\section{Mutational signature analysis}
\label{sec:suppl_mutsign}

Starting from the set of 520,521 balancer- or wild type-specific \acp{snv},
I removed the ones which are present in the DGRP freeze 2.0 \snv call set.
Then I used the R package \textsc{SomaticSignatures} \citep{Gehring2015} to
count base substitutions and their contexts of the remaining 58,457 variants
and plotted their relative frequencies in \cref{fig:signatures}. The absence
of striking differences between balancer and wild type spectra demotivated me
from deeper investigations of mutational signatures.

\figuretextplusmargin{snp_signatures_dgrp_removed.pdf}
    {signatures}
    {\Ac{snv} mutation spectrum}
    {Frequency of the different
     base substitutions in their three-nucleotide context for balancer- and
     wild type-specific \acp{snv}. \Acp{snv} that are found in DGRP were
     removed, leaving 58,457 variants.}




\section{Deletion calling}
\label{sec:suppl_del}

I used \delly version 0.7.2 on the \wgs data of the $F_0$ and $F_1$ data
simultaneously and applied an extensive filtering procedure to reduce the number
of false positive calls. From the initial 10,421 deletion calls, 5,150 dropped
out because they were not flagged as ``QC PASS'', were not on one of the main
chromosomes (\ac{chr2}, \ac{chr3}, or \ac{chrX}), had a mapping quality value of
less than 60 or did not match the expected genotypes (i.e. balancer-specific,
wild type-specific, and common - together constituting more than 90\% of the calls).
Furthermore I required a minimum number of supporting read pairs for reference
and alternative allele combined, namely 40 read pairs for ``imprecise'' \delly
calls and 25 split reads for breakpoint-precise \delly calls.

Next, I developed a dynamic read depth ratio filter that was applied to deletion
predictions of 160~bp or larger.
To this end, the read count within the predicted deletion was normalized by the
summed read count in size-matched intervals flanking the locus and these values
were compared between samples. I required a minimum difference in the read depth
ratio between samples with different genotypes and this threshold increases
dynamically with \sv size. This is motivated by the fact that for larger deletions
the average read depth signal is more robust against local fluctuations in coverage.
To give an example, this filter removed a number of predictions above 100~kb in
size, which could be clearly identified as false positives by inspecting additional
(e.g. Hi-C) data. At last I overlapped deletions with a mappability map to
classify them into high (at least 50\% in a uniquely mappable region) or
low-confidence loci. Eventually we obtained four call sets: 3,072 calls with
high-confidence and below 50~bp, 737 calls with high confidence and from
50~-~159~bp, 395 large calls with high confidence and 75 large ones with low
confidence.

As a validation \yad performed \pcr on randomly selected loci in the latter three
categories. I designed primers using a lab-internal extension to \primerthree
and \yad amplified 25 loci per category in both samples via \pcr.
In the size range 50-159~bp 24 out of 25 loci validated, also 24/25 loci
validated for high confidence calls of 160~bp, and 25/25 loci validated for
low-confidence calls, yielding an estimated FDR of 2.66\%.
At last we merged the set of \delly deletion calls into the set of small
deletions called by \freebayes and chose a lower size cutoff of 15~bp. During
the merging process \freebayes calls were given priority over matching \delly
calls (based on 50\% reciprocal overlap). The final data set (referred to as
``deletions'' in the main text) contains 8,340 deletions on chromosomes 2, 3 and
X.




\section{Duplication calling and filtering}
\label{sec:suppl_dup}

\figuretextwidth{dup_validation_example_wt.pdf}{dup_validation_wt} % DUP00014221
    {Example for a wild type-specific duplication}
    {Wild-type specific tandem duplication at locus
    \textit{chr3R:26,960,008−26,964,205}. The bi-allelic frequnency signal
    clearly suggests a heterozygous duplication in the $F_1$ cross and the
    increased read depth in the wild type sample identifies it as present on
    the wild type chromosome.}

\figuretextwidth{dup_validation_example_false.pdf}{dup_validation_false} %
    {Example for a false duplication prediction}
    {Predicted duplication locus \textit{chr3R:22,101,264−22,107,086} is not
    validated by the bi-allelic frequncy signal.}

I used \delly version 0.7.5 in tandem duplication mode and supplied both mate
pair and \wgs libraries for $F_0$ and $F_1$ samples simultaneously. Duplication
calls were initially filtered by the ``quality PASS'' criteria reported by
\delly and by their combined genotypes, which were required to be heterozygous
in the $F_1$ sample. We do not require homozygosity in the $F_0$ sample due to a
known issue of the duplication classifier, which reports many homozygous tandem
duplications as heterozygous. For all remaining 352 calls I generated detailed
overview plots that contained multiple lines of information: a total read-depth
track, a mappability track, overlapping gene annotations and, importantly, a
track showing the bi-allelic frequency measured at SNV positions. These plots
allowed me to sort out false positives, leaving 122 manually curated
high-quality tandem duplications. Aside from tandem duplications I further
inspected the bi-allelic frequency ratio across the genomes and unraveled three
non-tandem duplications of 4.3~kb, 10.4~kb and 258~kb size. Both sets together
are summarized by ``duplications'' in the main text.





\section{Breakpoints of the balancer chromosomes}
\label{sec:suppl_balancer_breakpoints}

\begin{table}[ht]
    \centering
    {\small
    \begin{tabu}{llrrrl}
        \toprule
        Bal      & Chr & $5\prime$ bp. & $3\prime$ bp. &        Del/Dup & Genes affected  \\
        \midrule
        \cyo     & 2L    & 2,137,075   & 2,137,067   & +9               & GlyP: $3\prime$ UTR      \\
                 & 2L    & 12,704,657  & 12,704,649  & +9               & nAChRalpha6: intron \\
                 & 2L    & 9,805,567   & 9,805,575   & -7               & CG5776, spict     \\
                 & 2R    & 14,067,771  & 14,067,782  & -10              & Src42A: $3\prime$ UTR     \\
                 & 2R    & 6,012,739   & 6,012,459   & +281             & Prosap: intron    \\
                 & 2R    & 21,971,918  & 21,972,072  & -153             & -                 \\
        \midrule
        \ac{tm3} & 3L    & 6,925,034   & 6,926,125   & -1,090           & -                 \\
                 & 3R    & 9,943,831   & 9,944,040   & -208             & Glut4EF: exon/$3\prime$ UTR \\
                 & 3L    & 15,150,269  & 15,150,272  & -2               & FucTA: intron     \\
                 & 3R    & 23,050,763  & 23,050,764  & 0                & p53: intron       \\
                 & 3L    & 19,386,273  & 19,388,151  & -1,877           & GC32206: intron/exon \\
                 & 3R    & 20,637,930  & 20,637,930  & +1               & Lrrk: exon        \\
                 & 3L    & 22,637,876  & 22,637,952  & -75              & CG14459: $3\prime$ UTR    \\
                 & 3R    & 31,653,695  & 31,653,707  & -11              & kek6: $5\prime$ UTR       \\
                 & 3R    & \textit{20,308,200} & \textit{20,325,700} & \textit{-17,500} & CG42668, CG42668  \\
                 & 3R    & -           & -           & -                & unknown           \\
        \bottomrule
    \end{tabu}
    }
    \tabcap{blancer_breakpoints}{Breakpoint positions of balancer chromosomes}{
        Breakpoint positions of the major rearrangements of both balancer
        chromosomes (Bal) were identified in Hi-C data and refined based on their
        paired-end signature uncovered by \delly. Values in italic could not be
        resolved at the base pair level. Between the $3\prime$ and $5\prime$ ends
        of the breakpoints deletions or duplications can have occurred:
        Deletions are states as negative integers, duplications as poitive
        integers (Del/Dup).
    }
\end{table}



\section{Detail on \texorpdfstring{\ase}{ASE} detection using DESeq2}
\label{sec:suppl_deseq}
The main idea is explained in \cref{sec:ase_impl}. Haplotype-resolved fragment
counts were calculated using \textsc{HTSeq-count} \citep{Anders2015}.
In the main analysis we then tested genes for \ase by inserting these
haplotype-specific counts of all four replicates (2x \Nmat, 2x \Npat) into a
matrix and supplying it to \deseq. Genes were filtered for a minimum number of
reads (average of 50 fragments per gene per sample) and by chromosome (only 2L,
2R, 3L, and 3R were considered). \deseq was tested with a design
\texttt{\footnotesize \textasciitilde Replicate + Haplotype}. The resulting p-values were corrected using
\textsc{FDRtool} \citep{Strimmer2008}.



\section{Further \texorpdfstring{\ase}{ASE} related analyses}
\label{sec:suppl_ase_more}

\figuretextwidth{ASE_genes_around_bps.pdf}{suppl_ase_genes_around_bps}
    {Number of \texorpdfstring{\ase}{ASE} genes around the breakpoints of large
    chromosomal rearrangments}{This plot shows the number of significant \ase
    genes at given distances to one of the large rearrangments, or to random
    positions in the genome (quantiles of 500 random samplings are shown).}





\section{Hi-C analyses}
\label{sec:suppl_hic}
