\subsection{Traditional SV discovery}
\label{sec:sv_detection_old}

A traditional way to investigate the karyotype of cells is by arresting cells in
metaphase, staining chromosomes and observing them under a microscope
\citep{Speicher2005}. The images of each chromosome are then ordered by
chromosome to show an ideogram. By counting the number of each chromosome in the
cell, gross abnormalities such as aneuploidy can be detected, which is
schematically depicted under aneuploidy in \cref{fig:SV_classes}. Relevant
disease-linked forms of aneuploidy could be unraveled early on this way, e.g.
the trisomy of chromosome 21 \citep{Lejeune1959}. The classic technique has been
refined to allow higher specificity and resolution, mostly via improved ways of
staining, such as by quinacrine staining, Giemsa banding, or chromosome-specific
labeling based on in situ hybridization \citep{Speicher2005}. These techniques
principally allow the detection of \acp{cnv}, inversions and translocations,
yet the size range of these \acp{sv} has to be in the order of several Megabases
or larger. Fluorescence in situ hybridization \citep{Bauman1980}, which relies
on the annealing of fluorescently labeled DNA probes to their complement DNA,
is also applied in a targeted manner for validating predicted SVs of slightly
smaller size.

Other means of detecting \acp{sv} include optical mapping
\citep{Schwartz1993,Teague2010} and hybridization-based microarrays. The latter
ones, which are reviewed in \cite{Alkan2011}, used to be the dominating method
for \cnv detection before high-throughput sequencing became standard. One of the
two major techniques in this category, namely array comparative genomic
hybridization, utilizes the competition of the test sample's DNA and a reference
DNA to a hybridization probe (e.g. short oligonucleotides) to infer the relative
copy number of the tested locus \citep{Snijders2001}. Using high-density arrays,
this method can successfully detect deletions down to 500~bp in size
\citep{Conrad2010}. SNP arrays, on the other hand, utilize hybridization probes
at sites of polymorphic \acp{snv} to measure the allelic ratio within a single
sample, which is called \emph{\acf{baf}}. This way, \acp{cnv} but also \loh can
be detected.





\subsection{SV discovery in the era of massively parallel sequencing}
\label{sec:sv_detection_ngs}

Today, aforementioned techniques have largely been superseded by \sv detection
utilizing \acl{mps} data.
% although especially microarrays remain a cost-effective alternative.
\Ac{sv} detection methods based on \mps are often (also by \cite{Alkan2011})
separated into four different conceptual approaches, namely read pair analysis,
split-read analysis, read depth analysis, and sequence assembly. In practice,
\sv prediction tools do not necessarily fit into only one of these categories.
Below I will summarize the major ideas behind the different approaches as well
as representative software implementing them.

\paragraph{Paired-end analysis}
\Ac{sv} detection based on paired-end sequencing utilizes the orientation and
expected distance of two sequencing reads to another to detect rearrangements.
For instance, when their mapping distance on a reference genome is larger than
expected a deletion may have occurred in the test sample anywhere in between
those reads. Paired-end analysis can principally detect many different types of
\acp{sv}, including \acp{cnv}, inversions, translocations, insertions and
\acp{mei} (\cref{fig:SV_detection}, B). This technique had first been used on
bacterial artificial chromosome
\citep{Volik2003}, on fosmid libraries \citep{Tuzun2005} and then on mate pair
sequencing of human genomes \citep{Korbel2007}. It is nowadays one of the
dominating principles of \sv detection and is implemented in well-established
software tools such as \textsc{BreakDancer} \citep{Chen2009}, \delly,
\textsc{CLEVER} \citep{Marschall2012}, or \textsc{LUMPY} \citep{Layer2014}.
Besides the richness in detectable SV classes, an advantage of the paired-end
read signature is that it can identify the breakpoints of an \sv. The breakpoint
accuracy depends on sequencing coverage and insert size distribution, but
it is typically in the range of few to several hundred base pairs.

\paragraph{Split-read analysis}
Split-read approaches utilize the fact that sequencing reads, if long enough,
can be divided and separately assigned to different locations of the
reference assembly  (\cref{fig:SV_detection}, C). This is different from intra-read
gaps or mismatches, which are still tolerated within each alignment and
which are typically used to detect \acp{snv} and small indels.
Based on the position and orientation of the partial alignments, split-read
analysis detects \acp{sv} in the same way as paired-end analysis. The major
advantage, though, is that breakpoints can be determined much more accurately,
often down to the exact nucleotide.
Split-read approaches had been explored, for instance, early on during the
1000 Genomes Project (\cref{sec:1000G}) on 400~bp single-ended reads
\citep{Zhang2011}. Tools such as \textsc{Pindel} \citep{Ye2009} or
\textsc{BreakSeq} \citep{Lam2010} were among the first ones to specifically
implemented the split-read approach. Nowadays, strongly encouraged by increasing
read lengths form standard MPS machines (e.g. up to 2~x~300~bp on an Illumina
MiSeq platform nowadays), read mapping software has been refined towards the
ability to directly perform split-read mapping, as exemplified by tools such as
the widely used \bwamem or specialized tools like \textsc{YAHA}
\citep{Faust2012} and \textsc{SplazerS} \citep{Emde2012}.
This allowed other popular paired-end analysis detection tools to incorporate
the split-read approach, for example in \delly, \textsc{MATE-CLEVER}
\citep{Marschall2013}, and \textsc{LUMPY}.

\figuretextwidth[t]{SV_detection_methods_Tattini2015_new.png}{SV_detection}{
    Principles of SV detection in MPS data}{Schematic representation of the four
    distinct mechanisms for SV detection on three examples:  A deletion,
    an insertion (of duplicated sequence, for example), and an inversion.
    Figure modified from \citetitle{Tattini2015} \citep{Tattini2015} licensed
    under \acl{ccby4}.}

\paragraph{Read depth analysis}
A complementary method for detecting \acp{cnv} utilizes the total read depth
signal inside an \sv  (\cref{fig:SV_detection}, A). This resembles the
methodology of microarrays, yet with
improved resolution since all of the (mappable) genome can be covered instead
of selected loci. \textsc{SegSeq} \citep{Chiang2009} and \textsc{CNV-seq}
\citep{Xie2009} were among the first tools that utilized read depth in a
sample-vs.-control scenario to detect \acp{cnv}. \textsc{mrFast}
\citep{Alkan2009} and \textsc{CopySeq} \citep{Waszak2010} extended this
approach to single-sample \cnv calling, and later \textsc{CNVnator}
\citep{Abyzov2011} and \textsc{genome STRiP} \citep{Handsaker2015} gained more
popularity. Normalization of read depth is a major challenge in these approaches
owing to an uneven sequencing coverage, which is why these tools typically
perform best when an internal reference (e.g. a control sample) is available.
The currently popular population-scale \cnv caller \textsc{genome STRiP} even
suggests a minimum of 20 to 30 sample genomes in order to perform well. In
contrast to paired-end or split-read detection, the read depth method cannot
accurately predict breakpoints, which becomes a major disadvantage especially
for smaller variants. For very large variants, on the other hand, read-depth
methods can still robustly predict \acp{cnv} even when their breakpoints reside
in repetitive regions.

\paragraph{Sequence assembly} At last, sequence assembly-based methods do not rely on the
information provided by read mapping software but perform \textit{de novo}
assembly instead, as demonstrated for example by \citep{Li2011a}. A comparison,
e.g. via sequence alignment, of the sample sequence to the reference genome then
reveals the presence of \acp{sv}  (\cref{fig:SV_detection}, D).
While whole-genome assembly still remains computationally expensive
\citep{Bradnam2013}, there are methods that perform local re-assembly of reads,
for example \textsc{TIGRA} \citep{Chen2014b} or the recent \textsc{novoBreak}
\citep{Chong2017}. Especially for \sv types that are difficult to detect by
paired-end or split-read approaches, notably for insertion of novel DNA sequence,
assembly can yield a great benefit. Tools that address this are \textsc{NovelSeq}
\citep{Hajirasouliha2010}, \textsc{MindTheGap} \citep{Rizk2014}, and
\textsc{Basil/Anise} \citep{Holtgrewe2015}.

In practice, \sv detection using any of the four different approaches typically
requires an additional filtering step after initial \sv prediction. Such
filters can rely on the quality metrics provided by the prediction tool, but
optimally they involve an independent signal, such as read coverage for
paired-end predicted \acp{cnv}. One useful signal that shall be highlighted here
is \acl{baf}---this idea from microarrays is applicable to \mps data,
too. At sites of heterozygous \acp{snv} that reside within a putative \sv, the
sequencing reads supporting both alleles can be contrasted to infer the copy
number of the locus. Notably, in \cref{sec:balancer_cnv} this principle was
instrumental to validate predicted duplications.





\subsection{State of the art and limitations of SV studies}
\label{sec:limitations}

\Aclp{sv} in the human genome have been studied many times, driven by the
availability of new technology. Initial population studies mapped large \acp{cnv}
in several individuals using microarrays \citep{Sebat2004,Iafrate2004,Sharp2005,Redon2006}.
With refinements of these techniques, \cnv discovery could later be expanded to
hundreds of individuals and down to a detection size of 1~kb (or even 500~bp),
which scaled up the number of detected variants tremendously
\citep{McCarroll2008,Conrad2010}. Further improvements in the discernible size
range, the breakpoint accuracy and the types of SVs detectable were reached
with the application of \mps technologies. \cite{Korbel2007} and \cite{Kidd2008}
were among the first studies to utilize paired-end-like approaches to study
\acp{sv}, including balanced ones, in few individuals. A series of studies
followed that explored all the different technical approaches mentioned in
\cref{sec:sv_detection_ngs}.

The first phase of the 1000 Genomes Project presented the then most
comprehensive \sv call set, which was based on on low-coverage sequencing data
of 179 individuals \citep{Mills2011}. However, especially inversion detection
faced major limitations within the project, as I describe in more detail in
\cref{sec:complex_invs}. Further studies continued \sv characterization in the
human population (e.g. \citep{Sudmant2015a,Hehir-Kwa2016}) and in disease,
revealing thousands of copy number variable loci that are linked to pathological
phenotypes \citep{Swaminathan2012,Forbes2011}. \Acp{sv} were also mapped extensively in
cancer genomes \citep{Weischenfeldt2016,Campbell2017} and in other organisms such
as \textit{Drosophila melanogaster} \citep{Massouras2012,Zichner2013} or
\textit{C. elegans} \citep{Maydan2010}.

\figuretextwidth[t]{SV_limitations.pdf}{sv_limitations}{Examples for limitations
    of MPS-based SV detection}{Three examples for cases in which current
    \mps-based \sv detection methods fail. The upper half in each panel shows
    sample DNA carrying an \sv; below is shown where reads from that sample map
    to the reference assembly. \subpanel{A} A duplication of a repetitive element
    (shaded boxes) occurred, which is not detected because read mapping is masked
    within the repetitive region. \subpanel{B} An inversion flanked by repetitive
    elements. Because paired-end reads cannot be mapped uniquely inside the
    repeats, this inversion remains undetected. With an read length or insert
    size larger than the size of the repeats the inversion could be revealed.
    \subpanel{C} Insertion of a small piece of DNA into another chromosome leads
    to the prediction of a reciprocal trasnlocation. The fundamental principal
    behind this limitation is that standard \mps-based calling only detects the
    breakpoints of a copy-neutral rearrangement, but cannot reason about their
    inner state. This is different in a \cnv, for which the inner state (i.e.
    its read coverage) can be utilized for calling. The issue depicted here
    also arises for other \sv classes, notably inversions.}

Increases in sequencing depth improved breakpoint accuracy and sensitivity of
\sv detection, yet it did not overcome specific limitations owed to the
repetitive nature of the human genome. Notably, \acp{sv} attributed to \nahr are
known to be flanked by repeat sequence, in which read mapping (and consequently
paired-end \sv detection) often fails. This has been termed the ``short-read
dilemma'' \citep{Onishi-Seebacher2011}. Unfortunately, the human genome consists
to a large portion of repeats. Sequence analyses found that up to two third of
the human genome are derived from repetitive elements (mostly transposable
elements) \citep{DeKoning2011} with around 5\% of the genome containing large
(>10~kb) and highly identical segmental duplications \citep{Lander2001}.
Especially repeat-embedded inversions cannot be detected based on traditional
techniques (including \mps) ``at high throughput and high resolution''
\citep{Sanders2016}. \Cref{fig:sv_limitations} depicts three exemplatory
scenarios in which repeats confuse paired-end sequencing based \sv detection.

Challenges related to the repetitive nature of our genomes have been discussed
extensively in other areas, notably for de novo assembly \citep{Alkan2011_assembly}
and haplotype phasing \citep{Browning2011}. In both cases, blocks of information
(e.g. phasing blocks or contigs) fail to span through repetitive genomic regions.
These challenges are even greater in other species with more repetitive or
polyploidy genomes, which tend to also have of reference assemblies of lower
quality. Livestock and crop are two examples with outstanding environmental
relevance that suffer from these limitations \citep{Bickhart2014,Saxena2014}.

\todo{Bulk vs. single cell}
