Genomic \acfp{sv} are commonly defined as variants affecting more than
50 consecutive base pairs of the DNA. The main purpose of this definition is to
distinguish \acp{sv} from smaller indel variants or multi-nucleotide substitutions
(i.e. blocks of consecutive \acp{snv}) \citep{Alkan2011}. Indels are variants
(of up to 50~bp) that insert or delete nucleotides, which is the same situation
seen from different perspectives (hence the neologism \textit{indel}).
A more appealing definition than the arbitrary 50~bp threshold is that indels
are detectable inside a contiguously mapped DNA sequencing read (introduced in
\cref{sec:mps}) whereas \acp{sv} are detectable across alignments, yet also
this definition no longer fully applies in the light of novel long-read
sequencing technologies (\cref{sec:long_read_seq}). Fortunately, a clear
distinction is not biologically relevant. \Acp{sv} come in many different
flavors of which the major ones are described subsequently.

\Acp{sv} in the human genome are of particular relevance in health and disease.
For example, they cause the majority of genetic differences within the human
population are implicated in various  \citep{Weischenfeldt2013}.
However, this shall only be discussed later in \cref{sec:balancer_background},
as I present a study on the functional consequences of \acp{sv} in that chapter.




\subsection{Different classes of structural variation}
\label{sec:sv_classes}

\emph{\Acfp{cnv}} describe the focal loss or gain of genetic material. They are
termed \emph{imbalanced}, as they do not leave the balance of the two homologues
intact. A loss of DNA is called a \emph{deletion}, and a gain either
\emph{duplication}, \emph{triplication} or simply by its \emph{copy number}.
For example, a deletion has a copy number of one instead of the expected copy
number of two in a diploid organism. A duplication that arises on one of the
homologues leads to total copy number of three, and so on.
Duplications are in \emph{tandem} when the additional copy inserted in direct
proximity to the original locus instead of somewhere else in the genome. The
latter is referred to as \emph{interspersed} duplication (\cref{fig:SV_classes}).

The loss or gain of whole (or major parts of) chromosomes, historically visible
under under a microscope, is summarized as \emph{aneuploidy}.
Aneuploidy can range from a single chromosome (or at least the majority of the
chromosome) being lost or gained, up to a complete increase or decrease of the
ploidy level (of all chromosomes). The expected \explain{ploidy}{For humans, who
    are diploid, $N$ equals $23$---meaning we carry 46 chromosomes in our cells.
    Interestingly this number was falsely believed to be 48 for three decades
    before it was corrected by \cite{Tijo1956}.} in diploid organisms is
$2N$, where $N$ is the number of chromosomes and $2$ the number of homologous
copies. Ploidy can aberrantly increase to \emph{triploidy} ($3N$),
\emph{tetraploidy} ($4N$) or even higher states covered by the general term
\emph{polyploid}y. Cells can also be in a purely \emph{haploid} state ($1N$),
but they are rarely viable due to problems with chromosome segregation. Mixed
states, where only some chromosomes change their copy number or not all
chromosomes have the same copy number are referred to as hyperploidy
(\cref{fig:SV_classes}).

Other types of \acp{sv} do not change the total copy number of a locus. Notably,
\emph{inversions} reverse the orientation of a locus, but generally do not
include gains or losses (\cref{fig:SV_classes}). In fact, even an inversion can
introduce (or co-locate with) \ac{cnv} depending on its mechanism of formation,
which his is one of the major findings of \cref{sec:complex_invs}.
In cases where multiple \sv classes occur within the same allel we term them
\emph{complex}. The most prominent examples are \emph{inverted duplications},
which are duplications that insert in reverse orientation into the genome
(\cref{fig:SV_classes}). Nevertheless, non-complex, i.e. \emph{simple inversions}
are the prime example of balanced \acp{sv} as they re-structure the genome
without gaining or loosing genetic material.

\figuretextwidth{SVs.pdf}{SV_classes}{Types of structural variants}
    {Each case is depicted by the original locus on the left and the affected
    locus on the right, where dashed lines are used to highlight the orientation.
    \textit{Top:} Different types of focal \acp{sv} of a genomic locus (red)
    within double-stranded DNA (represented by grey line).  \textit{Bottom:}
    Chromosomes are depicted by double oval shapes. In the (balanced)
    translocation, chromosomes 12 and 17 are chosen exemplary to stress that
    exchange happens between non-homologous chromosomes. In the \loh event,
    though, the maternal and paternal homologue of the same chromosome are
    shown.}

Another class falling into the category of balanced \acp{sv} are
\emph{translocations}. For a translocation, genetic
material is exchanged between two non-homologous chromosomes. A \emph{reciprocal
translocation} is balanced because the total amount of genetic material does
not change, just the assignment of certain loci (potentially of whole arms)
to chromosomes.
But \emph{imbalanced translocations} can arise, too. Here, one chromosome remains
largely unchanged but a part of the homologue is duplicated and added to another
chromosome, which might itself loose genetic material at the same time. This can
involve whole chromosome arms, but also smaller loci---this is also covered by
the definition of translocation. Typically though, translocation refers to the
special case of a reciprocal translocation as shown in \cref{fig:SV_classes}.

Furthermore, when cells lack one of the two alleles of a larger genomic locus,
this is called a \emph{\acf{loh}}. \loh is an immdiate consequence of the
(partial) loss of a homologue, as for example happens in case of a deletion.
However, there are also copy-neutral \loh events in which the same haploid
genotype is present in two copies. This might for example occur when an
individual inherits two copies of a chromosome from one parent and none from the
other parent (uniparental disomy), but it can also occur via other mechanisms
(\cref{sec:mechanisms}). \Ac{loh} is often not observed directly, but indirectly
by looking at smaller variants (notably \acp{snv}) in a given genomic
region-â€“-the absence of heterozygous variants is an indicator of \loh.

Finally, various other forms of \acp{sv} exist that are of less relevance for
this work. One exception, which shall briefly be mentioned here, are
\emph{\acfp{mei}}. Mobile elements, notably \emph{transposons}, are DNA elements
that can ``jump'' within a host genome. The human consists to a large amount of
the remainders of such elements \citep{Haubold2006}, which are largely
prohibited from active transposition by repressive mechanisms in the host cell.
A \mei may occur in a cut-and-paste or a copy-and-paste fashion and, although
they principally resemble duplications or translocation, they are seen as
separate class due to the fundamentally different mechanisms of formation.





\subsection{Molecular mechanisms underlying the formation of SVs}
\label{sec:mechanisms}
